---
title: "KCinAD"
author: "Felicitas Pardow"
date: "2023-03-14"
output: html_document
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = TRUE)

hook_in <- function(x, options) {
    x <- x[!grepl("^#\\s+", x)]
    paste0("```r\n",
          paste0(x, collapse="\n"),
          "\n```")
}
# function that helps remove commented lines with a space (# test) from knitted Rmarkdown and only keeps commented lines without spaces or double-commended lines (#test, ## test)

#ctrl + shift + a makes selected code adhere to the tidyverse format

knitr::knit_hooks$set(source = hook_in)

library(SingleCellExperiment)
library(Seurat)
library(ggplot2)
library(dplyr)
library(openxlsx)
library(clusterProfiler)
library(org.Hs.eg.db)
library(ggvenn)
library(RColorBrewer)
library(progeny)
library(pheatmap)

library(ggbeeswarm)
library(slingshot)
library(SummarizedExperiment)
library(circlize)
library(rmarkdown)
library(tradeSeq)
library(decoupleR)
library(OmnipathR)

set.seed(1234)

# quick load
knitr::opts_knit$set(root.dir ="C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
roj_rds <-  readRDS("roj_rds.rds")
# basal3_dake <-  readRDS("basal3_dake.rds")

#colors
colors_disease <- c("HC" = "#228B22", "AD" = "#FF6347") 

colors_cellcycle <- c("G1" = "darkorange", "S" = "#66C2A5", "G2M" = "#FFD92F")

colors_subtypes <- c("basal III" = "#8DA0CB", "basal II" = "#66C2A5", "basal I" = "#FFD92F", "basal IV" = "#A6D854", "stratum spinosum" = "#FC8D62", "stratum granulosum" = "#E78AC3")

colors_subtypesBasIII <- c("basal III KRT15+" = "#6573A0", "basal III ASS1+" = "#B0B9E6", "basal II" = "#66C2A5", "basal I" = "#FFD92F", "basal IV" = "#A6D854", "stratum spinosum" = "#FC8D62", "stratum granulosum" = "#E78AC3")
```

# Setting up the workstream

When running this workflow for the first time, install the following R packages:

```{r installing, eval = F, echo = -(1:2) }
# # eval = F - does not run code
# # echo -(1:2) - only shows code from line 3 onward aka hides first 2 lines 
# 
# #install Rtools as described here:  https://cran.rstudio.com/bin/windows/Rtools/rtools40.html

# if (!requireNamespace("BiocManager", quietly = TRUE)) #checks if package BiocManager is not installed
#     install.packages("BiocManager") #if not: installs it

#biocmanager gives access to the Bioconductor Project Package Repository
# BiocManager::install("Seurat") 
# BiocManager::install("tidyverse")
# BiocManager::install("SingleCellExperiment") #update none
# BiocManager::install("clusterProfiler")
# BiocManager::install("openxlsx")
# BiocManager::install("org.Hs.eg.db")
# BiocManager::install("RColorBrewer")
# BiocManager::install("slingshot")
# BiocManager::install("ggvenn")
# BiocManager::install("limma")
# BiocManager::install("progeny")
# BiocManager::install("pheatmap")
# BiocManager::install("decoupleR")
# BiocManager::install("OmnipathR")
# BiocManager::install("ggpubr")
# BiocManager::install("tradeSeq")

```

I tried out some commands to get accustomed to single cell analysis with Seurat. 

```{r getting accustomed, echo = FALSE}
roj_rds

roj_rds@meta.data
table(roj_rds$svm_annotation)

DimPlot(roj_rds) 

basal3_dake@meta.data

objectrds@assays$progeny@meta.features

table(roj_rds$disease)

roj_rds[, roj_rds$disease == "HC"]

# Dimensional reduction plot for PCA or tSNE
DimPlot(object = pbmc, reduction = "tsne")
DimPlot(object = pbmc, reduction = "pca")

# Dimensional reduction plot, with cells colored by a quantitative feature
FeaturePlot(object = pbmc, features = "MS4A1")

# Scatter plot across single cells, replaces GenePlot
FeatureScatter(object = pbmc, feature1 = "MS4A1", feature2 = "PC_1")
FeatureScatter(object = pbmc, feature1 = "MS4A1", feature2 = "CD3D")

# Scatter plot across individual features, repleaces CellPlot
CellScatter(object = pbmc, cell1 = "AGTCTACTAGGGTG", cell2 = "CACAGATGGTTTCT")

VariableFeaturePlot(object = pbmc)

# Violin and Ridge plots
VlnPlot(object = pbmc, features = c("LYZ", "CCL5", "IL32"))
RidgePlot(object = pbmc, feature = c("LYZ", "CCL5", "IL32"))

# Heatmaps
DoHeatmap(object = pbmc, features = heatmap_markers)
DimHeatmap(object = pbmc, reduction = "pca", cells = 200)

# New things to try!  Note that plotting functions now return ggplot2 objects, so you can add themes, titles, and
# options onto them
VlnPlot(object = pbmc, features = "MS4A1", split.by = "disease")
DotPlot(object = roj_rds, features = c("LYZ", "CCL5", "IL32"), split.by = "disease")
FeaturePlot(object = pbmc, features = c("MS4A1", "CD79A"), blend = TRUE)
DimPlot(object = pbmc) + DarkTheme()
DimPlot(object = pbmc) + labs(title = "2,700 PBMCs clustered using Seurat and viewed\non a two-dimensional tSNE")

Idents(roj_rds) <- "svm_annotation"
DotPlot(object = roj_rds, features = c("KRT15", "ASS1"), split.by = "disease")
```

Renaming and renaming metadata categories.

```{r rename and order metadata}
# rename and reorder svm annotation
# Idents(roj_rds) <- roj_rds$svm_annotation
# roj_rds <- RenameIdents(roj_rds, "BASAL III" = "basal III", "BASAL II" = "basal II", "BASAL I" = "basal I", "BASAL IV" = "basal IV", "Spinosum" = "stratum spinosum", "Stratum Granulosum" = "stratum granulosum")
# levels(roj_rds) #check
# 
# roj_rds@meta.data[["svm_annotation"]] <- Idents(roj_rds)

roj_rds@meta.data[["disease"]] <- factor(roj_rds@meta.data[["disease"]], levels = c("HC", "AD")) #order cell identities
```

# Visualizations

```{r Visualization non-integrated}
object = roj_rds
ndims = 10

DefaultAssay(object) <-  "RNA"
object <- NormalizeData(object)
object <- ScaleData(object,verbose = F)
object <- FindVariableFeatures(object, selection.method = "vst", nfeatures = 2000, verbose = F) # nFeatures 2000 by default, selection method is default too 
object <- RunPCA(object, features = VariableFeatures(object = object), npcs = 50, verbose = F) # by default run on scaled variable features and 50 PCs
object <- FindNeighbors(object, dims = 1:ndims, verbose = F)
object <- FindClusters(object, resolution = 0.3 ,verbose = F)
object <- RunUMAP(object, dims = 1:ndims, verbose = F)

#visaualization samples + clusters
DimPlot(object, group.by = "disease")
DimPlot(object, group.by = "sample")
DimPlot(object, group.by = "svm_annotation",  cols = colors_subtypes,  repel = T,  label = T,  label.box = T) + NoLegend()

#visualization genes of interest
FeaturePlot(object, features = c("KRT5", "KRT14", "KRT1", "KRT10", "KRT15", "ASS1", "TP63"))
FeaturePlot(object, features = c("ITGB1", "ITGA6", "COL17A1", "LRIG1", "LGR5", "LGR6", "AXIN2", "FRMD4A", "DSG3", "DLL1"))
FeaturePlot(object, features = c('KRT15', "ASS1", "IGFBP3", "LGALS3BP", "PTEN", "ATF3", "ATF4", "FOSL", "WNT5A", "TP53"))
```

```{r Visualization dataset}
Idents(roj_rds) <- roj_rds$svm_annotation
DimPlot(roj_rds, cols = colors_subtypes,  repel = T,  label = T,  label.box = T) + NoLegend()
# DimPlot(roj_rds[, roj_rds$disease == "HC"], cols = colors_subtypes,  repel = T,  label = T,  label.box = T) + NoLegend()
# DimPlot(roj_rds[, roj_rds$disease == "AD"], cols = colors_subtypes,  repel = T,  label = T,  label.box = T) + NoLegend()

DimPlot(roj_rds, group.by = "disease")
DimPlot(roj_rds, group.by = "sample")

table(roj_rds[, roj_rds$disease == "HC"]$svm_added_anno)
table(roj_rds[, roj_rds$disease == "AD"]$svm_added_anno)
```

```{r Visualization genes of interest}
#visualizing epidermal layer markers in healthy KCs
VlnPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_added_anno" , features = c("IRG1", "KRT14", "KRT1", "KRT10"), stack = T, split.by = "disease", flip = T)
#epidermal stem cell markers
SC_markers <- c("ITGB1", "ITGA6", "CD71", "DSG3", "DLL1", "MCSP", "LRIG1", "FRMD4A", "CD46", "HDAC1", "AXIN2")
VlnPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_added_anno" , features = SC_markers, stack = T, split.by = "disease", flip = T)

#how do they change in disease?
VlnPlot(roj_rds, group.by = "svm_added_anno" , features = c("KRT5", "KRT14", "KRT1", "KRT10"), stack = T, split.by = "disease", flip = T)
VlnPlot(roj_rds, group.by = "svm_added_anno" , features = SC_markers, stack = T, split.by = "disease", flip = T)

#visualization KRT15/ASS1 HC vs AD
VlnPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_added_anno" , features = c("KRT15", "ASS1"), stack = T, split.by = "disease", flip = T)
VlnPlot(roj_rds, group.by = "svm_added_anno" , features = c("KRT15", "ASS1"), stack = T, split.by = "disease", flip = T)

#TP63 in HC vs AD
VlnPlot(roj_rds, group.by = "svm_annotation" , features = c("TP63"), split.by = "disease", flip = T)


DimPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "sample")
VlnPlot(roj_rds, features = c('KRT5', 'KRT14', "KRT1", "KRT10")) # assay = 'RNA'

VlnPlot(roj_rds, group.by = "svm_annotation" , features = c("MKI67", "TOP2A", "MCM3", "HELLS", "KRT15", "ASS1", "AQP3", "DSG1"), stack = T, split.by = "disease", split.plot = T, flip = T)

VlnPlot(roj_rds, features = c("MCM3", "UHRF1", "CDK4", "HELLS", 'MKI67', "CDK1", 'PTTG1', "TOP2A"), cols= c(rep("darkgreen", 4), rep("orange", 4)), stack = T, flip = T) + NoLegend()

VlnPlot(roj_rds, features = c('KRT5', 'KRT14', "KRT1", "KRT10"), cols = c(rep("gray", 4)), stack = T, flip = T) + NoLegend()

Idents(roj_rds) <- roj_rds$svm_annotation

VlnPlot(roj_rds, features = c('AQP3', "DSG1", 'DSC3'), cols= c(rep("lightgreen", 5)), stack = T, flip = T) + NoLegend()

VlnPlot(roj_rds, features = c('KRT15', "ASS1", 'POSTN', "CCL2", "TP63"), cols= c(rep("#8DA0CB", 5)), stack = T, flip = T) + NoLegend()

VlnPlot(roj_rds, group.by = "svm_annotation" , features = c("TP63"), split.by = "disease", flip = T)

#visualizing basal III subtypes in UMAP
Idents(roj_rds) <- "svm_added_anno"

DimPlot(roj_rds[, roj_rds$disease == "AD"], cols = colors_subtypes_b3sub, repel = T,  label = T,  label.box = T) + NoLegend()

#IVL, KRT16, S100A9, S100A8, KRT6A, KRT16 also possible
FeaturePlot(roj_rds[, roj_rds$disease == "HC"], features = c('KRT15', "ASS1"))
FeaturePlot(roj_rds[, roj_rds$disease == "HC"], features = c('KRT5', "KRT1", "DSC1"), ncol = 1)
FeaturePlot(roj_rds[, roj_rds$disease == "AD"], features = c('KRT5', "KRT1", "DSC1"), ncol = 1)
FeaturePlot(roj_rds[, roj_rds$disease == "HC"], features = c("FLG", "KRT6A", "S100A8"), ncol = 1)
FeaturePlot(roj_rds[, roj_rds$disease == "AD"], features = c("FLG", "KRT6A", "S100A8"), ncol = 1)
# FeaturePlot(roj_rds, split.by = "disease", features = c('KRT5', "KRT1", "DSC1", "FLG", "S100A8", "KRT6A")) 
# & theme(legend.position = c(0.1,0.2))
FeaturePlot(roj_rds[, roj_rds$disease == "HC"], features = c('LOR'), ncol = 1)

VlnPlot(roj_rds, group.by = "svm_annotation", split.by = "disease", features = c("KRT5", "KRT14", "KRT1", "KRT10", "DSC1", "DSG1", "COL17A1", "ITGB1", "ITGA6", 'KRT15', "IVL", 'FLG', "LOR", 'KRT6A', 'KRT16', "S100A8", "S100A9", "CA2"), stack = T, flip = T) + theme(axis.title.x = element_blank()) #cols = c(rep("#8DA0CB", 66)) 

#let's check for AD specific markers
FeaturePlot(roj_rds, split.by = "disease", features = c("CA2", "CCL27", "IFI27", "TYMP")) 

#visualization genes of interest
VlnPlot(roj_rds, group.by = "svm_annotation", features = c("KLF4"), split.by = "disease", flip = T, pt.size = 0) + theme(axis.title.x = element_blank()) 
VlnPlot(roj_rds, group.by = "svm_annotation", features = c("KLF4", "GATA3", "TAF10", "MAFB", "KRT10", "DSG1", "CLDN1", "SPRY1"), split.by = "disease", stack = T, flip = T, pt.size = 0) + theme(axis.title.x = element_blank()) 

VlnPlot(roj_rds, group.by = "svm_annotation", features = c("KLF5", "Akt1", "Akt3", "TP53"), split.by = "disease", stack = T, flip = T, pt.size = 0) + theme(axis.title.x = element_blank()) 
VlnPlot(roj_rds, group.by = "svm_annotation", features = c("AHR", "OVOL1", "ID1", "CEBPA"), split.by = "disease", stack = T, flip = T, pt.size = 0) + theme(axis.title.x = element_blank()) 

FeaturePlot(roj_rds[, roj_rds$disease == "HC"], features = c('IGFBP3', "IFITM1"))

# Fig 1C
FeaturePlot(roj_rds, features = c("KRT5", "KRT1", "FLG", "CA2"), split.by = "disease") & theme(legend.position = "right")
FeaturePlot(roj_rds, features = c("KRT5", "KRT1", "FLG", "S100A7"), split.by = "disease") & theme(legend.position = "right")
FeaturePlot(roj_rds, features = c("KRT5", "KRT1", "FLG", "KRT17"), split.by = "disease") & theme(legend.position = "right")

#Fig 1D
VlnPlot(roj_rds, group.by = "svm_annotation", split.by = "disease", features = c("KRT5", "KRT14", "KRT1", "KRT10", "DSC1", "CLDN4", 'IVL', "FLG", "LOR", 'KRT6A', 'KRT16', "CA2", "KRT17", "S100A7", "DEFB1"), stack = T, flip = T, cols = colors_disease) + theme(axis.title.x = element_blank()) #other markers of activated epithelium: S100A8, S100A9

#additional AD markers
VlnPlot(roj_rds, group.by = "svm_annotation", split.by = "disease", features = c("APOE", "CCL27", "STAT1",  "NEAT1"), stack = T, flip = T, cols = colors_disease) + theme(axis.title.x = element_blank()) #i also tried "CCL17", "TARC", "NOS2", "CTACK", "CCL27", "S100A12", "MMP12", "LDH", "CCL26", "SELE", "CCR7", "CCL19", "PIK3R1", "STAT6", "HBD2", "PI3", "MKI67", "CCL2", "CXCL2", AMPS: "LL37", "DEFB4", "S100A9", "DEFB2", "DEFB3", "HBD4", "SLPI"

#Stem cell markers
VlnPlot(roj_rds, group.by = "svm_annotation", split.by = "disease", features = c("COL17A1", "ITGB1", "ITGA6", 'KRT15'), stack = T, flip = T, cols = colors_disease) + theme(axis.title.x = element_blank()) #i also tried "LRIG1", "DLL1", "MCSP", "CD46", "FRMD4A", "OCT4", "MYC", "OCT3", "SOX2", "KLF4"

VlnPlot(roj_rds, group.by = "svm_annotation", split.by = "disease", features = c("KRT15", "ASS1", "MCM3", "MKI67", "KRT14"), stack = T, flip = T, cols = colors_disease) + theme(axis.title.x = element_blank())

VlnPlot(roj_rds, group.by = "svm_annotation", split.by = "disease", features = c("TP63"), flip = T, pt.size = 0, cols = colors_disease) + theme(axis.title.x = element_blank())

VlnPlot(roj_rds, group.by = "svm_annotation", features = c("KLF4", "GATA3", "TAF10", "MAFB", "KRT10", "DSG1", "CLDN1", "SPRY1"), split.by = "disease", stack = T, flip = T, pt.size = 0, cols = colors_disease) + theme(axis.title.x = element_blank()) 

VlnPlot(roj_rds, group.by = "svm_annotation", features = c("AHR", "NGFR"), split.by = "disease", stack = T, flip = T, pt.size = 0, cols = colors_disease) + theme(axis.title.x = element_blank()) 
```

```{r stacked barplot cell type percentage - HC vs AD}
percentage_df = function(seurat_object, cell_annotation = "seurat_clusters",variable = "orig.ident",facet = F){
  #variable = which columns
  #Change ID
  seurat_object = seurat_object@meta.data
  df_percentages = ""

  #list of variables
  my_vars = as.character(as.data.frame(table(seurat_object[[variable]]))[,1])
  count = 0
  #loop through each variable, add to dataframe
  for(var_x in my_vars){
    #Subset dataframe for specific variable
    seurat_object_tmp = seurat_object[seurat_object[[variable]] == var_x,]
    
    percent_total = as.data.frame(table(seurat_object_tmp[[cell_annotation]]))#Add cell count
    percent_total$Var1 = as.character(percent_total$Var1)#Change type factor -> character
    percent_total["percentage"] = percent_total$Freq/sum(percent_total$Freq)*100 #Add percentage
    percent_total["variable"] = var_x #add variable
    if(count == 0){
    df_percentages = percent_total #create new df for first
    }else{
    df_percentages[(nrow(df_percentages)+1):(nrow(df_percentages)+nrow(percent_total)),] = percent_total
    
    }
  count = count + 1

  }
  #formatting
  df_percentages$percentage =  signif(df_percentages$percentage,3)
  colnames(df_percentages) = c("Celltype"  ,  "Freq"    ,   "percentage", "Variable"   )
  #readable labels
  df_percentages = df_percentages[df_percentages$Freq != 0,]
  
  if(facet){
    df_percentages$con = df_percentages$Variable
    list_m_a = strsplit(df_percentages$Variable,"_")
    for(p in 1:length(list_m_a)){
      var = paste0(list_m_a[[p]][1], list_m_a[[p]][2])
      df_percentages$con[p] = var
    }
    df_percentages$sample = df_percentages$Variable
  }
  df_percentages$ann = df_percentages$percentage
  df_percentages$ann[df_percentages$ann < 1] = "" #remove labels of small fractions
  return(df_percentages)
}

cell_percentage  <- percentage_df(roj_rds, cell_annotation = "svm_annotation", variable = "disease", facet = F)
#reorder: 
cell_percentage$Celltype <- factor(cell_percentage$Celltype, levels = c("basal III", "basal II", "basal I", "basal IV", "stratum spinosum", "stratum granulosum"))
# factor(percentage_df$Celltype)
cell_percentage$Variable <- factor(cell_percentage$Variable, levels = c("HC", "AD")) 
# factor(cell_percentage$Variable)

colors_subtypes <- c("basal III" = "#8DA0CB", "basal II" = "#66C2A5", "basal I" = "#FFD92F", "basal IV" = "#A6D854", "stratum spinosum" = "#FC8D62", "stratum granulosum" = "#E78AC3")

ggplot(cell_percentage, aes(fill = Celltype, y = percentage, x = Variable)) + 
    geom_bar(position = "stack", stat = "identity") +
    theme_classic() +
    geom_text(aes(label = percentage), position = position_stack(vjust = 0.5), size = 3) + 
    theme(text = element_text(size = 10))+
    scale_fill_manual(values = colors_subtypes)
```

```{r stacked barplot cell type percentage - per sample}
percentage_df = function(seurat_object, cell_annotation = "seurat_clusters",variable = "orig.ident",facet = F){
  #variable = which columns
  #Change ID
  seurat_object = seurat_object@meta.data
  df_percentages = ""

  #list of variables
  my_vars = as.character(as.data.frame(table(seurat_object[[variable]]))[,1])
  count = 0
  #loop through each variable, add to dataframe
  for(var_x in my_vars){
    #Subset dataframe for specific variable
    seurat_object_tmp = seurat_object[seurat_object[[variable]] == var_x,]
    
    percent_total = as.data.frame(table(seurat_object_tmp[[cell_annotation]]))#Add cell count
    percent_total$Var1 = as.character(percent_total$Var1)#Change type factor -> character
    percent_total["percentage"] = percent_total$Freq/sum(percent_total$Freq)*100 #Add percentage
    percent_total["variable"] = var_x #add variable
    if(count == 0){
    df_percentages = percent_total #create new df for first
    }else{
    df_percentages[(nrow(df_percentages)+1):(nrow(df_percentages)+nrow(percent_total)),] = percent_total
    
    }
  count = count + 1

  }
  #formatting
  df_percentages$percentage =  signif(df_percentages$percentage,3)
  colnames(df_percentages) = c("Celltype"  ,  "Freq"    ,   "percentage", "Variable"   )
  #readable labels
  df_percentages = df_percentages[df_percentages$Freq != 0,]
  
  if(facet){
    df_percentages$con = df_percentages$Variable
    list_m_a = strsplit(df_percentages$Variable,"_")
    for(p in 1:length(list_m_a)){
      var = paste0(list_m_a[[p]][1], list_m_a[[p]][2])
      df_percentages$con[p] = var
    }
    df_percentages$sample = df_percentages$Variable
  }
  df_percentages$ann = df_percentages$percentage
  df_percentages$ann[df_percentages$ann < 1] = "" #remove labels of small fractions
  return(df_percentages)
}

cell_percentage  <- percentage_df(roj_rds, cell_annotation = "svm_annotation", variable = "sample", facet = F)
cell_percentage$disease[cell_percentage$Variable %in% c("HC6", "HC7")] <- c("HC")
cell_percentage$disease[cell_percentage$Variable %in% c("AD5", "AD6", "AD7", "AD8")] <- c("AD")

#reorder: 
cell_percentage$Celltype <- factor(cell_percentage$Celltype, levels = c("basal III", "basal II", "basal I", "basal IV", "stratum spinosum", "stratum granulosum"))
# factor(percentage_df$Celltype)
cell_percentage$Variable <- factor(cell_percentage$Variable, levels = c("HC6", "HC7", "AD5", "AD6", "AD7", "AD8")) 
factor(cell_percentage$Variable)

ggplot(cell_percentage, aes(fill = Celltype, y = percentage, x = Variable)) + 
    geom_bar(position = "stack", stat = "identity") +
    theme_classic() +
    geom_text(aes(label = percentage), position = position_stack(vjust = 0.5), size = 3) + 
    theme(text = element_text(size = 10))+
    scale_fill_manual(values = colors_subtypes)

t_basalIII <- t.test(c(33.2, 22.6), c(33.8, 56.6, 47.1, 37.7), data = cell_percentage, var.equal = TRUE)
t_basalII <- t.test(c(6.39,	5.19), c(3.59,	3.1,	3.22,	3.63), data = cell_percentage, var.equal = TRUE)
t_basalI <- t.test(c(9.06,	5.62), c(10.8,	6.4,	5.36,	8.99), data = cell_percentage, var.equal = TRUE)
t_basalIV <- t.test(c(5.33,	5.94), c(25.4,	20,	31,	38.8), data = cell_percentage, var.equal = TRUE)
t_spinosum <- t.test(c(33.6,	51.9), c(14.5,	9.3,	11.5,	6.12), data = cell_percentage, var.equal = TRUE)
t_granulosum <- t.test(c(12.4,	8.75), c(11.9,	4.55,	1.88,	4.78), data = cell_percentage, var.equal = TRUE)
```

# Differential expression analysis

```{r DEanalysis global - HC markers, svm annotation}
diffex_hc_svm <- readRDS("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD/diffex_hc_svm.rds")

# table(roj_rds$svm_annotation)
# 
# markers <- FindMarkers(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_annotation", ident.1 = "stratum granulosum") 
# colnames(markers) <- paste("HC_granulosum", colnames(markers), sep = "_")
# markers$gene <- rownames(markers)
# 
# # diffex_hc_svm <- data.frame(gene = rownames(markers))
# # diffex_hc_svm$pct_global <- markers$HC_basalIII_pct.2
# diffex_hc_svm <- merge(x = diffex_hc_svm, y = markers, by = "gene", all.x = TRUE, all.y = T)
# diffex_hc_svm <- diffex_hc_svm[ , -c(ncol(diffex_hc_svm)-1, ncol(diffex_hc_svm)-4)] #removes 2 columns
# 
# #replaces all NAs in pct.global
# i <- diffex_hc_svm[is.na(diffex_hc_svm$pct_global), "gene"]
# 
# for(gene_name in i){
#   pct_global <- markers[markers$gene == gene_name, 4] #find in markers
#   index <- which(diffex_hc_svm$gene == gene_name) #find row in diffex_hc
#   diffex_hc_svm$pct_global[index] <- pct_global #replace NA there
# }

# diffex_hc_svm <- diffex_hc[ ,-c(9, 10, 11)]
# diffex_hc_svm <- diffex_hc %>% rename(HC_basal_I_pct.1 =  HC_basalI_pct.1.y)

# setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
# saveRDS(diffex_hc_svm, file = "diffex_hc_svm.rds")
# write.xlsx(diffex_hc_svm, file = "diffex_hc_svm.xlsx")
```

```{r DEanalysis global - AD markers, svm annotation}
diffex_ad_svm <- readRDS("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD/diffex_ad_svm.rds")

# table(roj_rds$svm_annotation)
# 
# markers <- FindMarkers(roj_rds[, roj_rds$disease == "AD"], group.by = "svm_annotation", ident.1 = "stratum granulosum")
# colnames(markers) <- paste("AD_granulosum", colnames(markers), sep = "_")
# markers$gene <- rownames(markers)
# 
# # diffex_ad_svm <- data.frame(gene = rownames(markers))
# # diffex_ad_svm$pct_global <- markers$AD_basal_III_pct.2
# diffex_ad_svm <- merge(x = diffex_ad_svm, y = markers, by = "gene", all.x = TRUE, all.y = T)
# diffex_ad_svm <- diffex_ad_svm[ , -c(ncol(diffex_ad_svm)-1, ncol(diffex_ad_svm)-4)] #removes 2 columns
# 
# #replaces all NAs in pct.global
# i <- diffex_ad_svm[is.na(diffex_ad_svm$pct_global), "gene"]
# 
# for(gene_name in i){
#   pct_global <- markers[markers$gene == gene_name, 4] #find in markers
#   index <- which(diffex_ad_svm$gene == gene_name) #find row in diffex_ad
#   diffex_ad_svm$pct_global[index] <- pct_global #replace NA there
# }
# 
# # diffex_hc_svm <- diffex_hc[ ,-c(9, 10, 11)]
# # diffex_hc_svm <- diffex_hc %>% rename(HC_basal_I_pct.1 =  HC_basalI_pct.1.y)
# 
# setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
# saveRDS(diffex_ad_svm, file = "diffex_ad_svm.rds")
# write.xlsx(diffex_ad_svm, file = "diffex_ad_svm.xlsx")
```

```{r DEanalysis global - AD vs HC, svm annotated}
diffex_ad_hc_svm <- readRDS("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD/diffex_ad_hc_svm.rds")

#first one is for AD vs HC (bulk)

# markers <- FindMarkers(roj_rds, group.by = "disease", ident.1 = "AD", ident.2 = "HC") #comparison of changes in AD compared to HC, log2FC up means more expressed in first group (AD)
# 
# colnames(markers) <- paste("AD", colnames(markers), sep = "_")
# markers$gene <- rownames(markers)
# 
# diffex_ad_hc_svm <- data.frame(gene = rownames(markers))
# diffex_ad_hc_svm$pct_AD_global <- markers$AD_pct.1
# diffex_ad_hc_svm$pct_HC_global <- markers$AD_pct.2
# diffex_ad_hc_svm <- merge(x = diffex_ad_hc_svm, y = markers, by = "gene", all.x = TRUE, all.y = T)
# diffex_ad_hc_svm <- diffex_ad_hc_svm[ , -c(ncol(diffex_ad_hc_svm)-1, ncol(diffex_ad_hc_svm)-4)] #removes 2 columns
# diffex_ad_hc_svm <- diffex_ad_hc_svm[ , -(5)] #removes AD.pct1 (because we have this already as AD global)
# 
# #this one is for the comparison of cell types, has to be done for each cell type individually
# 
# table(roj_rds$svm_annotation)
# Idents(roj_rds) <- roj_rds$svm_annotation
# 
# markers <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "stratum granulosum", ident.1 = "AD", ident.2 = "HC") #comparison of changes in AD compared to HC, log2FC up means more expressed in first group (AD)
# colnames(markers) <- paste("ADvsHC_granulosum", colnames(markers), sep = "_")
# names <- colnames(markers)
# names <- gsub(patter = "pct.1", replacement = "pct.AD", x = names) #replaces pattern in a string
# names <- gsub(patter = "pct.2", replacement = "pct.HC", x = names) 
# colnames(markers) <- names
# markers$gene <- rownames(markers)
# 
# diffex_ad_hc_svm <- merge(x = diffex_ad_hc_svm, y = markers, by = "gene", all.x = TRUE, all.y = T)
# diffex_ad_hc_svm <- diffex_ad_hc_svm[ , -c(ncol(diffex_ad_hc_svm)-4)] #removes unadjusted p value column

# setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
# saveRDS(diffex_ad_hc_svm, file = "diffex_ad_hc_svm.rds")
# write.xlsx(diffex_ad_hc_svm, file = "diffex_ad_hc_svm.xlsx")
```

# Basal I - IV marker genes

```{r Visualization cell type markers}
m_bIII <- c("POSTN", "KRT15", "ASS1", "CCL2", "COL7A1", "COL18A1", "JAG2", "DKK3", "ITGB4", "WNT3") 
m_bII <- c("MCM3", "MCM5", "DNMT1", "GINS2", "HELLS", "MCM4", "CDT1") 
m_bI <- c("CENPF", "MKI67", "CCNB2", "CDKN3", "TOP2A", "ASPM", "BIRC5", "UBE2C", "CDC20", "CCNB1")
m_bIV <- c("MT1H", "VSNL1", "KRT6A", "RRP15", "SMAD2", "ITPRIPL2", "MAPK1IP1L")

VlnPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_annotation", features = c(m_bIII, m_bII, m_bI, m_bIV, "KRT5", "KRT14"), cols = c(rep("#8DA0CB", 10), rep("#66C2A5", 7), rep("#FFD92F", 10), c(rep("#A6D854", 7)), rep("#FC8D62", 2)), stack = T, flip = T) + NoLegend()

VlnPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_annotation", features = c(m_bII), cols = c(rep("#66C2A5", 10)), stack = T, flip = T) + NoLegend()

VlnPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_annotation", features = c(m_bI), cols = c(rep("#FFD92F", 11)), stack = T, flip = T) + NoLegend()

VlnPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_annotation", features = c(m_bIV), cols = c(rep("#A6D854", 12)), stack = T, flip = T) + NoLegend()

VlnPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_annotation", features = c("KRT16", "KRT5", "MT1G", "MT1X", "KRT6A", "LY6D", "MT1H", "KRT14", "MT1E", "S100A2", "S100A14"), cols = c(rep("#A6D854", 12)), stack = T, flip = T) + NoLegend() #potentially also krt16, ATF3, DSP, DSC3 are DE genes for basal IV

VlnPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_added_anno", features = c("KRT1", "KRT5", "KRT14", "COL17A1", "WNT4", "CCL2", "DSG1", "KRT15", "ASS1"), stack = T, flip = T) + NoLegend()

VlnPlot(roj_rds, group.by = "svm_annotation", split.by = "disease", features = c("KRT15", "ASS1", "MCM3", "MKI67", "KRT14"), stack = T, flip = T) + NoLegend()
```

```{r Visualization stem cell markers}
Idents(roj_rds) <- "svm_annotation"

VlnPlot(roj_rds, group.by = "svm_annotation", split.by = "disease", features = c("KRT5", "KRT14", "KRT1", "KRT10", "COL17A1", "ITGB1", "ITGA6", 'KRT15'), stack = T, flip = T)  + theme(axis.title.x = element_blank()) #cols = c(rep("#8DA0CB", 66))

#test more markers
VlnPlot(roj_rds, group.by = "svm_annotation", split.by = "disease", features = c("ITGA6", "ITGB1", "TFRC", "DLL1", "DSG1", "DSG2", "DSG3", "DSG4", "EGFR"), stack = T, flip = T) + NoLegend()

VlnPlot(roj_rds, group.by = "svm_added_anno", split.by = "disease", features = c("LGR5", "LGR6", "AXIN2", "LRIG1"), stack = T, flip = T) + NoLegend()

FeaturePlot(roj_rds, features = c('LGR6', "LRIG1"), split.by = "disease")


FeaturePlot(roj_rds, features = c("COL17A1", "ITGB1", "ITGA6", 'KRT15', "TFRC", "DLL1","LGR5", "LGR6", "AXIN2", "LRIG1"))  

#test more markers
VlnPlot(roj_rds, group.by = "svm_annotation", split.by = "disease", features = c("ITGA6", "ITGB1", "TFRC", "DLL1", "DSG1", "DSG2", "DSG3", "DSG4", "EGFR"), stack = T, flip = T) + NoLegend()

VlnPlot(roj_rds, group.by = "svm_added_anno", split.by = "disease", features = c("LGR5", "LGR6", "AXIN2", "LRIG1"), stack = T, flip = T) + NoLegend()

FeaturePlot(roj_rds, features = c('LGR6', "LRIG1"), split.by = "disease")



FeatureScatter(roj_rds[ roj_rds$disease == "HC"], feature1 = 'KRT15', feature2 = 'ASS1')
FeaturePlot(roj_rds[ ,roj_rds$disease == "HC"], features = c('KRT15', 'ASS1'), blend = T)
DotPlot(roj_rds[ ,roj_rds$disease == "HC"], features = c('KRT15', 'ASS1'))

roj_rds[, roj_rds$svm_annotation == "basal III"]

FeaturePlot(roj_rds, features = c('MKI67'), split.by = "disease")
FeaturePlot(roj_rds[ ,roj_rds$disease == "AD"], features = c('MKI67'))

VlnPlot(roj_rds, features = c('MKI67'), split.by = "disease")
```

```{r boxplot KRT15 / ASS1}
#create a table with features of interest (in the cells from a cluster of interest)
Idents(roj_rds) <- "svm_annotation"

roj_rds_basalIII <- subset(roj_rds, idents = "basal III" ) #cluster of interest 
roj_rds_basalIII_KRT15 <- subset(roj_rds_basalIII, features = c("KRT15")) #feature of interest 
roj_rds_basalIII_ASS1 <- subset(roj_rds_basalIII, features = c("ASS1")) #feature of interest

#reshape the table to get it in the correct format to create a boxplot
matrix_KRT15 <-as.matrix(roj_rds_basalIII_KRT15@assays[["RNA"]]@data)
matrix_ASS1 <-as.matrix(roj_rds_basalIII_ASS1@assays[["RNA"]]@data)

library(reshape2)
library(ggplot2)
library(ggpubr)                               

dataframe_with_columns_KRT15 <-setNames(melt(matrix_KRT15), c('rows', 'col' ,'expression')) #use matrix as input
dataframe_with_columns_KRT15$disease <- roj_rds_basalIII_KRT15@meta.data$disease
#dataframe_with_columns$test <- colnames(roj_rds_basalIII_KRT15)
#all(dataframe_with_columns$test == dataframe_with_columns$col) #true 

dataframe_with_columns_ASS1 <-setNames(melt(matrix_ASS1), c('rows', 'col' ,'expression')) #use matrix as input
dataframe_with_columns_ASS1$disease <- roj_rds_basalIII_ASS1@meta.data$disease


#create the boxplot
#On the x-axis are the features
#when checking with data pints : geom_jitter(color="black") +
pdf("/scratch/bacint/yquint/Felicitas_derma/report/AD_HC_KRT15 expression_basal_III.pdf")
ggplot(dataframe_with_columns_KRT15, aes(rows, expression, fill=disease)) +
  geom_boxplot()+
  stat_compare_means(label = "p.signif")  #wilcox.test
  #theme(legend.position = "none")
dev.off()

pdf("/scratch/bacint/yquint/Felicitas_derma/report/AD_HC_ASS1 expression_basal_III.pdf")
ggplot(dataframe_with_columns_ASS1, aes(rows, expression, fill=disease)) +
  geom_boxplot()+
  stat_compare_means(label = "p.signif")  #wilcox.test
#theme(legend.position = "none")
dev.off()

```

# Cell cycle analysis

```{r cell cycle analysis}
# cc.genes.updated.2019 #updated list of cell cycle marker genes
roj_rds <- CellCycleScoring(roj_rds, s.features = cc.genes.updated.2019$s.genes, g2m.features = cc.genes.updated.2019$g2m.genes, set.ident = TRUE) #adding CC identity to the object metaData
# the lists cc.genes.updated are part of the Seurat package, 2019 version is the latest update

# head(roj_rds@meta.data)
Idents(roj_rds) <- "Phase"
DimPlot(roj_rds, group.by = "Phase", cols = colors_cellcycle, order = c("G2M", "S", "G1"))

# RidgePlot(roj_rds, features = c("S.Score", "G2M.Score"), ncol = 2, group.by = "svm_annotation")

#Visualize the percentage of cell cycle phases
percentage_df = function(seurat_object, cell_annotation = "seurat_clusters",variable = "orig.ident",facet = F){
  #variable = which columns
  #Change ID
  seurat_object = seurat_object@meta.data
  df_percentages = ""

  #list of variables
  my_vars = as.character(as.data.frame(table(seurat_object[[variable]]))[,1])
  count = 0
  #loop through each variable, add to dataframe
  for(var_x in my_vars){
    #Subset dataframe for specific variable
    seurat_object_tmp = seurat_object[seurat_object[[variable]] == var_x,]
    
    percent_total = as.data.frame(table(seurat_object_tmp[[cell_annotation]]))#Add cell count
    percent_total$Var1 = as.character(percent_total$Var1)#Change type factor -> character
    percent_total["percentage"] = percent_total$Freq/sum(percent_total$Freq)*100 #Add percentage
    percent_total["variable"] = var_x #add variable
    if(count == 0){
    df_percentages = percent_total #create new df for first
    }else{
    df_percentages[(nrow(df_percentages)+1):(nrow(df_percentages)+nrow(percent_total)),] = percent_total
    
    }
  count = count + 1

  }
  #formatting
  df_percentages$percentage =  signif(df_percentages$percentage,3)
  colnames(df_percentages) = c("Celltype"  ,  "Freq"    ,   "percentage","Variable"   )
  #readable labels
  df_percentages = df_percentages[df_percentages$Freq != 0,]
  
  if(facet){
    df_percentages$con = df_percentages$Variable
    list_m_a = strsplit(df_percentages$Variable,"_")
    for(p in 1:length(list_m_a)){
      var = paste0(list_m_a[[p]][1], list_m_a[[p]][2])
      df_percentages$con[p] = var
    }
    df_percentages$sample = df_percentages$Variable
  }
  df_percentages$ann = df_percentages$percentage
  df_percentages$ann[df_percentages$ann < 1] = "" #remove labels of small fractions
  return(df_percentages)
}

phase_percentage  <- percentage_df(roj_rds[, roj_rds$disease == "HC"], cell_annotation = "Phase", variable = "svm_annotation", facet = F)

#reorder: 
phase_percentage$Variable <- factor(phase_percentage$Variable, levels = c("basal III", "basal II", "basal I", "basal IV", "stratum spinosum", "stratum granulosum"))
factor(phase_percentage$Variable)
phase_percentage$Celltype <- factor(phase_percentage$Celltype, levels = c("G1", "S", "G2M")) 
factor(phase_percentage$Celltype)

ggplot(phase_percentage, aes(fill = Celltype, y = percentage, x = Variable)) + 
    geom_bar(position = "stack", stat = "identity") +
    theme_classic() +
    geom_text(aes(label = percentage), position = position_stack(vjust = 0.5), size = 3) + 
    theme(text = element_text(size = 10), axis.title.x = element_blank()) +
    scale_fill_manual(values = colors_cellcycle)

#AD
phase_percentage  <- percentage_df(roj_rds[, roj_rds$disease == "AD"], cell_annotation = "Phase", variable = "svm_annotation", facet = F)

#reorder: 
phase_percentage$Variable <- factor(phase_percentage$Variable, levels = c("basal III", "basal II", "basal I", "basal IV", "stratum spinosum", "stratum granulosum"))
factor(phase_percentage$Variable)
phase_percentage$Celltype <- factor(phase_percentage$Celltype, levels = c("G1", "S", "G2M")) 
factor(phase_percentage$Celltype)

ggplot(phase_percentage, aes(fill = Celltype, y = percentage, x = Variable)) + 
    geom_bar(position = "stack", stat = "identity") +
    theme_classic() +
    geom_text(aes(label = percentage), position = position_stack(vjust = 0.5), size = 3) + 
    theme(text = element_text(size = 10), axis.title.x = element_blank()) +
    scale_fill_manual(values = colors_cellcycle)
```

# Functional analysis basal cell types HC
##Prepare gene lists

```{r annotate gene symbol with entrezID}
#annotate all expressed background genes
background <-  mapIds(
  org.Hs.eg.db,
  keys = roj_rds@assays[["RNA"]]@counts@Dimnames[[1]],
  column = "ENTREZID",
  keytype = "SYMBOL",
  multiVals = "first"
)

#annotate DE genes in the differential expression data frame
entrez_anno <- bitr(diffex_hc_svm$gene, fromType = "SYMBOL",
        toType = c("ENTREZID"),
        OrgDb = org.Hs.eg.db)

diffex_hc_svm <- merge(x = diffex_hc_svm, y = entrez_anno, by.x = "gene", by.y = "SYMBOL", all.x = T)

diffex_hc_svm <- diffex_hc_svm %>% relocate(ENTREZID, .after = gene)
diffex_hc_svm <- diffex_hc_svm %>% rename(entrezID = ENTREZID)

#which genes do not have an entrezID
table(is.na(diffex_hc_svm$entrezID))
diffex_hc_svm$gene[is.na(diffex_hc_svm$entrezID)]
```

```{r prepare input gene lists - over enrichment analysis}
#basal III
DEHC_basalIII <- diffex_hc_svm[diffex_hc_svm$HC_basal_III_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_III_p_val_adj), "entrezID"]
DEHC_basalIIIup <- diffex_hc_svm[diffex_hc_svm$HC_basal_III_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_III_p_val_adj) & diffex_hc_svm$HC_basal_III_avg_log2FC > 0, "entrezID"]
DEHC_basalIIIdown <- diffex_hc_svm[diffex_hc_svm$HC_basal_III_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_III_p_val_adj) & diffex_hc_svm$HC_basal_III_avg_log2FC < 0, "entrezID"]

#basal II
DEHC_basalII <- diffex_hc_svm[diffex_hc_svm$HC_basal_II_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_II_p_val_adj), "entrezID"]
DEHC_basalIIup <- diffex_hc_svm[diffex_hc_svm$HC_basal_II_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_II_p_val_adj) & diffex_hc_svm$HC_basal_II_avg_log2FC > 0, "entrezID"]
DEHC_basalIIdown <- diffex_hc_svm[diffex_hc_svm$HC_basal_II_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_II_p_val_adj) & diffex_hc_svm$HC_basal_II_avg_log2FC < 0, "entrezID"]

#basal I
DEHC_basalI <- diffex_hc_svm[diffex_hc_svm$HC_basal_I_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_I_p_val_adj), "entrezID"]
DEHC_basalIup <- diffex_hc_svm[diffex_hc_svm$HC_basal_I_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_I_p_val_adj) & diffex_hc_svm$HC_basal_I_avg_log2FC > 0, "entrezID"]
DEHC_basalIdown <- diffex_hc_svm[diffex_hc_svm$HC_basal_I_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_I_p_val_adj) & diffex_hc_svm$HC_basal_I_avg_log2FC < 0, "entrezID"]

#basal IV
DEHC_basalIV <- diffex_hc_svm[diffex_hc_svm$HC_basal_IV_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_IV_p_val_adj), "entrezID"]
DEHC_basalIVup <- diffex_hc_svm[diffex_hc_svm$HC_basal_IV_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_IV_p_val_adj) & diffex_hc_svm$HC_basal_IV_avg_log2FC > 0, "entrezID"]
DEHC_basalIVdown <- diffex_hc_svm[diffex_hc_svm$HC_basal_IV_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_IV_p_val_adj) & diffex_hc_svm$HC_basal_IV_avg_log2FC < 0, "entrezID"]

#spinosum
DEHC_spinosum <- diffex_hc_svm[diffex_hc_svm$HC_spinosum_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_spinosum_p_val_adj), "entrezID"]
DEHC_spinosumup <- diffex_hc_svm[diffex_hc_svm$HC_spinosum_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_spinosum_p_val_adj) & diffex_hc_svm$HC_spinosum_avg_log2FC > 0, "entrezID"]
DEHC_spinosumdown <- diffex_hc_svm[diffex_hc_svm$HC_spinosum_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_spinosum_p_val_adj) & diffex_hc_svm$HC_spinosum_avg_log2FC < 0, "entrezID"]

#granulosum
DEHC_granulosum <- diffex_hc_svm[diffex_hc_svm$HC_granulosum_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_granulosum_p_val_adj), "entrezID"]
DEHC_granulosumup <- diffex_hc_svm[diffex_hc_svm$HC_granulosum_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_granulosum_p_val_adj) & diffex_hc_svm$HC_granulosum_avg_log2FC > 0, "entrezID"]
DEHC_granulosumdown <- diffex_hc_svm[diffex_hc_svm$HC_granulosum_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_granulosum_p_val_adj) & diffex_hc_svm$HC_granulosum_avg_log2FC < 0, "entrezID"]
```

```{r prepare input gene lists - gsea}
# Idents(roj_rds) <- roj_rds$svm_annotation
# 
# #basal III
# DEAD_basalIII <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "basal III", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
# DEAD_basalIII_gsea <- DEAD_basalIII[ , "avg_log2FC"]
# names(DEAD_basalIII_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIII), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalIII_gsea <- sort(DEAD_basalIII_gsea, decreasing = T)
# 
# DEAD_basalIIIup_gsea <- DEAD_basalIII[DEAD_basalIII$avg_log2FC > 0, "avg_log2FC"]
# names(DEAD_basalIIIup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIII)[DEAD_basalIII$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalIIIup_gsea <- sort(DEAD_basalIIIup_gsea, decreasing = T)
# 
# DEAD_basalIIIdown_gsea <- DEAD_basalIII[DEAD_basalIII$avg_log2FC < 0, "avg_log2FC"]
# names(DEAD_basalIIIdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIII)[DEAD_basalIII$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalIIIdown_gsea <- sort(DEAD_basalIIIdown_gsea, decreasing = T)
# 
# #basal II
# DEAD_basalII <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "basal II", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
# DEAD_basalII_gsea <- DEAD_basalII[ , "avg_log2FC"]
# names(DEAD_basalII_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalII), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalII_gsea <- sort(DEAD_basalII_gsea, decreasing = T)
# 
# DEAD_basalIIup_gsea <- DEAD_basalII[DEAD_basalII$avg_log2FC > 0, "avg_log2FC"]
# names(DEAD_basalIIup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalII)[DEAD_basalII$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalIIup_gsea <- sort(DEAD_basalIIup_gsea, decreasing = T)
# 
# DEAD_basalIIdown_gsea <- DEAD_basalII[DEAD_basalII$avg_log2FC < 0, "avg_log2FC"]
# names(DEAD_basalIIdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalII)[DEAD_basalII$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalIIdown_gsea <- sort(DEAD_basalIIdown_gsea, decreasing = T)
# 
# #basal I
# DEAD_basalI <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "basal I", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
# DEAD_basalI_gsea <- DEAD_basalI[ , "avg_log2FC"]
# names(DEAD_basalI_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalI), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalI_gsea <- sort(DEAD_basalI_gsea, decreasing = T)
# 
# DEAD_basalIup_gsea <- DEAD_basalI[DEAD_basalI$avg_log2FC > 0, "avg_log2FC"]
# names(DEAD_basalIup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalI)[DEAD_basalI$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalIup_gsea <- sort(DEAD_basalIup_gsea, decreasing = T)
# 
# DEAD_basalIdown_gsea <- DEAD_basalI[DEAD_basalI$avg_log2FC < 0, "avg_log2FC"]
# names(DEAD_basalIdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalI)[DEAD_basalI$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalIdown_gsea <- sort(DEAD_basalIdown_gsea, decreasing = T)
# 
# #basal IV
# DEAD_basalIV <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "basal IV", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
# DEAD_basalIV_gsea <- DEAD_basalIV[ , "avg_log2FC"]
# names(DEAD_basalIV_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIV), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalIV_gsea <- sort(DEAD_basalIV_gsea, decreasing = T)
# 
# DEAD_basalIVup_gsea <- DEAD_basalIV[DEAD_basalIV$avg_log2FC > 0, "avg_log2FC"]
# names(DEAD_basalIVup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIV)[DEAD_basalIV$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalIVup_gsea <- sort(DEAD_basalIVup_gsea, decreasing = T)
# 
# DEAD_basalIVdown_gsea <- DEAD_basalIV[DEAD_basalIV$avg_log2FC < 0, "avg_log2FC"]
# names(DEAD_basalIVdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIV)[DEAD_basalIV$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_basalIVdown_gsea <- sort(DEAD_basalIVdown_gsea, decreasing = T)
# 
# #spinosum 
# DEAD_spinosum <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "stratum spinosum", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
# DEAD_spinosum_gsea <- DEAD_spinosum[ , "avg_log2FC"]
# names(DEAD_spinosum_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_spinosum), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_spinosum_gsea <- sort(DEAD_spinosum_gsea, decreasing = T)
# 
# DEAD_spinosumup_gsea <- DEAD_spinosum[DEAD_spinosum$avg_log2FC > 0, "avg_log2FC"]
# names(DEAD_spinosumup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_spinosum)[DEAD_spinosum$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_spinosumup_gsea <- sort(DEAD_spinosumup_gsea, decreasing = T)
# 
# DEAD_spinosumdown_gsea <- DEAD_spinosum[DEAD_spinosum$avg_log2FC < 0, "avg_log2FC"]
# names(DEAD_spinosumdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_spinosum)[DEAD_spinosum$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_spinosumdown_gsea <- sort(DEAD_spinosumdown_gsea, decreasing = T)
# 
# #granulosum 
# DEAD_granulosum <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "stratum granulosum", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
# DEAD_granulosum_gsea <- DEAD_granulosum[ , "avg_log2FC"]
# names(DEAD_granulosum_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_granulosum), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_granulosum_gsea <- sort(DEAD_granulosum_gsea, decreasing = T)
# 
# DEAD_granulosumup_gsea <- DEAD_granulosum[DEAD_granulosum$avg_log2FC > 0, "avg_log2FC"]
# names(DEAD_granulosumup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_granulosum)[DEAD_granulosum$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_granulosumup_gsea <- sort(DEAD_granulosumup_gsea, decreasing = T)
# 
# DEAD_granulosumdown_gsea <- DEAD_granulosum[DEAD_granulosum$avg_log2FC < 0, "avg_log2FC"]
# names(DEAD_granulosumdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_granulosum)[DEAD_granulosum$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# DEAD_granulosumdown_gsea <- sort(DEAD_granulosumdown_gsea, decreasing = T)
```

##GO analysis

```{r GO over enrichment}
DEHC_basalIII_GO <-  enrichGO(gene = DEHC_basalIII, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_basalIIIup_GO <-  enrichGO(gene = DEHC_basalIIIup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_basalIIIdown_GO <-  enrichGO(gene = DEHC_basalIIIdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEHC_basalII_GO <-  enrichGO(gene = DEHC_basalII, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_basalIIup_GO <-  enrichGO(gene = DEHC_basalIIup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_basalIIdown_GO <-  enrichGO(gene = DEHC_basalIIdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEHC_basalI_GO <-  enrichGO(gene = DEHC_basalI, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_basalIup_GO <-  enrichGO(gene = DEHC_basalIup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_basalIdown_GO <-  enrichGO(gene = DEHC_basalIdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEHC_basalIV_GO <-  enrichGO(gene = DEHC_basalIV, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_basalIVup_GO <-  enrichGO(gene = DEHC_basalIVup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_basalIVdown_GO <-  enrichGO(gene = DEHC_basalIVdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEHC_spinosum_GO <-  enrichGO(gene = DEHC_spinosum, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_spinosumup_GO <-  enrichGO(gene = DEHC_spinosumup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_spinosumdown_GO <-  enrichGO(gene = DEHC_spinosumdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEHC_granulosum_GO <-  enrichGO(gene = DEHC_granulosum, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_granulosumup_GO <-  enrichGO(gene = DEHC_granulosumup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEHC_granulosumdown_GO <-  enrichGO(gene = DEHC_granulosumdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

#save
list <- list("GO DE HC basal III" = DEHC_basalIII_GO,
             "GO DE HC basal IIIup" = DEHC_basalIIIup_GO,
             "GO DE HC basal IIIdown" = DEHC_basalIIIdown_GO,
             "GO DE HC basal II" = DEHC_basalII_GO,
             "GO DE HC basal IIup" = DEHC_basalIIup_GO,
             "GO DE HC basal IIdown" = DEHC_basalIIdown_GO,
             "GO DE HC basal I" = DEHC_basalI_GO,
             "GO DE HC basal Iup" = DEHC_basalIup_GO,
             "GO DE HC basal Idown" = DEHC_basalIdown_GO,
             "GO DE HC basal IV" = DEHC_basalIV_GO,
             "GO DE HC basal IVup" = DEHC_basalIVup_GO,
             "GO DE HC basal IVdown" = DEHC_basalIVdown_GO,
             "GO DE HC spinosum" = DEHC_spinosum_GO,
             "GO DE HC spinosumup" = DEHC_spinosumup_GO,
             "GO DE HC spinosumdown" = DEHC_spinosumdown_GO,
             "GO DE HC granulosum" = DEHC_granulosum_GO,
             "GO DE HC granulosumup" = DEHC_granulosumup_GO,
             "GO DE HC granulosumdown" = DEHC_granulosumdown_GO)
write.xlsx(list, file = "GO DE HC2.xlsx")

##visualization
GO_HC_vis <- DEHC_basalIIIup_GO[c(1, 4)]
GO_HC_vis <- rbind(GO_HC_vis, DEHC_basalIIup_GO[c(1, 2)])
GO_HC_vis <- rbind(GO_HC_vis, DEHC_basalIup_GO[c(1, 2)])
GO_HC_vis <- rbind(GO_HC_vis, DEHC_basalIVup_GO[c(8, 22)])
GO_HC_vis <- rbind(GO_HC_vis, DEHC_spinosum_GO[c(9, 10)])
GO_HC_vis <- rbind(GO_HC_vis, DEHC_granulosumup_GO[c(3, 17, 18)])
GO_HC_vis$celltype <- c(rep("basal III", 2), rep("basal II", 2), rep("basal I", 2), rep("basal IV", 2), rep("spinosum", 2), rep("granulosum", 3))
GO_HC_vis$celltype <- factor(GO_HC_vis$celltype, levels = c("basal III", "basal II", "basal I", "basal IV", "spinosum", "granulosum"))

ggplot(GO_HC_vis, aes(x = Count, y = Description)) +
        geom_col(aes(fill = p.adjust)) +
        facet_grid("celltype", scale ="free") +
        scale_fill_gradient(high = "blue", low = "red", guide = guide_colourbar(reverse = TRUE)) +
        theme_bw(base_size = 13) + theme(axis.text.y = element_text(size = 13)) +
        ylab(NULL) + xlab("count")

# cnetplot(DEAD_basalIIIup_GO, showCategory = DEAD_basalIIIup_GOdf$Description[c(72, 175)])
# 
# #cnetplot custom GO's
# DEAD_basalIIIup_GOdf <- as.data.frame(DEAD_basalIIIup_GO)
# cnetplot(DEAD_basalIIIup_GO, categorySize = "geneNum", foldChange = genelist_c31,
#          showCategory = DEAD_basalIIIup_GOdf$Description[c(72, 175)]) +
#   scale_color_gradient2(name = "fold change", low = "blue", mid = "gray95", high = "red", midpoint = 0,
#                         limits = c(min(genelist_c31) , max(genelist_c31))) +
#   ggtitle("fold change TCDD vs control", subtitle = "padj < 0.05, background: baseMean > 1, BP & MF & CC together")
```

# Functional analysis basal cell types AD
##Prepare gene lists

```{r annotate gene symbol with entrezID}
#annotate all expressed background genes
background <-  mapIds(
  org.Hs.eg.db,
  keys = roj_rds@assays[["RNA"]]@counts@Dimnames[[1]],
  column = "ENTREZID",
  keytype = "SYMBOL",
  multiVals = "first"
)

#annotate DE genes in the differential expression data frame
entrez_anno <- bitr(diffex_ad_svm$gene, fromType = "SYMBOL",
        toType = c("ENTREZID"),
        OrgDb = org.Hs.eg.db)

diffex_ad_svm <- merge(x = diffex_ad_svm, y = entrez_anno, by.x = "gene", by.y = "SYMBOL", all.x = T)

diffex_ad_svm <- diffex_ad_svm %>% relocate(ENTREZID, .after = gene)
diffex_ad_svm <- diffex_ad_svm %>% rename(entrezID = ENTREZID)

#which genes do not have an entrezID
table(is.na(diffex_ad_svm$entrezID))
diffex_ad_svm$gene[is.na(diffex_ad_svm$entrezID)]
```

```{r prepare input gene lists - over enrichment analysis}
#basal III
DEAD_basalIII <- diffex_ad_svm[diffex_ad_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_III_p_val_adj), "entrezID"]
DEAD_basalIIIup <- diffex_ad_svm[diffex_ad_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_III_p_val_adj) & diffex_ad_svm$AD_basal_III_avg_log2FC > 0, "entrezID"]
DEAD_basalIIIdown <- diffex_ad_svm[diffex_ad_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_III_p_val_adj) & diffex_ad_svm$AD_basal_III_avg_log2FC < 0, "entrezID"]

#basal II
DEAD_basalII <- diffex_ad_svm[diffex_ad_svm$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_II_p_val_adj), "entrezID"]
DEAD_basalIIup <- diffex_ad_svm[diffex_ad_svm$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_II_p_val_adj) & diffex_ad_svm$AD_basal_II_avg_log2FC > 0, "entrezID"]
DEAD_basalIIdown <- diffex_ad_svm[diffex_ad_svm$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_II_p_val_adj) & diffex_ad_svm$AD_basal_II_avg_log2FC < 0, "entrezID"]

#basal I
DEAD_basalI <- diffex_ad_svm[diffex_ad_svm$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_I_p_val_adj), "entrezID"]
DEAD_basalIup <- diffex_ad_svm[diffex_ad_svm$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_I_p_val_adj) & diffex_ad_svm$AD_basal_I_avg_log2FC > 0, "entrezID"]
DEAD_basalIdown <- diffex_ad_svm[diffex_ad_svm$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_I_p_val_adj) & diffex_ad_svm$AD_basal_I_avg_log2FC < 0, "entrezID"]

#basal IV
DEAD_basalIV <- diffex_ad_svm[diffex_ad_svm$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_IV_p_val_adj), "entrezID"]
DEAD_basalIVup <- diffex_ad_svm[diffex_ad_svm$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_IV_p_val_adj) & diffex_ad_svm$AD_basal_IV_avg_log2FC > 0, "entrezID"]
DEAD_basalIVdown <- diffex_ad_svm[diffex_ad_svm$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_basal_IV_p_val_adj) & diffex_ad_svm$AD_basal_IV_avg_log2FC < 0, "entrezID"]

#spinosum
DEAD_spinosum <- diffex_ad_svm[diffex_ad_svm$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_spinosum_p_val_adj), "entrezID"]
DEAD_spinosumup <- diffex_ad_svm[diffex_ad_svm$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_spinosum_p_val_adj) & diffex_ad_svm$AD_spinosum_avg_log2FC > 0, "entrezID"]
DEAD_spinosumdown <- diffex_ad_svm[diffex_ad_svm$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_spinosum_p_val_adj) & diffex_ad_svm$AD_spinosum_avg_log2FC < 0, "entrezID"]

#granulosum
DEAD_granulosum <- diffex_ad_svm[diffex_ad_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_granulosum_p_val_adj), "entrezID"]
DEAD_granulosumup <- diffex_ad_svm[diffex_ad_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_granulosum_p_val_adj) & diffex_ad_svm$AD_granulosum_avg_log2FC > 0, "entrezID"]
DEAD_granulosumdown <- diffex_ad_svm[diffex_ad_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_svm$AD_granulosum_p_val_adj) & diffex_ad_svm$AD_granulosum_avg_log2FC < 0, "entrezID"]
```

##GO analysis

```{r GO over enrichment}
DEAD_basalIII_GO <-  enrichGO(gene = DEAD_basalIII, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIIIup_GO <-  enrichGO(gene = DEAD_basalIIIup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIIIdown_GO <-  enrichGO(gene = DEAD_basalIIIdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEAD_basalII_GO <-  enrichGO(gene = DEAD_basalII, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIIup_GO <-  enrichGO(gene = DEAD_basalIIup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIIdown_GO <-  enrichGO(gene = DEAD_basalIIdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEAD_basalI_GO <-  enrichGO(gene = DEAD_basalI, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIup_GO <-  enrichGO(gene = DEAD_basalIup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIdown_GO <-  enrichGO(gene = DEAD_basalIdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEAD_basalIV_GO <-  enrichGO(gene = DEAD_basalIV, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIVup_GO <-  enrichGO(gene = DEAD_basalIVup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIVdown_GO <-  enrichGO(gene = DEAD_basalIVdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEAD_spinosum_GO <-  enrichGO(gene = DEAD_spinosum, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_spinosumup_GO <-  enrichGO(gene = DEAD_spinosumup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_spinosumdown_GO <-  enrichGO(gene = DEAD_spinosumdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEAD_granulosum_GO <-  enrichGO(gene = DEAD_granulosum, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_granulosumup_GO <-  enrichGO(gene = DEAD_granulosumup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_granulosumdown_GO <-  enrichGO(gene = DEAD_granulosumdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

#save
list <- list("GO DE AD basal III" = DEAD_basalIII_GO,
             "GO DE AD basal IIIup" = DEAD_basalIIIup_GO,
             "GO DE AD basal IIIdown" = DEAD_basalIIIdown_GO,
             "GO DE AD basal II" = DEAD_basalII_GO,
             "GO DE AD basal IIup" = DEAD_basalIIup_GO,
             "GO DE AD basal IIdown" = DEAD_basalIIdown_GO,
             "GO DE AD basal I" = DEAD_basalI_GO,
             "GO DE AD basal Iup" = DEAD_basalIup_GO,
             "GO DE AD basal Idown" = DEAD_basalIdown_GO,
             "GO DE AD basal IV" = DEAD_basalIV_GO,
             "GO DE AD basal IVup" = DEAD_basalIVup_GO,
             "GO DE AD basal IVdown" = DEAD_basalIVdown_GO,
             "GO DE AD spinosum" = DEAD_spinosum_GO,
             "GO DE AD spinosumup" = DEAD_spinosumup_GO,
             "GO DE AD spinosumdown" = DEAD_spinosumdown_GO,
             "GO DE AD granulosum" = DEAD_granulosum_GO,
             "GO DE AD granulosumup" = DEAD_granulosumup_GO,
             "GO DE AD granulosumdown" = DEAD_granulosumdown_GO)
# write.xlsx(list, file = "GO DE AD.xlsx")

# ##visualization
# GO_AD_vis <- DEAD_basalIIIup_GO[c(1, 4)]
# GO_AD_vis <- rbind(GO_AD_vis, DEAD_basalIIup_GO[c(1, 2)])
# GO_AD_vis <- rbind(GO_AD_vis, DEAD_basalIup_GO[c(1, 2)])
# GO_AD_vis <- rbind(GO_AD_vis, DEAD_basalIVup_GO[c(8, 22)])
# GO_AD_vis <- rbind(GO_AD_vis, DEAD_spinosum_GO[c(9, 10)])
# GO_AD_vis <- rbind(GO_AD_vis, DEAD_granulosumup_GO[c(3, 17, 18)])
# GO_AD_vis$celltype <- c(rep("basal III", 2), rep("basal II", 2), rep("basal I", 2), rep("basal IV", 2), rep("spinosum", 2), rep("granulosum", 3))
# GO_AD_vis$celltype <- factor(GO_AD_vis$celltype, levels = c("basal III", "basal II", "basal I", "basal IV", "spinosum", "granulosum"))

ggplot(GO_AD_vis, aes(x = Count, y = Description)) +
        geom_col(aes(fill = p.adjust)) +
        facet_grid("celltype", scale ="free") +
        scale_fill_gradient(high = "blue", low = "red", guide = guide_colourbar(reverse = TRUE)) +
        theme_bw(base_size = 13) + theme(axis.text.y = element_text(size = 13)) +
        ylab(NULL) + xlab("count")

# cnetplot(DEAD_basalIIIup_GO, showCategory = DEAD_basalIIIup_GOdf$Description[c(72, 175)])
# 
# #cnetplot custom GO's
# DEAD_basalIIIup_GOdf <- as.data.frame(DEAD_basalIIIup_GO)
# cnetplot(DEAD_basalIIIup_GO, categorySize = "geneNum", foldChange = genelist_c31,
#          showCategory = DEAD_basalIIIup_GOdf$Description[c(72, 175)]) +
#   scale_color_gradient2(name = "fold change", low = "blue", mid = "gray95", high = "red", midpoint = 0,
#                         limits = c(min(genelist_c31) , max(genelist_c31))) +
#   ggtitle("fold change TCDD vs control", subtitle = "padj < 0.05, background: baseMean > 1, BP & MF & CC together")
```

# Functional analysis AD vs HC
##Prepare gene lists

```{r annotate gene symbol with entrezID}
#annotate all expressed background genes
background <-  mapIds(
  org.Hs.eg.db,
  keys = roj_rds@assays[["RNA"]]@counts@Dimnames[[1]],
  column = "ENTREZID",
  keytype = "SYMBOL",
  multiVals = "first"
)

#annotate DE genes in the differential expression data frame
entrez_anno <- bitr(diffex_ad_hc_svm$gene, fromType = "SYMBOL",
        toType = c("ENTREZID"),
        OrgDb = org.Hs.eg.db)

# diffex_ad_hc_svm <- merge(x = diffex_ad_hc_svm, y = entrez_anno, by.x = "gene", by.y = "SYMBOL", all.x = T)
# 
# diffex_ad_hc_svm <- diffex_ad_hc_svm %>% relocate(ENTREZID, .after = gene)
# diffex_ad_hc_svm <- diffex_ad_hc_svm %>% rename(entrezID = ENTREZID)

table(is.na(diffex_ad_hc_svm$entrezID))
```

```{r prepare input gene lists - over enrichment analysis}
#basal III
DEAD_basalIII <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_III_p_val_adj), "entrezID"]
DEAD_basalIIIup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_III_p_val_adj) & diffex_ad_hc_svm$AD_basal_III_avg_log2FC > 0, "entrezID"]
DEAD_basalIIIdown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_III_p_val_adj) & diffex_ad_hc_svm$AD_basal_III_avg_log2FC < 0, "entrezID"]

#basal II
DEAD_basalII <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_II_p_val_adj), "entrezID"]
DEAD_basalIIup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_II_p_val_adj) & diffex_ad_hc_svm$AD_basal_II_avg_log2FC > 0, "entrezID"]
DEAD_basalIIdown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_II_p_val_adj) & diffex_ad_hc_svm$AD_basal_II_avg_log2FC < 0, "entrezID"]

#basal I
DEAD_basalI <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_I_p_val_adj), "entrezID"]
DEAD_basalIup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_I_p_val_adj) & diffex_ad_hc_svm$AD_basal_I_avg_log2FC > 0, "entrezID"]
DEAD_basalIdown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_I_p_val_adj) & diffex_ad_hc_svm$AD_basal_I_avg_log2FC < 0, "entrezID"]

#basal IV
DEAD_basalIV <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_IV_p_val_adj), "entrezID"]
DEAD_basalIVup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_IV_p_val_adj) & diffex_ad_hc_svm$AD_basal_IV_avg_log2FC > 0, "entrezID"]
DEAD_basalIVdown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_IV_p_val_adj) & diffex_ad_hc_svm$AD_basal_IV_avg_log2FC < 0, "entrezID"]

#spinosum
DEAD_spinosum <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_spinosum_p_val_adj), "entrezID"]
DEAD_spinosumup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_spinosum_p_val_adj) & diffex_ad_hc_svm$AD_spinosum_avg_log2FC > 0, "entrezID"]
DEAD_spinosumdown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_spinosum_p_val_adj) & diffex_ad_hc_svm$AD_spinosum_avg_log2FC < 0, "entrezID"]

#granulosum
DEAD_granulosum <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_granulosum_p_val_adj), "entrezID"]
DEAD_granulosumup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_granulosum_p_val_adj) & diffex_ad_hc_svm$AD_granulosum_avg_log2FC > 0, "entrezID"]
DEAD_granulosumdown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_granulosum_p_val_adj) & diffex_ad_hc_svm$AD_granulosum_avg_log2FC < 0, "entrezID"]
```

```{r prepare input gene lists - gsea}
Idents(roj_rds) <- roj_rds$svm_annotation

#basal III
DEAD_basalIII <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "basal III", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
DEAD_basalIII_gsea <- DEAD_basalIII[ , "avg_log2FC"]
names(DEAD_basalIII_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIII), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalIII_gsea <- sort(DEAD_basalIII_gsea, decreasing = T)

DEAD_basalIIIup_gsea <- DEAD_basalIII[DEAD_basalIII$avg_log2FC > 0, "avg_log2FC"]
names(DEAD_basalIIIup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIII)[DEAD_basalIII$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalIIIup_gsea <- sort(DEAD_basalIIIup_gsea, decreasing = T)

DEAD_basalIIIdown_gsea <- DEAD_basalIII[DEAD_basalIII$avg_log2FC < 0, "avg_log2FC"]
names(DEAD_basalIIIdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIII)[DEAD_basalIII$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalIIIdown_gsea <- sort(DEAD_basalIIIdown_gsea, decreasing = T)

#basal II
DEAD_basalII <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "basal II", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
DEAD_basalII_gsea <- DEAD_basalII[ , "avg_log2FC"]
names(DEAD_basalII_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalII), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalII_gsea <- sort(DEAD_basalII_gsea, decreasing = T)

DEAD_basalIIup_gsea <- DEAD_basalII[DEAD_basalII$avg_log2FC > 0, "avg_log2FC"]
names(DEAD_basalIIup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalII)[DEAD_basalII$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalIIup_gsea <- sort(DEAD_basalIIup_gsea, decreasing = T)

DEAD_basalIIdown_gsea <- DEAD_basalII[DEAD_basalII$avg_log2FC < 0, "avg_log2FC"]
names(DEAD_basalIIdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalII)[DEAD_basalII$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalIIdown_gsea <- sort(DEAD_basalIIdown_gsea, decreasing = T)

#basal I
DEAD_basalI <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "basal I", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
DEAD_basalI_gsea <- DEAD_basalI[ , "avg_log2FC"]
names(DEAD_basalI_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalI), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalI_gsea <- sort(DEAD_basalI_gsea, decreasing = T)

DEAD_basalIup_gsea <- DEAD_basalI[DEAD_basalI$avg_log2FC > 0, "avg_log2FC"]
names(DEAD_basalIup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalI)[DEAD_basalI$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalIup_gsea <- sort(DEAD_basalIup_gsea, decreasing = T)

DEAD_basalIdown_gsea <- DEAD_basalI[DEAD_basalI$avg_log2FC < 0, "avg_log2FC"]
names(DEAD_basalIdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalI)[DEAD_basalI$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalIdown_gsea <- sort(DEAD_basalIdown_gsea, decreasing = T)

#basal IV
DEAD_basalIV <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "basal IV", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
DEAD_basalIV_gsea <- DEAD_basalIV[ , "avg_log2FC"]
names(DEAD_basalIV_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIV), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalIV_gsea <- sort(DEAD_basalIV_gsea, decreasing = T)

DEAD_basalIVup_gsea <- DEAD_basalIV[DEAD_basalIV$avg_log2FC > 0, "avg_log2FC"]
names(DEAD_basalIVup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIV)[DEAD_basalIV$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalIVup_gsea <- sort(DEAD_basalIVup_gsea, decreasing = T)

DEAD_basalIVdown_gsea <- DEAD_basalIV[DEAD_basalIV$avg_log2FC < 0, "avg_log2FC"]
names(DEAD_basalIVdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_basalIV)[DEAD_basalIV$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_basalIVdown_gsea <- sort(DEAD_basalIVdown_gsea, decreasing = T)

#spinosum 
DEAD_spinosum <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "stratum spinosum", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
DEAD_spinosum_gsea <- DEAD_spinosum[ , "avg_log2FC"]
names(DEAD_spinosum_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_spinosum), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_spinosum_gsea <- sort(DEAD_spinosum_gsea, decreasing = T)

DEAD_spinosumup_gsea <- DEAD_spinosum[DEAD_spinosum$avg_log2FC > 0, "avg_log2FC"]
names(DEAD_spinosumup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_spinosum)[DEAD_spinosum$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_spinosumup_gsea <- sort(DEAD_spinosumup_gsea, decreasing = T)

DEAD_spinosumdown_gsea <- DEAD_spinosum[DEAD_spinosum$avg_log2FC < 0, "avg_log2FC"]
names(DEAD_spinosumdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_spinosum)[DEAD_spinosum$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_spinosumdown_gsea <- sort(DEAD_spinosumdown_gsea, decreasing = T)

#granulosum 
DEAD_granulosum <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "stratum granulosum", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0.1) #logFC threshhold is 0.25 by default
DEAD_granulosum_gsea <- DEAD_granulosum[ , "avg_log2FC"]
names(DEAD_granulosum_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_granulosum), column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_granulosum_gsea <- sort(DEAD_granulosum_gsea, decreasing = T)

DEAD_granulosumup_gsea <- DEAD_granulosum[DEAD_granulosum$avg_log2FC > 0, "avg_log2FC"]
names(DEAD_granulosumup_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_granulosum)[DEAD_granulosum$avg_log2FC > 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_granulosumup_gsea <- sort(DEAD_granulosumup_gsea, decreasing = T)

DEAD_granulosumdown_gsea <- DEAD_granulosum[DEAD_granulosum$avg_log2FC < 0, "avg_log2FC"]
names(DEAD_granulosumdown_gsea) <- mapIds(org.Hs.eg.db, keys = rownames(DEAD_granulosum)[DEAD_granulosum$avg_log2FC < 0], column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
DEAD_granulosumdown_gsea <- sort(DEAD_granulosumdown_gsea, decreasing = T)
```

##GO analysis

```{r GO over enrichment}
DEAD_basalIII_GO <-  enrichGO(gene = DEAD_basalIII, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIIIup_GO <-  enrichGO(gene = DEAD_basalIIIup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIIIdown_GO <-  enrichGO(gene = DEAD_basalIIIdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEAD_basalII_GO <-  enrichGO(gene = DEAD_basalII, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIIup_GO <-  enrichGO(gene = DEAD_basalIIup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIIdown_GO <-  enrichGO(gene = DEAD_basalIIdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEAD_basalI_GO <-  enrichGO(gene = DEAD_basalI, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIup_GO <-  enrichGO(gene = DEAD_basalIup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIdown_GO <-  enrichGO(gene = DEAD_basalIdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEAD_basalIV_GO <-  enrichGO(gene = DEAD_basalIV, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIVup_GO <-  enrichGO(gene = DEAD_basalIVup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_basalIVdown_GO <-  enrichGO(gene = DEAD_basalIVdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEAD_spinosum_GO <-  enrichGO(gene = DEAD_spinosum, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_spinosumup_GO <-  enrichGO(gene = DEAD_spinosumup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_spinosumdown_GO <-  enrichGO(gene = DEAD_spinosumdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

DEAD_granulosum_GO <-  enrichGO(gene = DEAD_granulosum, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_granulosumup_GO <-  enrichGO(gene = DEAD_granulosumup, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)
DEAD_granulosumdown_GO <-  enrichGO(gene = DEAD_granulosumdown, OrgDb = org.Hs.eg.db, universe = background, keyType = 'ENTREZID', ont = "BP", readable = T)

#save
list <- list("GO DE AD basal III" = DEAD_basalIII_GO,
             "GO DE AD basal IIIup" = DEAD_basalIIIup_GO,
             "GO DE AD basal IIIdown" = DEAD_basalIIIdown_GO,
             "GO DE AD basal II" = DEAD_basalII_GO,
             "GO DE AD basal IIup" = DEAD_basalIIup_GO,
             "GO DE AD basal IIdown" = DEAD_basalIIdown_GO,
             "GO DE AD basal I" = DEAD_basalI_GO,
             "GO DE AD basal Iup" = DEAD_basalIup_GO,
             "GO DE AD basal Idown" = DEAD_basalIdown_GO,
             "GO DE AD basal IV" = DEAD_basalIV_GO,
             "GO DE AD basal IVup" = DEAD_basalIVup_GO,
             "GO DE AD basal IVdown" = DEAD_basalIVdown_GO,
             "GO DE AD spinosum" = DEAD_spinosum_GO,
             "GO DE AD spinosumup" = DEAD_spinosumup_GO,
             "GO DE AD spinosumdown" = DEAD_spinosumdown_GO,
             "GO DE AD granulosum" = DEAD_granulosum_GO,
             "GO DE AD granulosumup" = DEAD_granulosumup_GO,
             "GO DE AD granulosumdown" = DEAD_granulosumdown_GO)
write.xlsx(list, file = "GO DE AD.xlsx")

##visualization
barplot(DEAD_basalIIIup_GO, showCategory = DEAD_basalIIIup_GO$Description[c(186, 268)]) #proliferation
barplot(DEAD_basalIIIup_GO, showCategory = DEAD_basalIIIup_GO$Description[c(47, 7, 58, 148)]) #inflammation
barplot(DEAD_basalIIIup_GO, showCategory = DEAD_basalIIIup_GO$Description[c(19, 72, 244)]) #differentiation + migration
barplot(DEAD_basalIIIup_GO, showCategory = DEAD_basalIIIup_GO$Description[c(1, 31)]) #apoptosis, NFkB

barplot(DEAD_basalIIIup_GO, showCategory = DEAD_basalIIIup_GO$Description[c(186, 268, 47, 7, 58, 148, 19, 72, 244, 1, 31)]) 

cnetplot(DEAD_basalIIIup_GO, showCategory = DEAD_basalIIIup_GOdf$Description[c(72, 175)])
# cnetplot(DEAD_basalIIIup_GO, showCategory = DEAD_basalIIIup_GOdf$Description[c(72, 175, 268, 371)])

#change order manually
# DEAD_basalIIIup_GOdf <- as.data.frame(DEAD_basalIIIup_GO)
# DEAD_basalIIIup_GOdf <- DEAD_basalIIIup_GOdf[c(186, 198, 268), ]
# DEAD_basalIIIup_GOdf$x <- factor(data1$x, levels = c("B", "D", "E", "C", "A"))
# 
# ggplot(DEAD_basalIIIup_GOdf[c(186, 198, 268, 47, 7, 58, 148), ], aes(x = Count, y = Description)) + 
#         geom_col(aes(fill = p.adjust)) +
#         scale_fill_gradient(high = "blue", low = "red", guide = guide_colourbar(reverse = TRUE)) +
#         theme_bw(base_size = 13) + theme(axis.text.y = element_text(size = 13)) +
#         ylab(NULL) + xlab("count")
# #reorder(Description, desc(p.adjust))

#cnetplot custom GO's
DEAD_basalIIIup_GOdf <- as.data.frame(DEAD_basalIIIup_GO)
cnetplot(DEAD_basalIIIup_GO, categorySize = "geneNum", foldChange = genelist_c31,
         showCategory = DEAD_basalIIIup_GOdf$Description[c(72, 175)]) +
  scale_color_gradient2(name = "fold change", low = "blue", mid = "gray95", high = "red", midpoint = 0,
                        limits = c(min(genelist_c31) , max(genelist_c31))) +
  ggtitle("fold change TCDD vs control", subtitle = "padj < 0.05, background: baseMean > 1, BP & MF & CC together")


```

```{r GO GSEA}
#do GO GSEA analysis
#basal III
DEAD_basalIII_gseaGO <- gseGO(geneList = DEAD_basalIII_gsea, OrgDb = org.Hs.eg.db)
DEAD_basalIII_gseaGO <- setReadable(DEAD_basalIII_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIIIup_gseaGO <- gseGO(geneList = DEAD_basalIIIup_gsea, OrgDb = org.Hs.eg.db, scoreType = "pos")
DEAD_basalIIIup_gseaGO <- setReadable(DEAD_basalIIIup_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIIIdown_gseaGO <- gseGO(geneList = DEAD_basalIIIdown_gsea, OrgDb = org.Hs.eg.db, scoreType = "neg")
DEAD_basalIIIdown_gseaGO <- setReadable(DEAD_basalIIIdown_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#basal II
DEAD_basalII_gseaGO <- gseGO(geneList = DEAD_basalII_gsea, OrgDb = org.Hs.eg.db)
DEAD_basalII_gseaGO <- setReadable(DEAD_basalII_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIIup_gseaGO <- gseGO(geneList = DEAD_basalIIup_gsea, OrgDb = org.Hs.eg.db, scoreType = "pos")
DEAD_basalIIup_gseaGO <- setReadable(DEAD_basalIIup_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIIdown_gseaGO <- gseGO(geneList = DEAD_basalIIdown_gsea, OrgDb = org.Hs.eg.db, scoreType = "neg")
DEAD_basalIIdown_gseaGO <- setReadable(DEAD_basalIIdown_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#basal I
DEAD_basalI_gseaGO <- gseGO(geneList = DEAD_basalI_gsea, OrgDb = org.Hs.eg.db)
DEAD_basalI_gseaGO <- setReadable(DEAD_basalI_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIup_gseaGO <- gseGO(geneList = DEAD_basalIup_gsea, OrgDb = org.Hs.eg.db, scoreType = "pos")
DEAD_basalIup_gseaGO <- setReadable(DEAD_basalIup_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIdown_gseaGO <- gseGO(geneList = DEAD_basalIdown_gsea, OrgDb = org.Hs.eg.db, scoreType = "neg")
DEAD_basalIdown_gseaGO <- setReadable(DEAD_basalIdown_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#basal IV
DEAD_basalIV_gseaGO <- gseGO(geneList = DEAD_basalIV_gsea, OrgDb = org.Hs.eg.db)
DEAD_basalIV_gseaGO <- setReadable(DEAD_basalIV_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIVup_gseaGO <- gseGO(geneList = DEAD_basalIVup_gsea, OrgDb = org.Hs.eg.db, scoreType = "pos")
DEAD_basalIVup_gseaGO <- setReadable(DEAD_basalIVup_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIVdown_gseaGO <- gseGO(geneList = DEAD_basalIVdown_gsea, OrgDb = org.Hs.eg.db, scoreType = "neg")
DEAD_basalIVdown_gseaGO <- setReadable(DEAD_basalIVdown_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#spinosum
DEAD_spinosum_gseaGO <- gseGO(geneList = DEAD_spinosum_gsea, OrgDb = org.Hs.eg.db)
DEAD_spinosum_gseaGO <- setReadable(DEAD_spinosum_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_spinosumup_gseaGO <- gseGO(geneList = DEAD_spinosumup_gsea, OrgDb = org.Hs.eg.db, scoreType = "pos")
DEAD_spinosumup_gseaGO <- setReadable(DEAD_spinosumup_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_spinosumdown_gseaGO <- gseGO(geneList = DEAD_spinosumdown_gsea, OrgDb = org.Hs.eg.db, scoreType = "neg")
DEAD_spinosumdown_gseaGO <- setReadable(DEAD_spinosumdown_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#granulosum
DEAD_granulosum_gseaGO <- gseGO(geneList = DEAD_granulosum_gsea, OrgDb = org.Hs.eg.db)
DEAD_granulosum_gseaGO <- setReadable(DEAD_granulosum_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_granulosumup_gseaGO <- gseGO(geneList = DEAD_granulosumup_gsea, OrgDb = org.Hs.eg.db, scoreType = "pos")
DEAD_granulosumup_gseaGO <- setReadable(DEAD_granulosumup_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_granulosumdown_gseaGO <- gseGO(geneList = DEAD_granulosumdown_gsea, OrgDb = org.Hs.eg.db, scoreType = "neg")
DEAD_granulosumdown_gseaGO <- setReadable(DEAD_granulosumdown_gseaGO, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#save
list <- list("GSEA GO AD basal III " = DEAD_basalIII_gseaGO,
             "GSEA GO AD basal IIIup" = DEAD_basalIIIup_gseaGO,
             "GSEA GO AD basal IIIdown" = DEAD_basalIIIdown_gseaGO,
             "GSEA GO AD basal II" = DEAD_basalII_gseaGO,
             "GSEA GO AD basal IIup" = DEAD_basalIIup_gseaGO,
             "GSEA GO AD basal IIdown" = DEAD_basalIIdown_gseaGO,
             "GSEA GO AD basal I" = DEAD_basalI_gseaGO,
             "GSEA GO AD basal Iup" = DEAD_basalIup_gseaGO,
             "GSEA GO AD basal Idown" = DEAD_basalIdown_gseaGO,
             "GSEA GO AD basal IV" = DEAD_basalIV_gseaGO,
             "GSEA GO AD basal IVup" = DEAD_basalIVup_gseaGO,
             "GSEA GO AD basal IVdown" = DEAD_basalIVdown_gseaGO,
             "GSEA GO AD spinosum" = DEAD_spinosum_gseaGO,
             "GSEA GO AD spinosum up" = DEAD_spinosumup_gseaGO,
             "GSEA GO AD spinosum down" = DEAD_spinosumdown_gseaGO,
             "GSEA GO AD granulosum" = DEAD_granulosum_gseaGO,
             "GSEA GO AD granulosum up" = DEAD_granulosumup_gseaGO,
             "GSEA GO AD granulosum down" = DEAD_granulosumdown_gseaGO)
write.xlsx(list, file = "GO GSEA logFC 0.1 AD.xlsx")
```

```{r GO basal III KRT15/ASS1}
GO_KRT15_hc = enrichGO(
  gene = hc[hc$avg_log2FC > 0, ]$gene,
  OrgDb = org.Hs.eg.db,
  universe = roj_rds@assays[["RNA"]]@counts@Dimnames[[1]],
  keyType = 'SYMBOL',
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff  = 0.05
)
hc[hc$avg_log2FC > 0, ]$gene
# hc[hc$avg_log2FC < 0, ]$gene

GO_ASS1_hc = enrichGO(
  gene = hc[hc$avg_log2FC < 0, ]$gene,
  OrgDb = org.Hs.eg.db,
  universe = roj_rds@assays[["RNA"]]@counts@Dimnames[[1]],
  keyType = 'SYMBOL',
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff  = 0.05
)

dogo_hc = enrichGO(gene = hc[hc$avg_log2FC<0,]$gene,OrgDb = org.Hs.eg.db, 
                   universe= roj_rds@assays[["RNA"]]@counts@Dimnames[[1]], 
                  keyType = 'SYMBOL',ont = "BP",pAdjustMethod = "BH",pvalueCutoff = 0.01, qvalueCutoff  = 0.05)
GO_ASS1_hc

GO_KRT15_ad = enrichGO(
  gene = ad[ad$avg_log2FC > 0, ]$gene,
  OrgDb = org.Hs.eg.db,
  universe = roj_rds@assays[["RNA"]]@counts@Dimnames[[1]],
  keyType = 'SYMBOL',
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff  = 0.05
)

GO_ASS1_ad = enrichGO(
  gene = ad[ad$avg_log2FC < 0, ]$gene,
  OrgDb = org.Hs.eg.db,
  universe = roj_rds@assays[["RNA"]]@counts@Dimnames[[1]],
  keyType = 'SYMBOL',
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff  = 0.05
)

# write.xlsx(rbind("GO HC KRT15+ basal III", as.data.frame(GO_KRT15_hc), rep("", 29),
#                  "GO HC ASS1+ basal III", as.data.frame(GO_ASS1_hc), rep("", 29),
#                  "GO AD KRT15+ basal III", as.data.frame(GO_KRT15_ad), rep("", 29),
#                  "GO AD ASS1+ basal III", as.data.frame(GO_ASS1_ad), rep("", 29)), file = "GO_basalIIIsubtypes.xlsx")
```

```{r GO basal IV}
basalIV <- diffex_hc_svm[diffex_hc_svm$HC_basal_IV_p_val_adj < 0.05 & !is.na(diffex_hc_svm$HC_basal_IV_p_val_adj), "gene"]

GO_basalIV_hc = enrichGO(
  gene = basalIV,
  OrgDb = org.Hs.eg.db,
  universe = roj_rds@assays[["RNA"]]@counts@Dimnames[[1]],
  keyType = 'SYMBOL',
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff  = 0.05
)

GO_basalIV_hc_df <- as.data.frame(GO_basalIV_hc)
GO_basalIV_hc_df$ratio <- sapply(strsplit(GO_basalIV_hc_df$GeneRatio, split = "/"), function(x) as.numeric(x[1]) / as.numeric(x[2]))/
                                        sapply(strsplit(GO_basalIV_hc_df$BgRatio, split = "/"), function(x) as.numeric(x[1]) / as.numeric(x[2]))
# / as.numeric(x[2]
ggplot(GO_basalIV_hc_df[c(8, 18, 4, 24), ], aes(x = ratio, y = reorder(Description, desc(p.adjust)))) + 
        geom_point(aes(color = p.adjust), size = 6) +
        scale_colour_gradient(high = "blue", low = "red", guide = guide_colourbar(reverse = TRUE)) +
        theme_bw(base_size = 13) + theme(axis.text.y = element_text(size = 13)) +
        ylab(NULL) + xlab("gene to background ratio")
```

##KEGG pathway enrichment

```{r KEGG over enrichment}
#do KEGG enrichment analysis
DEAD_basalIII_kegg <- enrichKEGG(gene = DEAD_basalIII, universe = background, pvalueCutoff  = 0.05)
DEAD_basalIIIup_kegg <- enrichKEGG(gene = DEAD_basalIIIup, universe = background, pvalueCutoff  = 0.05)
DEAD_basalIIIdown_kegg <- enrichKEGG(gene = DEAD_basalIIIdown, universe = background, pvalueCutoff  = 0.05)

DEAD_basalII_kegg <- enrichKEGG(gene = DEAD_basalII, universe = background, pvalueCutoff  = 0.05)
DEAD_basalIIup_kegg <- enrichKEGG(gene = DEAD_basalIIup, universe = background, pvalueCutoff  = 0.05)
DEAD_basalIIdown_kegg <- enrichKEGG(gene = DEAD_basalIIdown, universe = background, pvalueCutoff  = 0.05)

DEAD_basalI_kegg <- enrichKEGG(gene = DEAD_basalI, universe = background, pvalueCutoff  = 0.05)
DEAD_basalIup_kegg <- enrichKEGG(gene = DEAD_basalIup, universe = background, pvalueCutoff  = 0.05)
DEAD_basalIdown_kegg <- enrichKEGG(gene = DEAD_basalIdown, universe = background, pvalueCutoff  = 0.05)

DEAD_basalIV_kegg <- enrichKEGG(gene = DEAD_basalIV, universe = background, pvalueCutoff  = 0.05)
DEAD_basalIVup_kegg <- enrichKEGG(gene = DEAD_basalIVup, universe = background, pvalueCutoff  = 0.05)
DEAD_basalIVdown_kegg <- enrichKEGG(gene = DEAD_basalIVdown, universe = background, pvalueCutoff  = 0.05)

DEAD_spin_kegg <- enrichKEGG(gene = DEAD_spin, universe = background, pvalueCutoff  = 0.05)
DEAD_spinup_kegg <- enrichKEGG(gene = DEAD_spinup, universe = background, pvalueCutoff  = 0.05)
DEAD_spindown_kegg <- enrichKEGG(gene = DEAD_spindown, universe = background, pvalueCutoff  = 0.05)

DEAD_gran_kegg <- enrichKEGG(gene = DEAD_gran, universe = background, pvalueCutoff  = 0.05)
DEAD_granup_kegg <- enrichKEGG(gene = DEAD_granup, universe = background, pvalueCutoff  = 0.05)
DEAD_grandown_kegg <- enrichKEGG(gene = DEAD_grandown, universe = background, pvalueCutoff  = 0.05)

# make result read.table
DEAD_basalIII_kegg <- setReadable(DEAD_basalIII_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_basalIIIup_kegg <- setReadable(DEAD_basalIIIup_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_basalIIIdown_kegg <- setReadable(DEAD_basalIIIdown_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalII_kegg <- setReadable(DEAD_basalII_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_basalIIup_kegg <- setReadable(DEAD_basalIIup_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_basalIIdown_kegg <- setReadable(DEAD_basalIIdown_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalI_kegg <- setReadable(DEAD_basalI_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_basalIup_kegg <- setReadable(DEAD_basalIup_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_basalIdown_kegg <- setReadable(DEAD_basalIdown_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIV_kegg <- setReadable(DEAD_basalIV_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_basalIVup_kegg <- setReadable(DEAD_basalIVup_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_basalIVdown_kegg <- setReadable(DEAD_basalIVdown_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_spin_kegg <- setReadable(DEAD_spin_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_spinup_kegg <- setReadable(DEAD_spinup_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_spindown_kegg <- setReadable(DEAD_spindown_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_gran_kegg <- setReadable(DEAD_gran_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_granup_kegg <- setReadable(DEAD_granup_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")
DEAD_grandown_kegg <- setReadable(DEAD_grandown_kegg, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#save
list <- list("KEGG DE AD basal III" = DEAD_basalIII_kegg,
             "KEGG DE AD basal IIIup" = DEAD_basalIIIup_kegg,
             "KEGG DE AD basal IIIdown" = DEAD_basalIIIdown_kegg,
             "KEGG DE AD basal II" = DEAD_basalII_kegg,
             "KEGG DE AD basal IIup" = DEAD_basalIIup_kegg,
             "KEGG DE AD basal IIdown" = DEAD_basalIIdown_kegg,
             "KEGG DE AD basal I" = DEAD_basalI_kegg,
             "KEGG DE AD basal Iup" = DEAD_basalIup_kegg,
             "KEGG DE AD basal Idown" = DEAD_basalIdown_kegg,
             "KEGG DE AD basal IV" = DEAD_basalIV_kegg,
             "KEGG DE AD basal IVup" = DEAD_basalIVup_kegg,
             "KEGG DE AD basal IVdown" = DEAD_basalIVdown_kegg,
             "KEGG DE AD spinosum" = DEAD_spin_kegg,
             "KEGG DE AD spinosum up" = DEAD_spinup_kegg,
             "KEGG DE AD spinosum down" = DEAD_spindown_kegg,
             "KEGG DE AD granulosum" = DEAD_gran_kegg,
             "KEGG DE AD granulosum up" = DEAD_granup_kegg,
             "KEGG DE AD granulosum down" = DEAD_grandown_kegg)
write.xlsx(list, file = "KEGG diffexp AD.xlsx")

##visualization
barplot(DEAD_basalIIIup_kegg, showCategory = DEAD_basalIIIup_kegg$Description[c(40, 42)]) #proliferation
barplot(DEAD_basalIIIup_kegg, showCategory = DEAD_basalIIIup_kegg$Description[c(1, 35)]) #inflammation
barplot(DEAD_basalIIIup_kegg, showCategory = DEAD_basalIIIup_kegg$Description[c(19, 22)]) #differentiation, migration
barplot(DEAD_basalIIIup_kegg, showCategory = DEAD_basalIIIup_kegg$Description[c(4)]) #apoptosis, NFkB

barplot(DEAD_basalIIIup_kegg, showCategory = DEAD_basalIIIup_kegg$Description[c(40, 42, 1, 35, 19, 22, 4)])
```

```{r KEGG GSEA}
#do kegg GSEA analysis
#basal III
DEAD_basalIII_gseaKEGG <- gseKEGG(geneList = DEAD_basalIII_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05)
DEAD_basalIII_gseaKEGG <- setReadable(DEAD_basalIII_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIIIup_gseaKEGG <- gseKEGG(geneList = DEAD_basalIIIup_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "pos")
DEAD_basalIIIup_gseaKEGG <- setReadable(DEAD_basalIIIup_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIIIdown_gseaKEGG <- gseKEGG(geneList = DEAD_basalIIIdown_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "neg")
DEAD_basalIIIdown_gseaKEGG <- setReadable(DEAD_basalIIIdown_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#basal II
DEAD_basalII_gseaKEGG <- gseKEGG(geneList = DEAD_basalII_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05)
DEAD_basalII_gseaKEGG <- setReadable(DEAD_basalII_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIIup_gseaKEGG <- gseKEGG(geneList = DEAD_basalIIup_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "pos")
DEAD_basalIIup_gseaKEGG <- setReadable(DEAD_basalIIup_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIIdown_gseaKEGG <- gseKEGG(geneList = DEAD_basalIIdown_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "neg")
DEAD_basalIIdown_gseaKEGG <- setReadable(DEAD_basalIIdown_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#basal I
DEAD_basalI_gseaKEGG <- gseKEGG(geneList = DEAD_basalI_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05)
DEAD_basalI_gseaKEGG <- setReadable(DEAD_basalI_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIup_gseaKEGG <- gseKEGG(geneList = DEAD_basalIup_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "pos")
DEAD_basalIup_gseaKEGG <- setReadable(DEAD_basalIup_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIdown_gseaKEGG <- gseKEGG(geneList = DEAD_basalIdown_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "neg")
DEAD_basalIdown_gseaKEGG <- setReadable(DEAD_basalIdown_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#basal IV
DEAD_basalIV_gseaKEGG <- gseKEGG(geneList = DEAD_basalIV_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05)
DEAD_basalIV_gseaKEGG <- setReadable(DEAD_basalIV_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIVup_gseaKEGG <- gseKEGG(geneList = DEAD_basalIVup_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "pos")
DEAD_basalIVup_gseaKEGG <- setReadable(DEAD_basalIVup_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_basalIVdown_gseaKEGG <- gseKEGG(geneList = DEAD_basalIVdown_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "neg")
DEAD_basalIVdown_gseaKEGG <- setReadable(DEAD_basalIVdown_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#spinosum
DEAD_spinosum_gseaKEGG <- gseKEGG(geneList = DEAD_spinosum_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05)
DEAD_spinosum_gseaKEGG <- setReadable(DEAD_spinosum_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_spinosumup_gseaKEGG <- gseKEGG(geneList = DEAD_spinosumup_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "pos")
DEAD_spinosumup_gseaKEGG <- setReadable(DEAD_spinosumup_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_spinosumdown_gseaKEGG <- gseKEGG(geneList = DEAD_spinosumdown_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "neg")
DEAD_spinosumdown_gseaKEGG <- setReadable(DEAD_spinosumdown_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#granulosum
DEAD_granulosum_gseaKEGG <- gseKEGG(geneList = DEAD_granulosum_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05)
DEAD_granulosum_gseaKEGG <- setReadable(DEAD_granulosum_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_granulosumup_gseaKEGG <- gseKEGG(geneList = DEAD_granulosumup_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "pos")
DEAD_granulosumup_gseaKEGG <- setReadable(DEAD_granulosumup_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

DEAD_granulosumdown_gseaKEGG <- gseKEGG(geneList = DEAD_granulosumdown_gsea, organism = 'hsa', keyType = "kegg", pvalueCutoff = 0.05, scoreType = "neg")
DEAD_granulosumdown_gseaKEGG <- setReadable(DEAD_granulosumdown_gseaKEGG, OrgDb = org.Hs.eg.db, keyType = "ENTREZID")

#save
list <- list("GSEA KEGG AD basal III " = DEAD_basalIII_gseaKEGG,
             "GSEA KEGG AD basal IIIup" = DEAD_basalIIIup_gseaKEGG,
             "GSEA KEGG AD basal IIIdown" = DEAD_basalIIIdown_gseaKEGG,
             "GSEA KEGG AD basal II" = DEAD_basalII_gseaKEGG,
             "GSEA KEGG AD basal IIup" = DEAD_basalIIup_gseaKEGG,
             "GSEA KEGG AD basal IIdown" = DEAD_basalIIdown_gseaKEGG,
             "GSEA KEGG AD basal I" = DEAD_basalI_gseaKEGG,
             "GSEA KEGG AD basal Iup" = DEAD_basalIup_gseaKEGG,
             "GSEA KEGG AD basal Idown" = DEAD_basalIdown_gseaKEGG,
             "GSEA KEGG AD basal IV" = DEAD_basalIV_gseaKEGG,
             "GSEA KEGG AD basal IVup" = DEAD_basalIVup_gseaKEGG,
             "GSEA KEGG AD basal IVdown" = DEAD_basalIVdown_gseaKEGG,
             "GSEA KEGG AD spinosum" = DEAD_spinosum_gseaKEGG,
             "GSEA KEGG AD spinosum up" = DEAD_spinosumup_gseaKEGG,
             "GSEA KEGG AD spinosum down" = DEAD_spinosumdown_gseaKEGG,
             "GSEA KEGG AD granulosum" = DEAD_granulosum_gseaKEGG,
             "GSEA KEGG AD granulosum up" = DEAD_granulosumup_gseaKEGG,
             "GSEA KEGG AD granulosum down" = DEAD_granulosumdown_gseaKEGG)
write.xlsx(list, file = "KEGG GSEA logFC 0.1 AD.xlsx")
```

##Progeny

###Classical progeny pipeline

```{r progeny classic pipeline}
objectrds <- roj_rds[, roj_rds$disease == "HC"]
# objectrds <- objectrds[, objectrds$svm_added_anno == "Basal III KRT15+" | objectrds$svm_added_anno == "Basal III ASS1+"]

Idents(objectrds) <- "svm_annotation"

org <- c("Human")
features <- 2000
diffOrder <- c("basal III ", "basal II", "basal I", "basal IV", "stratum spinosum", "stratum granulosum")
# pathway_order <- c("Estrogen", "MAPK", "VEGF", "EGFR", "PI3K", "JAK-STAT", "WNT", 'Androgen', 'TGFb', 'NFkB', 'TNFa', "p53", "Hypoxia", "Trail")

#data frame assigning identities to each cell
CellsClusters <- data.frame(
    Cell = names(Idents(objectrds)),
    CellType = as.character(Idents(objectrds)),
    stringsAsFactors = FALSE
  )

#calculation progeny, added as inactive assay to rds
objectrds <- progeny(objectrds, scale = F, organism = org, top = features, perm = 1, return_assay = TRUE)
  
# objectrds@assays$progeny

#We can now directly apply Seurat functions in our Progeny scores, like scaling the pathway activity scores
objectrds <- Seurat::ScaleData(objectrds, assay = "progeny")

#We transform Progeny scores into a data frame to better handling the results
progeny_scores_df <- as.data.frame(t(GetAssayData(objectrds, assay = "progeny", slot = "scale.data"))) %>%
  tibble::rownames_to_column("Cell") %>%
  tidyr::pivot_longer(names_to = "Pathway", values_to = "Activity", c(-"Cell")) #transforms the table into a longer table format - assigning a separate row for every pathways + coordinating activity per cell 
  
#We match Progeny scores with the cell clusters.
progeny_scores_df <- inner_join(progeny_scores_df, CellsClusters, by = c("Cell"))
  
#We summarize the Progeny scores by cell population
summarized_progeny_scores <- progeny_scores_df %>%
  group_by(Pathway, CellType) %>%
  summarise(avg = mean(Activity), std = sd(Activity))

#We prepare the data for the plot
summarized_progeny_scores_df <- summarized_progeny_scores %>%
  dplyr::select(-std) %>%   
  tidyr::spread(Pathway, avg) %>%
  data.frame(row.names = "CellType", check.names = FALSE, stringsAsFactors = FALSE) 

summarized_progeny_scores_df <- summarized_progeny_scores_df[order(diffOrder), ]
# summarized_progeny_scores_df_HC <- summarized_progeny_scores_df_HC[ , order(pathway_order)]

#we prepare the plot
paletteLength = 100
myColor = colorRampPalette(c("darkgreen", "white", "darkorange"))(paletteLength)

# to get the scale balanced and centered around 0
progenyBreaks <-  c(
  seq(min(summarized_progeny_scores_df), 0, length.out = ceiling(paletteLength / 2) + 1),
  seq(max(summarized_progeny_scores_df) / paletteLength, max(summarized_progeny_scores_df),  length.out = floor(paletteLength / 2)))
# seq creates a sequence lowest to highest number with length.out determining length of sequence

#we plot
progeny_hmap <- pheatmap(t(summarized_progeny_scores_df),
                         color = myColor, breaks = progenyBreaks,
                         cluster_cols = F, treeheight_col = 0, cluster_rows = T, treeheight_row = 0, 
                         border_color = NA, 
                         main =  paste0("PROGENy (", features , " features)"), fontsize = 14, fontsize_row = 10, angle_col = 45)
```

###Progeny comparison AD and HC separately analyzed

```{r progeny HC}
objectrds <- roj_rds[, roj_rds$disease == "HC"]
# objectrds <- objectrds[, objectrds$svm_added_anno == "Basal III KRT15+" | objectrds$svm_added_anno == "Basal III ASS1+"]

Idents(objectrds) <- "svm_annotation"

org <- c("Human")
features <- 1000
diffOrder <- c("basal III", "basal II", "basal I", "basal IV", "stratum spinosum", "stratum granulosum")
pathway_order <- c("Estrogen", "MAPK", "VEGF", "EGFR", "PI3K", "Trail", "p53", "Androgen", "JAK-STAT", "WNT", "TNFa", "NFkB", "TGFb", "Hypoxia")

#data frame assigning identities to each cell
CellsClusters <- data.frame(
    Cell = names(Idents(objectrds)),
    CellType = as.character(Idents(objectrds)),
    stringsAsFactors = FALSE
  )

#calculation progeny, added as inactive assay to rds
objectrds <- progeny(objectrds, scale = F, organism = org, top = features, perm = 1, return_assay = TRUE)

# objectrds@assays$progeny

#We can now directly apply Seurat functions in our Progeny scores, like scaling the pathway activity scores
objectrds <- Seurat::ScaleData(objectrds, assay = "progeny")

#We transform Progeny scores into a data frame to better handling the results
progeny_scores_df <- as.data.frame(t(GetAssayData(objectrds, assay = "progeny", slot = "scale.data"))) %>%
  tibble::rownames_to_column("Cell") %>%
  tidyr::pivot_longer(names_to = "Pathway", values_to = "Activity", c(-"Cell")) #transforms the table into a longer table format - assigning a separate row for every pathways + coordinating activity per cell

#We match Progeny scores with the cell clusters.
progeny_scores_df <- inner_join(progeny_scores_df, CellsClusters, by = c("Cell"))

#We summarize the Progeny scores by cell population
summarized_progeny_scores <- progeny_scores_df %>%
  group_by(Pathway, CellType) %>%
  summarise(avg = mean(Activity), std = sd(Activity))

#We prepare the data for the plot
summarized_progeny_scores_df_HC <- summarized_progeny_scores %>%
  dplyr::select(-std) %>%
  tidyr::spread(Pathway, avg) %>%
  data.frame(row.names = "CellType", check.names = FALSE, stringsAsFactors = FALSE)

summarized_progeny_scores_df_HC <- summarized_progeny_scores_df_HC[diffOrder, ]
summarized_progeny_scores_df_HC <- summarized_progeny_scores_df_HC[, pathway_order]

#we prepare the plot
paletteLength = 100
myColor = colorRampPalette(c("darkgreen", "white", "darkorange"))(paletteLength)

# to get the scale balanced and centered around 0
progenyBreaks <-  c(
  seq(min(summarized_progeny_scores_df_HC), 0, length.out = ceiling(paletteLength / 2) + 1),
  seq(max(summarized_progeny_scores_df_HC) / paletteLength, max(summarized_progeny_scores_df_HC),  length.out = floor(paletteLength / 2)))
# seq creates a sequence lowest to highest number with length.out determining length of sequence

#we plot
progeny_hmap <- pheatmap(t(summarized_progeny_scores_df_HC), 
                        cluster_cols = F, #treeheight_col = 0, 
                        cluster_rows = F, #treeheight_row = 0, 
                        color = myColor, breaks = progenyBreaks, border_color = NA,
                        main =  paste0("PROGENy pathway activity HC"), angle_col = 45) #fontsize = 14, fontsize_row = 10
```

```{r progeny AD}
objectrds <- roj_rds[, roj_rds$disease == "AD"]

Idents(objectrds) <- "svm_annotation"

org <- c("Human")
features <- 1000
# diffOrder <- c("basal III ", "basal II", "basal I", "basal IV", "stratum spinosum", "stratum granulosum")
# pathway_order <- c("Estrogen", "MAPK", "VEGF", "EGFR", "PI3K", "Trail", "p53", "Androgen", "JAK-STAT", "WNT", "TNFa", "NFkB", "TGFb", "Hypoxia")

#data frame assigning identities to each cell
CellsClusters <- data.frame(
    Cell = names(Idents(objectrds)),
    CellType = as.character(Idents(objectrds)),
    stringsAsFactors = FALSE
  )

#calculation progeny, added as inactive assay to rds
objectrds <- progeny(objectrds, scale = F, organism = org, top = features, perm = 1, return_assay = TRUE)

# objectrds@assays$progeny

#We can now directly apply Seurat functions in our Progeny scores, like scaling the pathway activity scores
objectrds <- Seurat::ScaleData(objectrds, assay = "progeny")

#We transform Progeny scores into a data frame to better handling the results
progeny_scores_df <- as.data.frame(t(GetAssayData(objectrds, assay = "progeny", slot = "scale.data"))) %>%
  tibble::rownames_to_column("Cell") %>%
  tidyr::pivot_longer(names_to = "Pathway", values_to = "Activity", c(-"Cell")) #transforms the table into a longer table format - assigning a separate row for every pathways + coordinating activity per cell

#We match Progeny scores with the cell clusters.
progeny_scores_df <- inner_join(progeny_scores_df, CellsClusters, by = c("Cell"))

#We summarize the Progeny scores by cell population
summarized_progeny_scores <- progeny_scores_df %>%
  group_by(Pathway, CellType) %>%
  summarise(avg = mean(Activity), std = sd(Activity))

#We prepare the data for the plot
summarized_progeny_scores_df_AD <- summarized_progeny_scores %>%
  dplyr::select(-std) %>%
  tidyr::spread(Pathway, avg) %>%
  data.frame(row.names = "CellType", check.names = FALSE, stringsAsFactors = FALSE)

summarized_progeny_scores_df_AD <- summarized_progeny_scores_df_AD[diffOrder, ]
summarized_progeny_scores_df_AD <- summarized_progeny_scores_df_AD[ , pathway_order]

paletteLength = 100
myColor = colorRampPalette(c("darkgreen", "white", "darkorange"))(paletteLength)

#to get the scale balanced and centered around 0
progenyBreaks = c(
 seq(min(summarized_progeny_scores_df_AD), 0, length.out = ceiling(paletteLength / 2) + 1),
 seq(max(summarized_progeny_scores_df_AD) / paletteLength, max(summarized_progeny_scores_df_AD),  length.out = floor(paletteLength / 2)))
# seq creates a sequence lowest to highest number with length.out determining length of sequence

progeny_hmap <- pheatmap(t(summarized_progeny_scores_df_AD), 
                        cluster_cols = F, #treeheight_col = 0, 
                        cluster_rows = F, #treeheight_row = 0, 
                        color = myColor, breaks = progenyBreaks, border_color = NA,
                        main =  paste0("PROGENy pathway activity AD"), angle_col = 45) #fontsize = 14, fontsize_row = 10
```

This is not a good way to do it because it masks systemic differences between AD and HC because both analysis are centered within themselves. 

```{r progeny comparison}
# #we prepare the plot
# summarized_progeny_scores_df_comp = summarized_progeny_scores_df_AD - summarized_progeny_scores_df_HC #relative activity between two variables
# summarized_progeny_scores_df_comp <- summarized_progeny_scores_df_comp[order(diffOrder), ]
# 
# paletteLength = 100
# myColor = colorRampPalette(c("darkgreen", "white", "darkorange"))(paletteLength)
# 
# #to get the scale balanced and centered around 0
# progenyBreaks = c(
#  seq(min(summarized_progeny_scores_df_comp), 0, length.out = ceiling(paletteLength / 2) + 1),
#  seq(max(summarized_progeny_scores_df_comp) / paletteLength, max(summarized_progeny_scores_df_comp),  length.out = floor(paletteLength / 2)))
# # seq creates a sequence lowest to highest number with length.out determining length of sequence
# 
# progeny_hmap <- pheatmap(t(summarized_progeny_scores_df_comp),fontsize = 14,
#                         fontsize_row = 10, cluster_cols = FALSE,
#                         color = myColor, breaks = progenyBreaks,
#                         main =  paste0("PROGENy pathway activity AD relative to Healthy"), angle_col = 45,
#                         treeheight_col = 0, treeheight_row = 0, border_color = NA, Colv = NA)
```

```{r progeny Jos}
seur_obj  <- roj_rds

net <- decoupleR::get_progeny(organism = 'human', top = 100)
  # Extract the normalized log-transformed counts
  mat <- as.matrix(seur_obj@assays$RNA@data)
  
  # Run wmean
  acts <- run_wmean(mat = mat, net = net, .source='source', .target='target',
                    .mor='weight', times = 100, minsize = 5)
  
  # Extract norm_wmean and store it in pathways_wmean in data
  seur_obj[['pathwayswmean']] <- acts %>%
    filter(statistic == 'norm_wmean') %>%
    pivot_wider(id_cols = 'source', names_from = 'condition',
                values_from = 'score') %>%
    column_to_rownames('source') %>%
    Seurat::CreateAssayObject(.)
  
  # Change assay
  DefaultAssay(object = seur_obj) <- "pathwayswmean"
  
  # Scale the data
  seur_obj <- ScaleData(seur_obj)
  seur_obj@assays$pathwayswmean@data <- seur_obj@assays$pathwayswmean@scale.data
  # saveRDS(seur_obj, file = rds_file)

# Extract activities from object as a long dataframe
df <- t(as.matrix(seur_obj@assays$pathwayswmean@data)) %>%
  as.data.frame() %>%
  mutate(cluster = Idents(seur_obj)) %>%
  pivot_longer(cols = -cluster, names_to = "source", values_to = "score") %>%
  group_by(cluster, source) %>%
  summarise(mean = mean(score))

# Transform to wide matrix
top_acts_mat <- df %>%
  pivot_wider(id_cols = 'cluster', names_from = 'source',
              values_from = 'mean') %>%
  column_to_rownames('cluster') %>%
  as.matrix()

# Choose color pallette
palette_length = 100
my_color = colorRampPalette(c("Darkblue", "white","red"))(palette_length)

my_breaks <- c(seq(-2, 0, length.out=ceiling(palette_length/2) + 1),
               seq(0.05, 2, length.out=floor(palette_length/2)))

# Plot
# pdf(paste(resultsdir, "10b_progeny_enrichment.pdf", sep = '/'), width = 5, height = 5)
DimPlot(seur_obj, reduction = "umap", label = TRUE, pt.size = 0.5)
pheatmap(top_acts_mat, border_color = NA, color=my_color, breaks = my_breaks) 

for (pathway in rownames(seur_obj@assays$pathwayswmean)){
  print(FeaturePlot(seur_obj, features = c(pathway)) & 
  scale_colour_gradient2(low = 'blue', mid = 'white', high = 'red')) +
  ggtitle(paste(pathway,'activity', sep = ' '))
}

# dev.off()
#output progeny table file
output_df <- df %>%
  pivot_wider(id_cols = 'cluster', names_from = 'source',
              values_from = 'mean') %>%
  column_to_rownames('cluster') %>%  t() %>% as.data.frame()

output_df[['pathway']] <- rownames(output_df)
#get the per catagory stdev
# Get top tfs with more variable means across clusters
std_df <- df %>%
  group_by(source) %>%
  summarise(std = sd(mean)) %>%
  arrange(-abs(std)) 
colnames(std_df) <- c('pathway', 'std')

output_df <- merge(output_df,std_df, by = 'pathway')
write.table(output_df, paste(resultsdir, "10b_progeny_enrichment.csv", sep = '/'), sep = ',')
```

###Progeny comparison AD and HC analyzed together

Comparing this way is much better, i talked with Sybren and here the pathway activities are normalized and centered together. Since the scaling seems to be linear, one can substract them from each other. 

```{r progeny together}
pathway_features <- 2000
colorder <- c("HC_basal III", "HC_basal II", "HC_basal I", "HC_basal IV", "HC_stratum spinosum", "HC_stratum granulosum", "AD_basal III", "AD_basal II", "AD_basal I", "AD_basal IV", "AD_stratum spinosum", "AD_stratum granulosum")
pathway_order <- c("JAK-STAT", "TNFa", "NFkB", "Estrogen", "MAPK", "VEGF", "EGFR", "PI3K", "Androgen", "p53", "TGFb", "Trail", "Hypoxia", "WNT")

objectrds <- roj_rds
Idents(objectrds) <-  paste0(objectrds$disease,"_",objectrds$svm_annotation)
  
  require(progeny)
  require(pheatmap)
  require(tidyr)
  require(tibble)

  #data frame assigning identities to each cell
  CellsClusters <- data.frame(
    Cell = names(Idents(objectrds)),
    CellType = as.character(Idents(objectrds)),
    stringsAsFactors = FALSE
  )

  #calculation progeny, added as inactive assay to rds
  objectrds <- progeny(objectrds, scale = F, organism = "Human", top = pathway_features, perm = 1, return_assay = TRUE, verbose = T)

  #We can now directly apply Seurat functions, like scaling our Progeny scores
  objectrds <- Seurat::ScaleData(objectrds, assay = "progeny")
  
  #We transform Progeny scores into a data frame to better handling the results
  progeny_scores_df <- as.data.frame(t(GetAssayData(
    objectrds,
    slot = "scale.data",
    assay = "progeny"
  ))) %>%
    rownames_to_column("Cell") %>%
    gather(Pathway, Activity,-Cell) #transforms the table into a longer table format - assigning a separate row for every pathways + coordinating activity per cell 
  
  #We match Progeny scores with the cell clusters.
  progeny_scores_df <- inner_join(progeny_scores_df, CellsClusters)
  
  #We summarize the Progeny scores by cell population
  summarized_progeny_scores <- progeny_scores_df %>%
    group_by(Pathway, CellType) %>%
    summarise(avg = mean(Activity), std = sd(Activity))
 
  #We prepare the data for the plot
  summarized_progeny_scores_df <- summarized_progeny_scores %>%
    dplyr::select(-std) %>%   
    spread(Pathway, avg) %>%
    data.frame(row.names = "CellType", check.names = FALSE, stringsAsFactors = FALSE)
  
  summarized_progeny_scores_df <- summarized_progeny_scores_df[colorder, ]
  summarized_progeny_scores_df <- summarized_progeny_scores_df[, pathway_order, ]
  
  # #we prepare the plot
  # nr = nrow(summarized_progeny_scores_df) / 2
  # df1 = summarized_progeny_scores_df[1:nr, ]
  # df2 = summarized_progeny_scores_df[(nr + 1):c(2 * nr), ]
  # summarized_progeny_scores_df = df1 - df2 #relative activity between two variables
  
  paletteLength = 100
  myColor = colorRampPalette(c("darkgreen", "white", "darkorange"))(paletteLength) #potentially darkblue, indianred
  
  #to get the scale balanced and centered around 0
  progenyBreaks = c(
    seq(min(summarized_progeny_scores_df), 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(summarized_progeny_scores_df) / paletteLength, max(summarized_progeny_scores_df),  length.out = floor(paletteLength / 2)))
  # seq creates a sequence lowest to highest number with length.out determining length of sequence

  #we plot
  progeny_hmap <- pheatmap(t(summarized_progeny_scores_df),
                           fontsize = 12, fontsize_row = 9, 
                           cluster_rows = F, cluster_cols = F, angle_col = 45, treeheight_col = 0,  border_color = NA, 
                           color = myColor, breaks = progenyBreaks, 
                           main =  paste0("PROGENy pathway activity"))
```

```{r progeny function comparison}
progenyfv_2 = function(objectrds, features = 2000, org = "Human", colorder, clustering = F, pathway_order = c("Trail", "Hypoxia", "WNT", "p53", "Estrogen", "MAPK", "EGFR", "PI3K", "Androgen", "TGFb", "VEGF", "NFkB", "TNFa", "JAK-STAT")){ 
  
  # objectrds <- roj_rds
  # features = 2000
  # org = "Human"
  # colorder = c("AD_basal III", "AD_basal II", "AD_basal I", "AD_basal IV", "AD_stratum spinosum", "AD_stratum granulosum", "HC_basal III", "HC_basal II", "HC_basal I", "HC_basal IV", "HC_stratum spinosum", "HC_stratum granulosum")
  # clustering = T
  # pathway_order <- c("Trail", "Hypoxia", "WNT", "p53", "Estrogen", "MAPK", "EGFR", "PI3K", "Androgen", "TGFb", "VEGF", "NFkB", "TNFa", "JAK-STAT")
  
  Idents(objectrds) <-  paste0(objectrds$disease,"_",objectrds$svm_annotation)
  colorder <- x
  
  require(progeny)
  require(pheatmap)
  require(tidyr)
  require(tibble)

  #data frame assigning identities to each cell
  CellsClusters <- data.frame(
    Cell = names(Idents(objectrds)),
    CellType = as.character(Idents(objectrds)),
    stringsAsFactors = FALSE
  )

  #calculation progeny, added as inactive assay to rds
  objectrds <- progeny(objectrds, scale = F, organism = org, top = features, perm = 1, return_assay = TRUE)

  #We can now directly apply Seurat functions, like scaling our Progeny scores
  objectrds <- Seurat::ScaleData(objectrds, assay = "progeny")
  
  #We transform Progeny scores into a data frame to better handling the results
  progeny_scores_df <- as.data.frame(t(GetAssayData(
    objectrds,
    slot = "scale.data",
    assay = "progeny"
  ))) %>%
    rownames_to_column("Cell") %>%
    gather(Pathway, Activity,-Cell) #transforms the table into a longer table format - assigning a separate row for every pathways + coordinating activity per cell 
  
  #We match Progeny scores with the cell clusters.
  progeny_scores_df <- inner_join(progeny_scores_df, CellsClusters)
  
  #We summarize the Progeny scores by cell population
  summarized_progeny_scores <- progeny_scores_df %>%
    group_by(Pathway, CellType) %>%
    summarise(avg = mean(Activity), std = sd(Activity))
 
  #We prepare the data for the plot
  summarized_progeny_scores_df <- summarized_progeny_scores %>%
    dplyr::select(-std) %>%   
    spread(Pathway, avg) %>%
    data.frame(row.names = "CellType", check.names = FALSE, stringsAsFactors = FALSE)
  
  summarized_progeny_scores_df <- summarized_progeny_scores_df[order(colorder), ]
  
  #we prepare the plot
  nr = nrow(summarized_progeny_scores_df) / 2
  df1 = summarized_progeny_scores_df[1:nr, ]
  df2 = summarized_progeny_scores_df[(nr + 1):c(2 * nr), ]
  summarized_progeny_scores_df = df1 - df2 #relative activity between two variables
  summarized_progeny_scores_df <- summarized_progeny_scores_df[ , pathway_order]
  
  paletteLength = 100
  myColor = colorRampPalette(c("darkgreen", "white", "darkorange"))(paletteLength) #potentially darkblue, indianred
  
  #to get the scale balanced and centered around 0
  progenyBreaks = c(
    seq(min(summarized_progeny_scores_df), 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(summarized_progeny_scores_df) / paletteLength, max(summarized_progeny_scores_df),  length.out = floor(paletteLength / 2)))
  # seq creates a sequence lowest to highest number with length.out determining length of sequence

  #we plot
  progeny_hmap <- pheatmap(t(summarized_progeny_scores_df),
                           fontsize = 12, fontsize_row = 9, 
                           cluster_rows = clustering, cluster_cols = F, angle_col = 45, treeheight_col = 0,  border_color = NA, 
                           color = myColor, breaks = progenyBreaks, 
                           main =  paste0("PROGENy pathway activity relative to Healthy"))
  return(progeny_hmap)
}

x <- c("AD_basal III", "AD_basal II", "AD_basal I", "AD_basal IV", "AD_stratum spinosum", "AD_stratum granulosum", "HC_basal III", "HC_basal II", "HC_basal I", "HC_basal IV", "HC_stratum spinosum", "HC_stratum granulosum")
#it is important that AD all together and before HC otherwise relation is not correct / inverse
#order within AD or healthy can be as you like
y <- c("Trail", "Hypoxia", "WNT", "p53", "Estrogen", "MAPK", "EGFR", "PI3K", "Androgen", "TGFb", "VEGF", "NFkB", "TNFa", "JAK-STAT")

Idents(roj_rds) = paste0(roj_rds$disease,"_",roj_rds$svm_annotation)
progenyfv_2(roj_rds, features = 2000, org = "Human", colorder = x, clustering = F) #result only interpretable if equal amount of AD vs HC conditions!
```



###Extract pathway contributing genes

Progeny has a weighted data set of genes contributing to each pathway. This we overlap with genes changing in basal III between AD and HC in our data set (also works for any other comparison). 

First we calculate the t statistic of the genes expression changes. Here we calculate it for the gene expression changes of basal III cells between AD and HC. It can also be done for other comparisons though. We take the whole list of genes into account (standard for progeny). 

```{r calculcate t statistic for each gene}
library(limma)
Idents(roj_rds) <- roj_rds$svm_annotation
basalIII <- subset(roj_rds, idents = "basal III")
counts <- as.matrix(GetAssayData(basalIII))

# Filter out genes that are 0 for every cell in this cluster
notexpr <- which(rowSums(counts) == 0)
expressed <- counts[-notexpr,]

mm <- model.matrix(~ 0 + disease, data = basalIII@meta.data)
fit <- lmFit(expressed, mm)  
head(coef(fit))# means in each sample for each gene
# test <- coef(fit)
# test[rownames(test) == "KRT6A", ] #test to see

contr <- makeContrasts(diseaseAD - diseaseHC, levels = colnames(coef(fit)))
tmp <- contrasts.fit(fit, contrasts = contr)
tmp <- eBayes(tmp)
# topTable(tmp, sort.by = "P", n = 20) # top 20 DE genes

basalIII_tstat <- as.data.frame(topTable(tmp, n = 20000)) #since there are around 18000 genes, top 2000 includes all genes
basalIII_tstat$gene <- rownames(basalIII_tstat)
basalIII_tstat <- basalIII_tstat[ ,c("gene", "t")] # traditional t statistic "t"
# rownames(basalIII_tstat) <- NULL
```

Theoretically, we could limit the list of genes we're comparing with the progeny pathways to only all significantly DE genes I also tried using fold changes of significantly DE expressed genes between AD and HC for plotting but then I end up with only 800 genes while the t statistic include much more genes. 

```{r using FC values for the scatterplot}
# basalIII_topgenes_FC <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_III_p_val_adj), ]
# basalIII_topgenes_FC$AD_basal_III_avg_FC <- 2^basalIII_topgenes_FC$AD_basal_III_avg_log2FC
# basalIII_topgenes_FC <- basalIII_topgenes_FC [ , c("gene", "AD_basal_III_avg_FC")]
```

Then we extract the genes contributing to each pathway. Here we focus on the top 2000 pathway genes since we also used those to calculate the progeny scores before. 

```{r pathway genes}
progeny_weight <- get_progeny(organism = 'human', top = 2000) #alternative way obtaining progeny pathway contributing genes
```

These scores are then overlapped with the t statistics from gene changes between basal III AD compared to HC. 

x-axis: contribution of the gene to the pathway (positive / negative contribution) - determined by progeny
y axis: statistic significance of gene in your comparison - depends on gene expression in dataset

```{r extract contributing pathway genes - modified progenyScatter function}
pathway <- 'WNT'

df <- progeny_weight %>%
  filter(source == pathway) %>%
  mutate(ID = target, color = "3", quadrant = "0") %>%
  dplyr::select(-p_value) %>% 
  tibble::column_to_rownames('target')

inter <- sort(intersect(basalIII_tstat$gene,rownames(df)))
df <- df[inter, ]
df['logFC'] <- basalIII_tstat[inter, "logFC"]
df['t_value'] <- basalIII_tstat[inter, "t"]
df['p_adj'] <- basalIII_tstat[inter, "adj.P.Val"]
df <- df[ , c(1, 3, 2, 6, 7, 8, 5, 4)]
df <- df %>%
  mutate(color = if_else(weight > 0 & t_value > 0, '1', color)) %>%
  mutate(color = if_else(weight > 0 & t_value < 0, '2', color)) %>%
  mutate(color = if_else(weight < 0 & t_value > 0, '2', color)) %>%
  mutate(color = if_else(weight < 0 & t_value < 0, '1', color)) #color gets assignment dependent on weight and t value
df <- df %>%
  mutate(quadrant = if_else(weight > 0 & t_value > 0, '1', quadrant)) %>%
  mutate(quadrant = if_else(weight < 0 & t_value > 0, '2', quadrant)) %>%
  mutate(quadrant = if_else(weight < 0 & t_value < 0, '3', quadrant)) %>% 
  mutate(quadrant = if_else(weight > 0 & t_value < 0, '4', quadrant))#quadrant gets assignment dependent on weight and t value

# options(ggrepel.max.overlaps= Inf) # Include all gene labels
 
#normally you plot t value, I opted to put log
scatter <- ggplot(df, aes(x = weight, y = logFC, color = color)) + geom_point() +
  scale_colour_manual(values = c("darkorange","darkgreen","white")) +
  # ggrepel::geom_label_repel(aes(label = ID)) +
  theme_minimal() +
  theme(legend.position = "none") +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  ggtitle(pathway)

plot(scatter)
# hist_x <- ggplot(df, aes(x = weight, fill = "blue")) + geom_density() + 
#                 scale_fill_manual(values = c("#00c5ff")) + 
#                 theme_minimal() + 
#                 theme(legend.position = "none", axis.text.x = element_blank(),
#                 axis.ticks.x = element_blank(), axis.title.y = element_blank(),
#                 axis.text.y = element_blank(), axis.ticks.y = element_blank(),
#                 panel.grid.major = element_blank(),
#                 panel.grid.minor = element_blank()) 
# 
# hist_y <- ggplot(df, aes(x = logFC, fill = "blue")) + geom_density() + coord_flip() + 
#                 scale_fill_manual(values = c("#00c5ff")) + 
#                 theme_minimal() + 
#                 theme(legend.position = "none", axis.text.x = element_blank(),
#                 axis.ticks.x = element_blank(), axis.title.y = element_blank(),
#                 axis.text.y = element_blank(), axis.ticks.y = element_blank(),
#                 panel.grid.major = element_blank(),
#                 panel.grid.minor = element_blank())
# 
# ggpubr::ggarrange(hist_x, NULL, scatter, hist_y, ncol = 2, nrow = 2,  align = "hv", widths = c(2, 1), heights = c(1, 2))

table(df$quadrant)
```

Alternative: 

```{r extract pathway contributing genes - progenyScatter original function}
# progeny_weight <- getModel("Human", top = 2000) %>%
#   as.data.frame() %>%
#   tibble::rownames_to_column("GeneID")
# 
# # options(ggrepel.max.overlaps= Inf) # Include all gene labels
# 
# table(progeny_weight$GeneID %in% basalIII_tstat$gene)
# progeny_weight$GeneID[!(progeny_weight$GeneID %in% basalIII_tstat$gene == T)] #not all progeny genes are expressed in my dataset
# 
# scat_plots <- progeny::progenyScatter(df = basalIII_tstat, dfID = 1, weight_matrix = progeny_weight, weightID = 1, statName = "logFC", verbose = F) #statName is purely the textual axis label
# plot(scat_plots[[1]]$'WNT')
# 
# pathway <- progeny_weight$GeneID[!(progeny_weight$NFkB == 0)]
# pathway_genes <- basalIII_tstat[basalIII_tstat$gene %in% pathway, ]
```

Let's check if the downstream Wnt pathway genes that are changing in AD basal III are TP63 targets.
That is not really logical though because TP63 ChIP shows only direct TP63 targets and not downstream affected by Wnt.

```{r direct TP63 in Wnt basal III AD changes?}
#load targets p63 ChIP-seq
p63targetsall <- readxl::read_excel("embr0016-0863-sd13.xlsx", sheet = "ST12A") #open sup 13, table12 of Kouwenhoven EN, Oti M, Niehues H, van Heeringen SJ, Schalkwijk J, Stunnenberg HG, van Bokhoven H, Zhou H. Transcription factor p63 bookmarks and regulates dynamic enhancers during epidermal differentiation. EMBO Rep. 2015 Jul;16(7):863-78. doi: 10.15252/embr.201439941. Epub 2015 Jun 1. PMID: 26034101; PMCID: PMC4515125.
p63targetsall <- p63targetsall$...10 #select gene list
p63targetsall <- unique(p63targetsall[3:41598]) #Trim header to remove non gene values

#background
counts_sum <- rowSums(roj_rds@assays[["RNA"]])
background <- rownames(roj_rds@assays[["RNA"]][counts_sum > 1])

p63targetsexpr <- p63targetsall[p63targetsall %in% background]

#let's see if TP63 target genes are enriched in all downregulated WNT targets

#compared to all basal III AD WNT targets
# table(df$ID[df$quadrant == 2 | df$quadrant == 4] %in% p63targetsexpr)
ggvenn::ggvenn(list('Expressed TP63 ChIP-seq targets' = p63targetsexpr, 'WNT downregulated in AD basal III' = df$ID[df$quadrant == 2 | df$quadrant == 4]))
ggvenn::ggvenn(list('Expressed TP63 ChIP-seq targets' = p63targetsexpr, 'all WNT targets in AD basal III' = df$ID))

#compared to all epressed genes
ggvenn::ggvenn(list('TP63 ChIP-seq targets' = p63targetsall, 'WNT downregulated in AD basal III' = df$ID[df$quadrant == 2 | df$quadrant == 4]))
ggvenn::ggvenn(list('TP63 ChIP-seq targets' = p63targetsall, 'all expressed genes' = background))

#that is not very clear - let's check if TP63 target genes are enriched in all DE downregulated WNT targets
WNTdownsign <- df[df$quadrant == 2 | df$quadrant == 4, ]
WNTdownsign <- WNTdownsign[WNTdownsign$p_adj < 0.05, ]
WNTdownsign$ID

ggvenn::ggvenn(list('Expressed TP63 ChIP-seq targets' = p63targetsexpr, 'WNT DE downregulated in AD basal III' = WNTdownsign$ID))
ggvenn::ggvenn(list('Expressed TP63 ChIP-seq targets' = p63targetsexpr, 'all DE WNT targets in AD basal III' = df$ID[df$p_adj < 0.05]))
ggvenn::ggvenn(list('Expressed TP63 ChIP-seq targets' = p63targetsexpr, 'all WNT targets in AD basal III' = df$ID))
ggvenn::ggvenn(list('TP63 ChIP-seq targets' = p63targetsall, 'all expressed genes' = background))
```

BUT: I can check the EEC patient affected genes. 

```{r downstream TP63 targets in Wnt basal III AD changes?}
p63LOF <- readxl::read_excel("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/Shen - TP63 KO patients can be rescued by PRIMA (2013).xls") #a list of genes with an expression change >1.5 in a microarray assay of a monolayer cell culture of 3 TP63 LOF mutated patient lines. Flaws: 2D vs 3D vs single-cell!, 1.5 FC very high, microarray also captures a limited number of genes
p63LOFexpr <- p63LOF[p63LOF$gene %in% background, ]

#let's see if TP63 downstream target genes are enriched in all downregulated WNT targets

#compared to all basal III AD WNT targets
# table(df$ID[df$quadrant == 2 | df$quadrant == 4] %in% p63targetsexpr)
ggvenn::ggvenn(list('Expressed TP63 ChIP-seq targets' = p63targetsexpr, 'WNT downregulated in AD basal III' = df$ID[df$quadrant == 2 | df$quadrant == 4]))
ggvenn::ggvenn(list('Expressed TP63 ChIP-seq targets' = p63targetsexpr, 'all WNT targets in AD basal III' = df$ID))
```



```{r heatmap of enriched genes}
Idents(roj_rds) <- roj_rds$svm_annotation

#extract scaled pseudobulk expression data
roj_rds <- ScaleData(roj_rds)
x <- GetAssayData(roj_rds, slot = 'scale.data')
pseudobulk_scaled <- AverageExpression(roj_rds, assays = "RNA", slot = 'scale.data', group.by = c('disease', 'svm_annotation'))
pseudobulk_scaled <- as.data.frame(pseudobulk_scaled$RNA)

wnt_enriched_genes <- pseudobulk_scaled[rownames(pseudobulk_scaled) %in% c("DUSP14","RAP2B","AHR","OVOL1","COX5A","SHB"),]
# heatmap
pheatmap(wnt_enriched_genes,fontsize=14, show_rownames = TRUE,color=myColor,cluster_cols = FALSE, main = "Genes contributing to WNT enrichment", angle_col = 45, treeheight_col = 0, border_color = NA)
```






# Pseudotime analysis

Using slingshot on the non-integrated data provides a more accurate representation of the cell trajectory(!!) when comparing HC vs AD. However, when looking at individual genes, pseudotime performed on the integrated umap but looking at non-integrated gene expression ensures having the same trajectory of AD and HC. This is what we do when we perform trajectory interference on integrated data, the tradeseq infered gene expression analysis is using non-integrated counts again. 
Conclusion: show pseudotime timeline derived from non-integrated umap. show pseudotime heatmap of genes derived from integrated umap (gene expression will remain from non-integrated RNA assay)

## Pseudotime on non-integrated data

```{r clustering}
#remove possible previous iterations
roj_rds$slingPseudotime_1 = NULL
roj_rds$slingPseudotime_2 = NULL

object <- roj_rds
ndims <- 10

#clustering
DefaultAssay(object) <-  "RNA"
object <- NormalizeData(object)
object <- ScaleData(object,verbose = F)
object <- FindVariableFeatures(object, selection.method = "vst", nfeatures = 2000, verbose = F) # nFeatures 2000 by default, selection method is default too 
object <- RunPCA(object, features = VariableFeatures(object = object), npcs = 50, verbose = F) # by default run on scaled variable features and 50 PCs
object <- FindNeighbors(object, dims = 1:ndims, verbose = F)
object <- FindClusters(object, resolution = 0.3 ,verbose = F)
object <- RunUMAP(object, dims = 1:ndims, verbose = F)

DimPlot(object, group.by = "sample")
```

Running Slingshot -> I get a different trajectory than Ward but he said it is completely okay and might likely be because of package differences (not sure about the last one). I might be able to replicate it if I specify clusterLabels (svm_annotation) but no starting cluster. 

```{r slingshot all together}
# object_sce <- as.SingleCellExperiment(object, assay = "RNA") #Seurat to SCE

# 
# # object_sce <- slingshot(object_sce, clusterLabels = 'svm_annotation', reducedDim = 'UMAP')
# object_sce <- slingshot(object_sce, clusterLabels = 'svm_annotation', reducedDim = 'UMAP', start.clus = "basal III", end.clus = "stratum granulosum") #strong recommendation to specify a starting cluster
#  
#Plot pseudotime on UMAP
# plot(reducedDims(object_sce)$UMAP, col = colors_subtypes[as.factor(object_sce$svm_annotation)], pch = 16,  asp = 0.5)
# lines(SlingshotDataSet(object_sce), lwd = 4)
```

```{r slingshot healthy}
if(exists("object$slingPseudotime")){
object$slingPseudotime_1 <- NULL
object$slingPseudotime_2 <- NULL
}

object_hc <- subset(object, disease == "HC")
# ndims <- 10
# object_hc <- RunUMAP(object_hc, dims = 1:ndims,verbose = F) #makes no difference 

object_hc_sce <- as.SingleCellExperiment(object_hc, assay = "RNA") #Seurat to SCE

if(exists("object_hc_sce$slingPseudotime")){
object_hc_sce$slingPseudotime_1 <- NULL
object_hc_sce$slingPseudotime_2 <- NULL
}

object_hc_sce <- slingshot(object_hc_sce, clusterLabels = 'svm_annotation', reducedDim = 'UMAP', start.clus = "basal III", end.clus = "stratum granulosum") #strong recommendation to specify a starting cluster

#Plot pseudotime on UMAP
plot(reducedDims(object_hc_sce)$UMAP, col = colors_subtypes[as.factor(object_hc_sce$svm_annotation)], pch = 16,  asp = 0.5)
lines(SlingshotDataSet(object_hc_sce), lwd = 4)

#plot pseudotime 1
dat <- object_hc_sce@colData[ ,c("svm_annotation","slingPseudotime_1","disease")]
dat <- as.data.frame(dat)
dat <- dat[!is.na(dat$slingPseudotime_1), ]

ggplot(dat, aes(x = svm_annotation, y = slingPseudotime_1, fill = svm_annotation, color = svm_annotation)) +
    geom_dotplot(binwidth = 0.3, stackdir = "centerwhole", binaxis = "y", stackratio = 0.1) + coord_flip() + theme_classic() + ggtitle("HC") + labs(x = NULL, y = NULL) +
    scale_fill_manual(values = colors_subtypes) + scale_y_continuous(breaks = c(0, 5, 10, 15, 20, 25)) +
    scale_color_manual(values = colors_subtypes) + theme(legend.position = "none") 
```

```{r slingshot AD}

if(exists("object$slingPseudotime")){
object$slingPseudotime_1 <- NULL
object$slingPseudotime_2 <- NULL
}

object_ad <- subset(object, disease == "AD")
# object_ad <- RunUMAP(object_ad, dims = 1:ndims,verbose = F)

object_ad_sce <- as.SingleCellExperiment(object_ad, assay = "RNA") #Seurat to SCE

if(exists("object_ad_sce$slingPseudotime")){
object_ad_sce$slingPseudotime_1 <- NULL
object_ad_sce$slingPseudotime_2 <- NULL
}

object_ad_sce <- slingshot(object_ad_sce, reducedDim = 'UMAP') #left out the clusterLabels to get just one lineage
# object_ad_sce <- slingshot(object_ad_sce, clusterLabels = 'svm_annotation', reducedDim = 'UMAP', start.clus = "basal III", end.clus = "stratum granulosum")

#Plot pseudotime on UMAP
plot(reducedDims(object_ad_sce)$UMAP, col = colors_subtypes[as.factor(object_ad_sce$svm_annotation)], pch = 16,  asp = 0.5)
lines(SlingshotDataSet(object_ad_sce), lwd = 4)

#plot pseudotime 1
dat <- object_ad_sce@colData[ ,c("svm_annotation","slingPseudotime_1","disease")]
dat <- as.data.frame(dat)
dat <- dat[!is.na(dat$slingPseudotime_1), ]
dat$slingPseudotime_1_mod <- max(dat$slingPseudotime_1) - dat$slingPseudotime_1

ggplot(dat, aes(x = svm_annotation, y = slingPseudotime_1_mod, fill = svm_annotation, color = svm_annotation)) +
    geom_dotplot(binwidth = 0.3, stackdir = "centerwhole", binaxis = "y", stackratio = 0.1) + coord_flip() + theme_classic() +
    scale_fill_manual(values = colors_subtypes) + scale_color_manual(values = colors_subtypes) +
    ggtitle("AD") + labs(x = NULL, y = NULL) + theme(legend.position = "none")
```

```{r plot pseudotime on UMAP}
#transfer pseudotime into seurat object
object_hc$slingPseudotime_1 <- object_hc_sce$slingPseudotime_1
object_ad$slingPseudotime_1 <- object_ad_sce$slingPseudotime_1

FeaturePlot(object_hc,features = c("slingPseudotime_1")) + ggtitle("HC_slingPseudotime") + scale_color_gradient2(low = "yellow",mid = "yellow",high ="red",na.value = "grey")
FeaturePlot(object_ad,features = c("slingPseudotime_1")) + ggtitle("AD_slingPseudotime") + scale_color_gradient2(low = "red",mid = "red", high ="yellow",na.value = "grey")
```

ASK WARD IF THAT IS OKAY, ADAPT legend -> same max, figure out how to plot lines on umap

```{r plot pseudotime on integrated UMAP}
if(exists("roj_rds$slingPseudotime")){
roj_rds$slingPseudotime_1 <- NULL
roj_rds$slingPseudotime_2 <- NULL
}

roj_rds_hc <- subset(roj_rds, disease == "HC")
roj_rds_ad <- subset(roj_rds, disease == "AD")

#transfer trajectories back to the seurat object
roj_rds_hc$slingPseudotime_1 <- object_hc_sce$slingPseudotime_1
roj_rds_ad$slingPseudotime_1 <- object_ad_sce$slingPseudotime_1
roj_rds_ad$slingPseudotime_1_mod <- max(roj_rds_ad$slingPseudotime_1) - roj_rds_ad$slingPseudotime_1

#visualize trajectories in HC and AC
FeaturePlot(roj_rds_hc,features = c("slingPseudotime_1")) + ggtitle("HC pseudotime") + scale_color_gradient2(low = "yellow",mid = "yellow",high ="red",na.value = "grey", breaks = c(0, 5, 10, 15, 20, 25)) #x1
FeaturePlot(roj_rds_ad,features = c("slingPseudotime_1_mod")) + ggtitle("AD pseudotime") + scale_color_gradient2(low = "yellow",mid = "yellow",high ="red",na.value = "grey",  breaks = c(0, 5, 10, 15, 20, 25)) #x2

# roj_hc_sce <- as.SingleCellExperiment(roj_rds_hc, assay = "RNA")
# test <- SlingshotDataSet(object_hc_sce)
# plot(reducedDims(roj_hc_sce)$UMAP, col = colors_subtypes[as.factor(roj_hc_sce$svm_annotation)], pch = 16,  asp = 0.5)
# lines(test, lwd = 4)
# 
# roj_ad_sce <- as.SingleCellExperiment(roj_rds_ad, assay = "RNA")
# plot(reducedDims(roj_ad_sce)$UMAP, col = colors_subtypes[as.factor(roj_ad_sce$svm_annotation)], pch = 16,  asp = 0.5)
# lines(SlingshotDataSet(roj_ad_sce), lwd = 4)

#print(x1+x2|x3+x4) #nice way of reordering plots when printing
```


## Pseudotime on integrated data

You have to calculate pseudotime on integrated umap but continue plotting non-integrated RNA assay when looking at gene expression

```{r calculate pseudotime}
#remove possible previous iterations
if(exists("roj_rds$slingPseudotime")){
roj_rds$slingPseudotime_1 <- NULL
roj_rds$slingPseudotime_2 <- NULL
}

#perform slingshot with the sce object
roj_rds_sce <- as.SingleCellExperiment(roj_rds, assay = "RNA") 

# roj_rds_sce <- slingshot(roj_rds_sce, clusterLabels = 'svm_annotation', reducedDim = 'UMAP', start.clus = "basal III", end.clus = "stratum granulosum") #reproduces Wards result
# roj_rds_sce <- slingshot(roj_rds_sce, clusterLabels = 'svm_annotation', reducedDim = 'UMAP') #looks also interesting
roj_rds_sce <- slingshot(roj_rds_sce, reducedDim = 'UMAP') #just 1 trajectory

#Plot pseudotime on UMAP
plot(reducedDims(roj_rds_sce)$UMAP, col = colors_subtypes[as.factor(roj_rds_sce$svm_annotation)], pch = 16,  asp = 1)
lines(SlingshotDataSet(roj_rds_sce), lwd = 4)

# plotGeneCount(curve = SlingshotDataSet(roj_rds_sce), clusters = roj_rds_sce$svm_annotation) # other command for drawing trajectory on UMAP

#transfer trajectories back to the seurat object
roj_rds$slingPseudotime_1 <- roj_rds_sce$slingPseudotime_1
```

```{r visualize HC and AD on UMAP}
#visualize trajectories in HC and AC
roj_rds_hc <- subset(roj_rds, disease == "HC")
roj_rds_ad <- subset(roj_rds, disease == "AD")

FeaturePlot(roj_rds_hc,features = c("slingPseudotime_1")) + ggtitle("HC pseudotime") + scale_color_gradient2(low = "yellow",mid = "yellow",high ="red",na.value = "grey") #x1
FeaturePlot(roj_rds_ad,features = c("slingPseudotime_1")) + ggtitle("AD pseudotime") + scale_color_gradient2(low = "yellow",mid = "yellow",high ="red",na.value = "grey") #x2

#print(x1+x2|x3+x4) #nice way of reordering plots when printing
```

This visualization is independent from the integrated vs non-integrated data because only cells and their pseudotime gets plot, independent from UMAP, just dependent on what data has been used for visualization. 

```{r pseudotime 1 on UMAP}
#plot pseudotime 1 HC
dat <- roj_rds_sce@colData[, c("svm_annotation", "slingPseudotime_1", "disease")]
dat <- as.data.frame(dat[dat$disease == "HC", ])
dat <- dat[!is.na(dat$slingPseudotime_1), ]

ggplot(dat, aes(x = svm_annotation, y = slingPseudotime_1, fill = svm_annotation, color = svm_annotation)) +
    geom_dotplot(binwidth = 0.3, stackdir = "centerwhole", binaxis = "y", stackratio = 0.1) + coord_flip() + theme_classic() +
    scale_fill_manual(values = colors_subtypes) + scale_color_manual(values = colors_subtypes) +
    ggtitle("HC") + labs(x = NULL, y = NULL) + theme(legend.position = "none")

#plot pseudotime 1 AD
dat <- roj_rds_sce@colData[, c("svm_annotation", "slingPseudotime_1", "disease")]
dat <- as.data.frame(dat[dat$disease == "AD", ])
dat <- dat[!is.na(dat$slingPseudotime_1), ]

ggplot(dat, aes(x = svm_annotation, y = slingPseudotime_1, fill = svm_annotation, color = svm_annotation)) +
    geom_dotplot(binwidth = 0.3, stackdir = "centerwhole", binaxis = "y", stackratio = 0.1) + coord_flip() + theme_classic() +
    scale_fill_manual(values = colors_subtypes) + scale_color_manual(values = colors_subtypes) +
    ggtitle("AD") + labs(x = NULL, y = NULL) + theme(legend.position = "none")
```

## Perform differential gene expression analysis along the pseudotime trajectory

Here we use tradeseq to analyze differential expression alongside the pseudotime trajectory.

For doing this, a fitting model is needed to infer a smoothed gene expression along the pseudotime trajectory. We first use evaluateK to  helps us to select a good number of knots to fit on. It is best to do that applying the filter that you will also use for your NB-GAM. 

```{r figure out best number of k to fit the NB-GAM}
set.seed(123)
icMat <- evaluateK(counts = roj_rds_sce@assays@data$counts, sds = SlingshotDataSet(roj_rds_sce), nGenes = 100)
```

Now we use a negative binomial general additive model to infer the smoothed gene expression of the highly variable genes with the knots we have determined before. 
You might want to play around with the filtering, maybe you also want to take the overlap of highly variable expressed genes with highly expressed genes or only use the highly expressed genes
-> reason is that you want to get rid of "noise" in the heatmap, mean genes that dont change alongside the trajectory 
-> I chose to display the highly variable genes but kept the code for overlaying it with the highly expressed genes in. 

```{r select highly expressed genes}
# counts <- as.data.frame(roj_rds@assays$RNA@counts)
# 
# #calculate log10 transformed counts for all genes
# gene_expr <- data.frame(gene_expr = rowSums(counts)) #calculates the genes total counts across all cells
# gene_expr$log10_counts <- log10(gene_expr$gene_expr + 1) #log10 transform that number 
# gene_expr$gene_expr <- NULL
# 
# #obtain percentage of cells expressing the gene 
# gene_expr$percent_cell <- apply(counts, 1, function(x){length(which(x != 0)) / ncol(counts)}) #margin = 1 indicates rows, #how many cells have expression > 0 
# 
# #set count and % cells expressing thresholds and indicate them by assigned color
# #current threshold: log10 counts > 3.1 and > 0.2 % of cells express the gene 
# gene_expr$highly_expressed <- "no"
# gene_expr[gene_expr$log10_counts > 3.1 & gene_expr$percent_cell > 0.2, ]$highly_expressed <- "yes"
# color <- c("black","red")
# names(color) <- c("no","yes")
# 
# #plot
# library(ggrepel)
# ggplot(data = gene_expr, aes(x = log10_counts, y = percent_cell, col = highly_expressed)) + geom_point() + 
#   scale_colour_manual(values = color) + 
#   geom_vline(xintercept = 3.1, col = "red") + 
#   geom_hline(yintercept = 0.2, col = "red") +
#   geom_text_repel(aes(label = ifelse(percent_cell > 0.2 & log10_counts > 4.5, rownames(gene_expr), "")), size = 3) + #these genes are labelled
#   theme_light() 
```

```{r calculate NB-GAM}
setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
roj_rds_sce_gam <- readRDS("roj_rds_sce_gam_counts_9knots_varfeaturesONLY.rds")
diffex_trajectory <- readRDS("diffex_trajectory_counts_9knots_varfeaturesONLY.rds")
  
# roj_rds <- FindVariableFeatures(roj_rds)
# 
# #figure out how many cores can be used 
# BPPARAM <- BiocParallel::bpparam() # bpnworkers = number of cores
# BPPARAM$workers <- 6 #use 6 cores, that's also the default
# 
# #calculating a negative binomial generalized additive model to interfere gene expression along the (slingshot calculated) trajectory 
# set.seed(123)
# roj_rds_sce_gam <- fitGAM(counts = as.matrix(roj_rds_hc@assays$RNA@counts), sds = SlingshotDataSet(roj_rds_sce), nknots = 9, parallel = T, genes = roj_rds@assays$RNA@var.features) #genes to fit, default = all
# #genes = intersect(roj_rds@assays$RNA@var.features, rownames(gene_expr[gene_expr$color == "yes", ]))
# # roj_rds_sce@assays@data$counts
# 
# # summary(roj_rds_sce_gam[["KRT6"]])

#statistical test to check if gene expression changes across trajectory
# diffex_trajectory <- associationTest(roj_rds_sce_gam) #lineages = T -> test for all lineages independently, needed if you have several lineages
# diffex_trajectory$gene <- rownames(diffex_trajectory)

# setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
# saveRDS(roj_rds_sce_gam, "roj_rds_sce_gam_counts_9knots_varfeaturesONLY.rds")
# saveRDS(diffex_trajectory, file = "diffex_trajectory_counts_9knots_varfeaturesONLY.rds")
# write.xlsx(diffex_trajectory, file = "diffex_trajectory_counts_9knots_varfeaturesONLY.xlsx")
```

Now we draw a heatmap to visualize gene expression across the trajectory. We use only the highly variable genes for this that are also differential expressed along the trajectory. !As mentioned, try out if you find high background without change and therefore want to refine the selection!

```{r heatmap settings}
library(ComplexHeatmap, verbose = F)

pseudotime_genes <- diffex_trajectory[diffex_trajectory$pvalue < 0.05 & !is.na(diffex_trajectory$pvalue), ]
# pseudotime_genes <- intersect(diffex_trajectory2$gene, rownames(gene_expr[gene_expr$highly_expressed == "yes", ]))

genes_to_anno <- c("KRT5", "KRT14", "KRT15", "ASS1", "MCM3", "MKI67", "AHR", "TFAP2A", "OVOL1", "GRHL3", "TP63", "ZNF750", "ATF3", "KRT1", "KRT10", "KRT6", "KRT16", "FLG", "IVL") #genes to annotate
heatmap_genes <- c(pseudotime_genes$gene, genes_to_anno)
```

First in healthy epidermis

```{r heatmap gene expression differentiation healthy}
#healthy trajectory
roj_rds_hc <- subset(roj_rds, disease == "HC")

heatmap_hc <- as.data.frame(roj_rds_hc@assays$RNA@counts)
heatmap_hc <- heatmap_hc[rownames(heatmap_hc) %in% heatmap_genes, ] # filter for desired genes

#order cells in pseudotime
pseudtime_cells <- as.data.frame(roj_rds_hc@meta.data[ , c("slingPseudotime_1","svm_annotation")]) #obtain pseudotime
heatmap_hc <- heatmap_hc[ , rownames(pseudtime_cells)] #sanity check: remove cells not included in pseudotime
heatmap_hc <- heatmap_hc[ , order(pseudtime_cells$slingPseudotime_1)] # order by pseudotime

#change gene counts to z score and smooths gene expression
heatmap_hc <- t(apply(heatmap_hc, 1, function(x) scale(x))) #1 specifies rows, scale calculates the z score
heatmap_hc <- t(apply(heatmap_hc, 1, function(x){zoo::rollmean(x, 100, align = "center")})) #computes a rolling mean to the rows, this smooths out the outliers
# tried with 20 and 50 but it produces a more patchy heatmap and the expression pattern stay the same 
heatmap_hc <- na.omit(heatmap_hc) #around 9 genes are skipped because they only have 0 counts (in HC),
heatmap_hc <- as.data.frame(heatmap_hc) #otherwise the combining with heatmap_hc later gives weird results

#annotate cell types
svm_annotation_hc <- pseudtime_cells[order(pseudtime_cells$slingPseudotime_1), ]$svm_annotation
svm_annotation_hc <- svm_annotation_hc[50 : (ncol(heatmap_hc) + 49)] #centered rolling mean -> first 50 cells don't have a direct output 

annot_celltype_hc <- HeatmapAnnotation(celltype = svm_annotation_hc, col = list(celltype = colors_subtypes))

#annotate genes of interest
annot_genes <- HeatmapAnnotation(
  GeneOfInterest = anno_mark(at = match(genes_to_anno, rownames(heatmap_hc)),
                             labels = genes_to_anno),
  which = "row"
  ) #show_annotation_name = T
 
#K means with 7 groups
heatmap_trajectory_hc <- draw(Heatmap(heatmap_hc, right_annotation = annot_genes, top_annotation = annot_celltype_hc,
                                      col = colorRamp2(c(min(heatmap_hc), 0, max(heatmap_hc)), c("blue", "lightyellow", "red")),
                                      name = "z-score", 
                                      km = 7, row_km_repeats = 1, row_dend_reorder = F, cluster_rows = T, clustering_method_rows = "single", cluster_row_slices = F, #clustering
                                      row_title_rot = 0, show_row_names = F, show_column_names = F,
                                      cluster_columns  = F))
```

And now in AD epidermis

```{r heatmap gene expression differentiation AD}
#AD trajectory
roj_rds_ad <- subset(roj_rds, disease == "AD")

heatmap_ad <- as.data.frame(roj_rds_ad@assays$RNA@counts)
heatmap_ad <- heatmap_ad[rownames(heatmap_ad) %in% heatmap_genes, ] # filter for desired genes

#order cells in pseudotime
pseudtime_cells <- as.data.frame(roj_rds_ad@meta.data[ , c("slingPseudotime_1","svm_annotation")]) #obtain pseudotime
heatmap_ad <- heatmap_ad[ , rownames(pseudtime_cells)] #sanity check: remove cells not included in pseudotime
heatmap_ad <- heatmap_ad[ , order(pseudtime_cells$slingPseudotime_1)] # order by pseudotime

#change gene counts to z score and smooths gene expression
heatmap_ad <- t(apply(heatmap_ad, 1, function(x) scale(x))) #1 specifies rows, scale calculates the z score
heatmap_ad <- t(apply(heatmap_ad, 1, function(x){zoo::rollmean(x, 100, align = "center")})) #computes a rolling mean to the rows, this smooths out the outliers
# tried with 20 and 50 but it produces a more patchy heatmap and the expression pattern stay the same 
heatmap_ad <- na.omit(heatmap_ad) #around 2 genes are skipped because they only have 0 counts (in ad)
heatmap_ad <- as.data.frame(heatmap_ad) #otherwise the combining with heatmap_hc later gives weird results

#annotate cell types
svm_annotation_ad <- pseudtime_cells[order(pseudtime_cells$slingPseudotime_1), ]$svm_annotation
svm_annotation_ad <- svm_annotation_ad[50 : (ncol(heatmap_ad) + 49)] #centered rolling mean -> first 50 cells don't have a direct output 

annot_celltype_ad <- HeatmapAnnotation(celltype = svm_annotation_ad, col = list(celltype = colors_subtypes))

#annotate genes of interest
annot_genes <- HeatmapAnnotation(
  GeneOfInterest = anno_mark(at = match(genes_to_anno, rownames(heatmap_ad)),
                             labels = genes_to_anno),
  which = "row"
  ) #show_annotation_name = T
 
#K means with 7 groups
heatmap_trajectory_ad <- draw(Heatmap(heatmap_ad, right_annotation = annot_genes, top_annotation = annot_celltype_ad,
                                      col = colorRamp2(c(min(heatmap_ad), 0, max(heatmap_ad)), c("blue", "lightyellow", "red")),
                                      name = "z-score", 
                                      km = 7, row_km_repeats = 1, row_dend_reorder = F, cluster_rows = T, clustering_method_rows = "single", cluster_row_slices = F, #clustering
                                      row_title_rot = 0, show_row_names = F, show_column_names = F,
                                      cluster_columns  = F))
```

And now both together in a side-by-side heatmap 

```{r heatmap gene expression differentiation HC and AD}
setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
readRDS("heatmap_trajectory_hcad.rds")

#match per gene ad and hc dataframe
shared_genes <- intersect(rownames(heatmap_hc), rownames(heatmap_ad)) #only use genes present in both heatmaps
heatmap_hcad <- cbind(heatmap_hc[shared_genes, ], heatmap_ad[shared_genes, ])

#annotate cell types
annot_celltype_hcad <- HeatmapAnnotation(
  celltype = c(svm_annotation_hc, svm_annotation_ad),
  col = list(celltype = colors_subtypes)
  )
  
#annotate genes of interest
annot_genes <- HeatmapAnnotation(
  GeneOfInterest = anno_mark(at = match(genes_to_anno, rownames(heatmap_hcad)),
                             labels = genes_to_anno),
  which = "row"
  ) #show_annotation_name = T

#K means with 7 groups
heatmap_trajectory_hcad <- draw(Heatmap(heatmap_hcad, right_annotation = annot_genes, top_annotation = annot_celltype_hcad, 
                                      col = colorRamp2(c(min(heatmap_ad), 0, max(heatmap_ad)), c("blue", "lightyellow", "red")),
                                      name = "z-score", 
                                      km = 7, row_km_repeats = 1, row_dend_reorder = F, cluster_rows = T, clustering_method_rows = "single", cluster_row_slices = F, #clustering
                                      row_title_rot = 0, show_row_names = F, show_column_names = F,
                                      cluster_columns  = F)) 

setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
saveRDS(heatmap_trajectory_hcad, file = "heatmap_trajectory_hcad.rds")
```

```{r quick GO check heatmap HC and AD}
c7_GO <- enrichGO(gene = c7, OrgDb = org.Hs.eg.db, universe = row.names(roj_rds@assays$RNA@counts), keyType = 'SYMBOL', ont = "ALL", readable = T)

c7_GOdf <- as.data.frame(c7_GO)

c4_GO <- enrichGO(gene = c4, OrgDb = org.Hs.eg.db, universe = row.names(roj_rds@assays$RNA@counts), keyType = 'SYMBOL', ont = "ALL", readable = T)

c4_GOdf <- as.data.frame(c4_GO)
View(c4_GOdf)
```

We see the a big influence of cell cycle genes. We therefore extract clusters of stem cell (before cell cycle) and late epidermal development genes. 

```{r dissect heatmap into early and late epidermal development genes}
summary(row_order(heatmap_trajectory_hcad))

c1 <- heatmap_hcad[row_order(heatmap_trajectory_hcad)[[1]], ]
c2 <- heatmap_hcad[row_order(heatmap_trajectory_hcad)[[2]], ]
c4 <- heatmap_hcad[row_order(heatmap_trajectory_hcad)[[4]], ]
c7 <- heatmap_hcad[row_order(heatmap_trajectory_hcad)[[7]], ] #k means: $`1`, split: [[1]]

heatmap_hcad_late <- rbind(c1, c2) 
heatmap_hcad_early <- rbind(c4, c7) 
```

We now plot early development genes. 

```{r optimize clustering heatmap hcad early}
clust_dist <- c("euclidean", "maximum", "manhattan", "canberra", "minkowski", "pearson", "spearman", "kendall", "binary") 
clust_methods <- c("ward.D","ward.D2",  "centroid", "complete", "average", "mcquitty", "median", "single") 
kmeans <- c(1, 2, 3, 4, 5, 6)
heatmap_all_compare <- vector(mode = "list") #creates a list, a list can have different elements as vectors, matrixes, other lists

library(ggplotify)

annot_celltype_hcad <- HeatmapAnnotation(
  celltype = c(svm_annotation_hc, svm_annotation_ad),
  col = list(celltype = colors_subtypes),
  show_annotation_name = F
  )

####clustering distance####
for (method in clust_dist){
  print(method)
  heatmap_all_compare[[method]] <- ggplotify::as.ggplot(
         Heatmap(heatmap_hcad_early, col = colorRamp2(c(min(heatmap_ad), 0, max(heatmap_ad)), c("blue", "lightyellow", "red")),
                 name = "z-score", clustering_distance_rows = method, clustering_method_rows = "single", 
                 row_dend_reorder = F, cluster_rows = T,  cluster_row_slices = F, 
                 row_title_rot = 0, show_row_names = F, show_column_names = F,
                 cluster_columns  = F, show_heatmap_legend = FALSE)
  )
}
  
# clustering_method_rows = "complete",
# km = 7, row_km_repeats = 1, 

setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
pdf("Heatmap_HCAD_early_clusteringDistance.pdf", height = 8, width = 12)
cowplot::plot_grid(plotlist = heatmap_all_compare) #another way of multiplotting
dev.off()  

####clustering method####
for (method in clust_methods){
  print(method)
   heatmap_all_compare[[method]] <- as.ggplot(
     Heatmap(heatmap_hcad_early, col = colorRamp2(c(min(heatmap_ad), 0, max(heatmap_ad)), c("blue", "lightyellow", "red")),
             name = "z-score", clustering_distance_rows = "XXXX", clustering_method_rows = method, 
             row_dend_reorder = F, cluster_rows = T,  cluster_row_slices = F,  
             row_title_rot = 0, show_row_names = F, show_column_names = F,
             cluster_columns  = F, show_heatmap_legend = FALSE)
  )
}

setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
pdf("Heatmap_HCAD_early_clusteringMethod.pdf", height = 8, width = 12)
cowplot::plot_grid(plotlist = heatmap_all_compare) #another way of multiplotting
dev.off()

####k means####
for (k in kmeans){
    print(k)
   heatmap_all_compare[[k]] <- as.ggplot(
     Heatmap(heatmap_hcad_early, top_annotation = annot_celltype_hcad,
             col = colorRamp2(c(min(heatmap_ad), 0, max(heatmap_ad)), c("blue", "lightyellow", "red")),
             name = "z-score", km = k, row_km_repeats = 4, show_heatmap_legend = FALSE,
             row_dend_reorder = F, cluster_rows = T,  cluster_row_slices = F, 
             row_title_rot = 0, show_row_names = F, show_column_names = F,
             cluster_columns  = F)
  )
}

setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
pdf("Heatmap_HCAD_early_kmeans1.pdf", height = 8, width = 12)
cowplot::plot_grid(plotlist = heatmap_all_compare) 
dev.off()
```

```{r heatmap early devlopment}
setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
# heatmap_trajectory_hcad_early_k5 <- readRDS("heatmap_trajectory_hcad_early_k5.rds")
heatmap_trajectory_hcad_early <- readRDS("heatmap_trajectory_hcad_early9.rds")


annot_celltype_hcad <- HeatmapAnnotation(
  celltype = c(svm_annotation_hc, svm_annotation_ad),
  col = list(celltype = colors_subtypes),
  show_annotation_name = T
  )


# genes_to_anno_hcearly <- c("KRT15", "ASS1", "KRT5", "KRT14", 
#                            "COL17A1", "ITGA6", "ITGB1", 
#                            "FOS", "JUN", 
#                            "TP63"
#                            )
# 
# #annotate genes of interest
# annot_genes_hcearly <- HeatmapAnnotation(
#   GeneOfInterest = anno_mark(at = match(genes_to_anno_hcearly, rownames(heatmap_hcad_early)),
#                              labels = genes_to_anno_hcearly),
#   which = "row"
#   ) #show_annotation_name = T
# 
# heatmap_trajectory_hcad_early9 <- draw(
#   Heatmap(heatmap_hcad_early, top_annotation = annot_celltype_hcad, right_annotation = annot_genes_hcearly,
#           col = colorRamp2(c(min(heatmap_ad), 0, max(heatmap_ad)), c("blue", "lightyellow", "red")),
#           name = "z-score", k = 5, row_km_repeats = 1, cluster_columns  = F,
#           row_dend_reorder = F, cluster_rows = T, cluster_row_slices = F,
#           row_title_rot = 0, show_row_names = F, show_column_names = F)
# )

# setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
# saveRDS(object = heatmap_trajectory_hcad_early9, file = "heatmap_trajectory_hcad_early9.rds")
```

```{r GO over enrichment trajectory genes - early}
summary(row_order(heatmap_trajectory_hcad_early))

c1 <- heatmap_hcad_early[row_order(heatmap_trajectory_hcad_early)[[1]], ]
c2 <- heatmap_hcad_early[row_order(heatmap_trajectory_hcad_early)[[2]], ]
c3 <- heatmap_hcad_early[row_order(heatmap_trajectory_hcad_early)[[3]], ]
c4 <- heatmap_hcad_early[row_order(heatmap_trajectory_hcad_early)[[4]], ]
c5 <- heatmap_hcad_early[row_order(heatmap_trajectory_hcad_early)[[5]], ] 

heatmap_hcad_early_ASS1down <- c(rownames(c1), rownames(c2))
heatmap_hcad_early_KRT15down <- c(rownames(c4), rownames(c5))
heatmap_hcad_early_nochange <- c(rownames(c3))

ASS1down_GO <- enrichGO(gene = heatmap_hcad_early_ASS1down, OrgDb = org.Hs.eg.db, universe = row.names(roj_rds@assays$RNA@counts), keyType = 'SYMBOL', ont = "ALL", readable = T)
ASS1down_GOdf <- as.data.frame(ASS1down_GO)
View(ASS1down_GOdf)

KRT15down_GO <- enrichGO(gene = heatmap_hcad_early_KRT15down, OrgDb = org.Hs.eg.db, universe = row.names(roj_rds@assays$RNA@counts), keyType = 'SYMBOL', ont = "ALL", readable = T)
KRT15down_GOdf <- as.data.frame(KRT15down_GO)
View(KRT15down_GOdf)

Nochange_GO <- enrichGO(gene = heatmap_hcad_early_nochange, OrgDb = org.Hs.eg.db, universe = row.names(roj_rds@assays$RNA@counts), keyType = 'SYMBOL', ont = "ALL", readable = T)
Nochange_GOdf <- as.data.frame(Nochange_GO)
View(Nochange_GOdf)

#save
list <- list("Early_ASS1down" = ASS1down_GO,
             "Early_KRT15down" = KRT15down_GO,
             "Early_nochange" = Nochange_GO
             )
write.xlsx(list, file = "GO Heatmap HCADtrajectory early2.xlsx")
```

Here we plot late epidermal development genes. 

```{r optimize clustering heatmap hcad late}
clust_dist <- c("euclidean", "maximum", "manhattan", "canberra", "minkowski", "pearson", "spearman", "binary") 
clust_methods <- c("ward.D","ward.D2",  "centroid", "complete", "average", "mcquitty", "median", "single") 
kmeans <- c(1, 2, 3, 4, 5, 6)
heatmap_all_compare <- vector(mode = "list") #creates a list, a list can have different elements as vectors, matrixes, other lists

annot_celltype_hcad <- HeatmapAnnotation(
  celltype = c(svm_annotation_hc, svm_annotation_ad),
  col = list(celltype = colors_subtypes),
  show_legend = F
  )

library(ggplotify)

####clustering distance####
for (method in clust_dist){
  print(method)
  heatmap_all_compare[[method]] <- ggplotify::as.ggplot(
         Heatmap(heatmap_hcad_late, top_annotation = annot_celltype_hcad, 
                                      col = colorRamp2(c(min(heatmap_ad), 0, max(heatmap_ad)), c("blue", "lightyellow", "red")),
                                      name = "z-score", clustering_distance_rows = method, clustering_method_rows = "single", 
                                      row_dend_reorder = F, cluster_rows = T,  cluster_row_slices = F, 
                                      row_title_rot = 0, show_row_names = F, show_column_names = F,
                                      cluster_columns  = F, show_heatmap_legend = FALSE)
  )
}
  
# clustering_method_rows = "complete",
# km = 7, row_km_repeats = 1, 

setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
pdf("Heatmap_HCAD_late_clusteringDistance.pdf", height = 8, width = 12)
cowplot::plot_grid(plotlist = heatmap_all_compare) #another way of multiplotting
dev.off()  

####clustering method####
for (method in clust_methods){
   print(method)
   heatmap_all_compare[[method]] <- as.ggplot(
     Heatmap(heatmap_hcad_late, right_annotation = annot_genes, top_annotation = annot_celltype_hcad, 
                                      col = colorRamp2(c(min(heatmap_ad), 0, max(heatmap_ad)), c("blue", "lightyellow", "red")),
                                      name = "z-score", clustering_distance_rows = "XXXX", clustering_method_rows = method, 
                                      row_dend_reorder = F, cluster_rows = T,  cluster_row_slices = F, 
                                      row_title_rot = 0, show_row_names = F, show_column_names = F,
                                      cluster_columns  = F, show_heatmap_legend = FALSE)
  )
}

setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
pdf("Heatmap_HCAD_late_clusteringMethod.pdf", height = 8, width = 12)
cowplot::plot_grid(plotlist = heatmap_all_compare) #another way of multiplotting
dev.off()

####k means####
for (k in kmeans){
  print (k)
   heatmap_all_compare[[k]] <- as.ggplot(
     Heatmap(heatmap_hcad_late, top_annotation = annot_celltype_hcad, 
             col = colorRamp2(c(min(heatmap_ad), 0, max(heatmap_ad)), c("blue", "lightyellow", "red")),
             name = "z-score", km = k, row_km_repeats = 4,
             row_dend_reorder = F, cluster_rows = T,  cluster_row_slices = F, 
             row_title_rot = 0, show_row_names = F, show_column_names = F,
             cluster_columns  = F, show_heatmap_legend = FALSE)
  )
}

setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
pdf("Heatmap_HCAD_late_kmeans.pdf", height = 8, width = 12)
cowplot::plot_grid(plotlist = heatmap_all_compare) 
dev.off()
```

```{r heatmap late devlopment}
# setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
# heatmap_trajectory_hcad_late <- readRDS("heatmap_trajectory_hcad_late9.rds")

annot_celltype_hcad <- HeatmapAnnotation(
  celltype = c(svm_annotation_hc, svm_annotation_ad),
  col = list(celltype = colors_subtypes),
  show_annotation_name = T
  )

genes_to_anno_hclate <- c("KRT1", "KRT10", "GRHL3","DSC1", "DSG1", "FLG", "IVL", "LOR")

#annotate genes of interest
annot_genes_hcelate <- HeatmapAnnotation(
  GeneOfInterest = anno_mark(at = match(genes_to_anno_hclate, rownames(heatmap_hcad_late)),
                             labels = genes_to_anno_hclate),
  which = "row"
  )

heatmap_trajectory_hcad_late9 <- draw(
  Heatmap(heatmap_hcad_late, top_annotation = annot_celltype_hcad, right_annotation = annot_genes_hcelate,
          col = colorRamp2(c(min(heatmap_ad), 0, max(heatmap_ad)), c("blue", "lightyellow", "red")),
          name = "z-score", k = 5, row_km_repeats = 3, cluster_columns  = F,
          row_dend_reorder = F, cluster_rows = T, cluster_row_slices = F,
          row_title_rot = 0, show_row_names = F, show_column_names = F)
)

heatmap_trajectory_hcad_late <- heatmap_trajectory_hcad_late9

# setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
# saveRDS(object = heatmap_trajectory_hcad_late9, file = "heatmap_trajectory_hcad_late9.rds")
```

```{r GO over enrichment trajectory genes - late}
summary(row_order(heatmap_trajectory_hcad_late))

c1 <- heatmap_hcad_late[row_order(heatmap_trajectory_hcad_late)[[1]], ]
c2 <- heatmap_hcad_late[row_order(heatmap_trajectory_hcad_late)[[2]], ]
c3 <- heatmap_hcad_late[row_order(heatmap_trajectory_hcad_late)[[3]], ]
c4 <- heatmap_hcad_late[row_order(heatmap_trajectory_hcad_late)[[4]], ]
c5 <- heatmap_hcad_late[row_order(heatmap_trajectory_hcad_late)[[5]], ] 

heatmap_hcad_late_downc2 <- c(rownames(c2))
heatmap_hcad_late_nochange <- c(rownames(c3))
heatmap_hcad_late_downc4 <- c(rownames(c4))
heatmap_hcad_late_down <- c(rownames(c1), rownames(c2), rownames(c4))
heatmap_hcad_late_up <- c(rownames(c5))

Nochange_GO <- enrichGO(gene = heatmap_hcad_late_nochange, OrgDb = org.Hs.eg.db, universe = row.names(roj_rds@assays$RNA@counts), keyType = 'SYMBOL', ont = "ALL", readable = T)
Nochange_GOdf <- as.data.frame(Nochange_GO)
View(Nochange_GOdf)

Late_downc2_GO <- enrichGO(gene = heatmap_hcad_late_downc2, OrgDb = org.Hs.eg.db, universe = row.names(roj_rds@assays$RNA@counts), keyType = 'SYMBOL', ont = "ALL", readable = T)
Late_downc2_GOdf <- as.data.frame(Late_downc2_GO)
View(Late_downc2_GOdf)

Late_downc4_GO <- enrichGO(gene = heatmap_hcad_late_downc4, OrgDb = org.Hs.eg.db, universe = row.names(roj_rds@assays$RNA@counts), keyType = 'SYMBOL', ont = "ALL", readable = T)
Late_downc4_GOdf <- as.data.frame(Late_downc4_GO)
View(Late_downc4_GOdf)

Late_down_GO <- enrichGO(gene = heatmap_hcad_late_down, OrgDb = org.Hs.eg.db, universe = row.names(roj_rds@assays$RNA@counts), keyType = 'SYMBOL', ont = "ALL", readable = T)
Late_down_GOdf <- as.data.frame(Late_down_GO)
View(Late_down_GOdf)

Late_upGO <- enrichGO(gene = heatmap_hcad_late_up, OrgDb = org.Hs.eg.db, universe = row.names(roj_rds@assays$RNA@counts), keyType = 'SYMBOL', ont = "ALL", readable = T)
Late_upGOdf <- as.data.frame(Late_upGO)
View(Late_upGOdf)

#save
list <- list("Late_nochange" = Nochange_GOdf,
             "Late_downC2" = Late_downc2_GOdf,
             "Late_downC4" = Late_downc4_GOdf,
             "Late_down" = Late_down_GOdf,
             "Late_up" = Late_upGOdf
             )
write.xlsx(list, file = "GO Heatmap HCADtrajectory late.xlsx")
```



# Separation of basal III subtypes

I want to increase the UMAP cluster resolution in a way that I can separate KRT15+ and ASS1+ basal III keratinocytes.

```{r UMAP separate basal III subclusters}
roj_rds <-  readRDS("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/Ward_TP63inAD_compWork/roj_rds_v3_rem1000f.rds")

#Isolate basal 3 cluster
basal3_dake <- subset(roj_rds,ident = "BASAL III")

#Calculate UMAP (10 PCs) using disease integrated data
DefaultAssay(basal3_dake) = "integrated"
basal3_dake <- RunPCA(basal3_dake, npcs = 10, verbose = F)
basal3_dake <- FindNeighbors(basal3_dake, dims = 1:10, verbose = F)
basal3_dake <- FindClusters(basal3_dake, resolution = 0.2, verbose = F)
basal3_dake <- RunUMAP(basal3_dake, dims = 1:10, verbose = F)

#Switch to raw data (for visualization)
DefaultAssay(basal3_dake) = "RNA"
# basal3_dake <- ScaleData(basal3_dake, verbose = F)
DimPlot(basal3_dake)

#Assign Annotations
Idents(basal3_dake) = basal3_dake$seurat_clusters
basal3_dake = RenameIdents(basal3_dake,
  "0" = "Basal III KRT15+",
  '1' = "Basal III ASS1+",
  '2' = "Basal III KRT15+",
  '3' = "Basal III KRT15+",
  '4' = "Basal III ASS1+"
)
basal3_dake$b3_subanno = Idents(basal3_dake)
basal3_dake$b3_subanno_dis = paste0(basal3_dake$b3_subanno, "_", basal3_dake$disease)

as.data.frame(table(basal3_dake$b3_subanno_dis))

#re-integrate back into the rojahn dataset
Idents(roj_rds, WhichCells(basal3_dake)) = basal3_dake$b3_subanno
roj_rds$svm_added_anno <-  Idents(roj_rds)

#rename and reorder svm KRT15/ ASS1 separating annotation

# Idents(roj_rds) <- roj_rds$svm_added_anno
# roj_rds <- RenameIdents(roj_rds,
#     "Basal III KRT15+" = "basal III KRT15+",
#     "Basal III ASS1+" = "basal III ASS1+",
#     "BASAL II" = "basal II",
#     "BASAL I" = "basal I",
#     "BASAL IV" = "basal IV",
#     "Spinosum" = "stratum spinosum",
#     "Stratum Granulosum" = "stratum granulosum")
# levels(roj_rds) #check
# roj_rds@meta.data[["svm_added_anno"]] <- Idents(roj_rds)

roj_rds@meta.data[["svm_added_anno"]] <- factor(roj_rds@meta.data[["svm_added_anno"]], levels = c("basal III KRT15+", "basal III ASS1+", "basal II", "basal I", "basal IV", "stratum spinosum", "stratum granulosum")) #order cell identities

```

```{r visualization}
Idents(roj_rds) <- "svm_added_anno"

DimPlot(roj_rds[, roj_rds$disease == "HC"], cols = colors_subtypesBasIII, repel = T,  label = T,  label.box = T) + NoLegend()
```

##Differences basal III subtypes - HC

```{r DEanalysis global - HC markers, KRT15/ASS1 separated}
diffex_hc_svmadd <- readRDS("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD/diffex_hc_svmadd.rds")

# table(roj_rds$svm_added_anno)
# 
# markers <- FindMarkers(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_added_anno", ident.1 = "Stratum Granulosum") 
# colnames(markers) <- paste("HC_granulosum", colnames(markers), sep = "_")
# markers$gene <- rownames(markers)
# 
# # diffex_hc_svmadd <- data.frame(gene = rownames(markers))
# # diffex_hc_svmadd$pct_global <- markers$HC_KRT15_pct.2
# diffex_hc_svmadd <- merge(x = diffex_hc_svmadd, y = markers, by = "gene", all.x = TRUE, all.y = T)
# diffex_hc_svmadd <- diffex_hc_svmadd[ , -c(ncol(diffex_hc_svmadd)-1, ncol(diffex_hc_svmadd)-4)] #removes 2 columns
# 
# #replaces all NAs in pct.global
# i <- diffex_hc_svmadd[is.na(diffex_hc_svmadd$pct_global), "gene"]
# 
# for(gene_name in i){
#   pct_global <- markers[markers$gene == gene_name, 4] #find in markers
#   index <- which(diffex_hc_svmadd$gene == gene_name) #find row in diffex_hc_svmadd
#   diffex_hc_svmadd$pct_global[index] <- pct_global #replace NA there
# }
# 
# diffex_hc <- diffex_hc %>% rename(HC_basal_III_avg_log2FC =  HC_basalIII_avg_log2FC)
# 
# diffex_hc$HC_basalIII_avg_log2FC

# setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
# saveRDS(diffex_hc_svmadd, file = "diffex_hc_svmadd.rds")
# write.xlsx(diffex_hc_svmadd, file = "diffex_hc_svmadd.xlsx")
```

```{r DEanalysis basal III subtypes HC - DEanalysis}
#krt15 global
krt15_global <- diffex_hc[diffex_hc$HC_KRT15_p_val_adj < 0.05 & !is.na(diffex_hc$HC_KRT15_p_val_adj), ]
krt15_global <- krt15_global[krt15_global$HC_ASS1_p_val_adj > 0.05 | is.na(krt15_global$HC_ASS1_p_val_adj), ]

#ass1 global
ass1_global <- diffex_hc[diffex_hc$HC_ASS1_p_val_adj < 0.05 & !is.na(diffex_hc$HC_ASS1_p_val_adj), ]
ass1_global <- ass1_global[ass1_global$HC_KRT15_p_val_adj > 0.05 | is.na(ass1_global$HC_KRT15_p_val_adj), ]

#krt15 vs ass1 direct comparison
krt15_ass1 <- FindMarkers(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_added_anno", ident.1 = "Basal III KRT15+", ident.2 = "Basal III ASS1+")
krt15_ass1$gene <- rownames(krt15_ass1)
krt15_ass1 <- krt15_ass1[krt15_ass1$p_val_adj < 0.05, ]
```

```{r basal III subtypes HC - GO}
#background
counts_sum <- rowSums(roj_rds@assays[["RNA"]])
background <- rownames(roj_rds@assays[["RNA"]][counts_sum > 1])

#krt15
k15up <- c(krt15_global$gene[krt15_global$HC_KRT15_avg_log2FC > 0], krt15_ass1$gene[krt15_ass1$avg_log2FC > 0])  #global krt15 unique + krt15_ass1 associated with krt15
k15up <- unique(k15up)

k15down <- c(krt15_global$gene[krt15_global$HC_KRT15_avg_log2FC < 0]) #krt15_ass1 down are actually associated with ass1
k15down <- unique(k15down)

upk15_all <- enrichGO(gene = k15up, universe = background, pvalueCutoff = 0.1, OrgDb = org.Hs.eg.db, keyType = 'SYMBOL', ont = "ALL")
# upk15_all <- as.data.frame(upk15_all)

downk15_all <- enrichGO(gene = k15down, universe = background, OrgDb = org.Hs.eg.db, keyType = 'SYMBOL', ont = "ALL")
# downk15_all <- as.data.frame(downk15_all)

#ass1
ass1up <- c(ass1_global$gene[ass1_global$HC_ASS1_avg_log2FC > 0], krt15_ass1$gene[krt15_ass1$avg_log2FC < 0])  #global ass1 unique + krt15_ass1 associated with ass1
ass1up <- unique(ass1up)

ass1down <- c(ass1_global$gene[ass1_global$HC_ASS1_avg_log2FC < 0]) #krt15_ass1 down are actually associated with ass1
ass1down <- unique(ass1down)

upass1_all <- enrichGO(gene = ass1up, universe = background, OrgDb = org.Hs.eg.db, keyType = 'SYMBOL', ont = "ALL")
# upass1_all <- as.data.frame(upass1_all)

downass1_all <- enrichGO(gene = ass1down, universe = background, OrgDb = org.Hs.eg.db, keyType = 'SYMBOL', ont = "ALL")
# downass1_all <- as.data.frame(downass1_all)

# #save
# list <- list("DE_KRT15_specific" = diffex_hc[diffex_hc$gene %in% k15up | diffex_hc$gene %in% k15down, ],
#              "GO_KRT15_up" = as.data.frame(upk15_all),
#              "GO_KRT15_down" = as.data.frame(downk15_all),
#              "DE_ASS1_specific" = diffex_hc[diffex_hc$gene %in% ass1up | diffex_hc$gene %in% ass1down, ],
#              "GO_ASS1_up" = as.data.frame(upass1_all),
#              "GO_ASS1_down" = as.data.frame(downass1_all))
# write.xlsx(list, file = "GO_basalIII_HC.xlsx")
```

```{r}
diffex_basalIII <- FindMarkers(roj_rds[ , roj_rds$disease == "HC"], group.by = "svm_added_anno", logfc.threshold = 0, ident.1 = "basal III KRT15+", ident.2 = "basal III ASS1+") #this is NOT DIFF expression!!!
hc <- diffex_basalIII[diffex_basalIII$p_val_adj < 0.05, ] # this is DIFF expression!!!
hc["gene"] <- rownames(hc)

#[hc$avg_log2FC > 0, ]
GO <- enrichGO(
  gene = hc[hc$avg_log2FC > 0, ]$gene,
  OrgDb = org.Hs.eg.db,
  universe = roj_rds@assays[["RNA"]]@counts@Dimnames[[1]],
  keyType = 'SYMBOL',
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff  = 0.05
)

View(as.data.frame(GO))

diffex_basalIII <- FindMarkers(roj_rds[ , roj_rds$disease == "HC"], group.by = "svm_added_anno", logfc.threshold = 0, ident.1 = "basal III KRT15+", ident.2 = "basal III ASS1+") #this is NOT DIFF expression!!!
hc <- diffex_basalIII[diffex_basalIII$p_val_adj < 0.05, ] # this is DIFF expression!!!
hc["gene"] <- rownames(hc)

#[hc$avg_log2FC > 0, ]
GO_ASS1 <- enrichGO(
  gene = hc[hc$avg_log2FC < 0, ]$gene,
  OrgDb = org.Hs.eg.db,
  universe = roj_rds@assays[["RNA"]]@counts@Dimnames[[1]],
  keyType = 'SYMBOL',
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff  = 0.05
)

View(as.data.frame(GO_ASS1))
```

```{r basal III subtypes HC - AHR signature}
AHRsig <- readxl::read_excel("C:/Users/spare/OneDrive - Radboudumc/Results/Pan-tissue AHR signature_Opitz(2020).xlsx") #open table S2 of Sadik et. al 2020
AHRsig <- AHRsig[-c(1:2), ]
colnames(AHRsig) <- c("gene", "entrezID")
AHRsig <- AHRsig$gene

ggvenn::ggvenn(list('AHRsig' = AHRsig, 'KC_KRT15' = c(k15up, k15down)))
AHRsig[AHRsig %in% c(k15up, k15down)]
```

```{r exploring DEanalysis}
#go
GO_KRT15_hc = enrichGO(gene = krt15$gene, OrgDb = org.Hs.eg.db, keyType = 'SYMBOL', ont = "ALL")
show <- as.data.frame(GO_KRT15_hc)

GO_KRT15_hc = enrichGO(gene = krt15$gene, OrgDb = org.Hs.eg.db, keyType = 'SYMBOL', ont = "ALL")
show <- as.data.frame(GO_KRT15_hc)

#gsea
krt15_ <- krt15 %>% dplyr::pull(HC_ASS1_avg_log2FC)
names(krt15_) <- krt15 %>% dplyr::pull(gene)
krt15_ <- sort(krt15_, decreasing = TRUE)

GSEA_KRT15_hc <- GSEA(krt15_, TERM2GENE = GSE_targets, pvalueCutoff = 1, eps = 0,
  maxGSSize = 10000,
  minGSSize = 1
  )


GSEA_KRT15_hc <- gseGO(krt15_, OrgDb = org.Hs.eg.db, ont = "ALL", keyType = "SYMBOL", nPermSimple = 10000) #scoreType = "pos"

table(krt15$HC_KRT15_avg_log2FC > 0)
table(krt15$HC_KRT15_avg_log2FC < 0)
```

OLD:

```{r DEanalysis basalIII in basal3_dake subset}
# #separating all basal III subclusters based on disease condition
# Idents(basal3_dake) <- basal3_dake$disease
# basal3_dake_hc <- subset(basal3_dake, ident = "HC")
# basal3_dake_ad <- subset(basal3_dake, ident = "AD")
# 
# diff_exp_hc <- FindMarkers(basal3_dake_hc, group.by = "b3_subanno", logfc.threshold = 0, ident.1 = "Basal III KRT15+", ident.2 = "Basal III ASS1+") #this is NOT DIFF expression!!!
# diff_exp_ad <- FindMarkers(basal3_dake_ad, group.by = "b3_subanno", logfc.threshold = 0, ident.1 = "Basal III KRT15+", ident.2 = "Basal III ASS1+")
# 
# hc <- diff_exp_hc[diff_exp_hc$p_val_adj < 0.05, ] # this is DIFF expression!!!
# hc["gene"] <- rownames(hc)
# 
# ad <- diff_exp_ad[diff_exp_ad$p_val_adj < 0.05, ]
# ad["gene"] <- rownames(ad)
# 
# #save DE list
# df_list <- list("DE_healthy" = hc, "DE_AD" = ad)
# # write.xlsx(df_list, file = "DE_basalIIIsubtypes.xlsx")
```

## Differences basal III subtypes - checking changes in AD (HC vs AD)

```{r DEanalysis comparison HC vs AD}
#complete
changeAD <- FindMarkers(roj_rds, group.by = "disease", ident.1 = "AD", ident.2 = "HC") 

#krt15
change_krt15 <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "Basal III KRT15+", ident.1 = "AD", ident.2 = "HC") #comparison of KRT15+ changes in AD compared to HC, log2FC up means more expressed in first group (AD)
change_krt15$gene <- rownames(change_krt15)

change_krt15 <- change_krt15[change_krt15$p_val_adj < 0.05, ]
table(change_krt15$avg_log2FC > 0)

krt15changeup <- change_krt15[change_krt15$avg_log2FC > 0, "gene"]
krt15changedown <- change_krt15[change_krt15$avg_log2FC < 0, "gene"]

GO_krt15changeup <- enrichGO(gene = krt15changeup, universe = background, pvalueCutoff = 0.1, OrgDb = org.Hs.eg.db, keyType = 'SYMBOL', ont = "ALL")
# GO_krt15changeup <- as.data.frame(GO_krt15changeup)
GO_krt15changedown <- enrichGO(gene = krt15changedown, universe = background, pvalueCutoff = 0.1, OrgDb = org.Hs.eg.db, keyType = 'SYMBOL', ont = "ALL")
GO_krt15changedown <- as.data.frame(GO_krt15changedown)

#ass1
change_ass1 <- FindMarkers(roj_rds, group.by = "disease", subset.ident = "Basal III ASS1+", ident.1 = "AD", ident.2 = "HC") #comparison of ASS1+ changes in AD compared to HC, log2FC up means more expressed in first group (AD)
change_ass1$gene <- rownames(change_ass1)

change_ass1 <- change_ass1[change_ass1$p_val_adj < 0.05, ]
table(change_ass1$avg_log2FC > 0)

ass1changeup <- change_ass1[change_ass1$avg_log2FC > 0, "gene"]
ass1changedown <- change_ass1[change_ass1$avg_log2FC < 0, "gene"]

GO_ass1changeup <- enrichGO(gene = ass1changeup, universe = background, pvalueCutoff = 0.1, OrgDb = org.Hs.eg.db, keyType = 'SYMBOL', ont = "ALL")
# GO_ass1changeup <- as.data.frame(GO_ass1changeup)
GO_ass1changedown <- enrichGO(gene = ass1changedown, universe = background, pvalueCutoff = 0.1, OrgDb = org.Hs.eg.db, keyType = 'SYMBOL', ont = "ALL")
# GO_ass1changedown <- as.data.frame(GO_ass1changedown)

# save
# list <- list("DE_KRT15_ADvsHC" = change_krt15,
#              "GO_KRT15_ADvsHC_up" = as.data.frame(GO_krt15changeup),
#              "GO_KRT15_ADvsHC_down" = as.data.frame(GO_krt15changedown),
#              "DE_ASS1_ADvsHC" = change_ass1,
#              "GO_ASS1_ADvsHC_up" = as.data.frame(GO_ass1changeup),
#              "GO_ASS1_ADvsHC_down" = as.data.frame(GO_ass1changedown))
# write.xlsx(list, file = "GO_basalIII_ADvsHC.xlsx")
```

```{r DEanalysis global - AD vs HC, KRT15/ASS1 separated}
diffex_ad_hc_svmadd <- readRDS("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD/diffex_ad_svmadd.rds")

# table(roj_rds$svm_added_anno)
# Idents(roj_rds) <- roj_rds$svm_added_anno
# 
# markers <- FindMarkers(roj_rds[, roj_rds$disease == "AD"], group.by = "svm_added_anno", ident.1 = "Stratum Granulosum") 
# colnames(markers) <- paste("AD_granulosum", colnames(markers), sep = "_")
# markers$gene <- rownames(markers)
# 
# # diffex_ad_svmadd <- data.frame(gene = rownames(markers))
# # diffex_ad_svmadd$pct_global <- markers$AD_KRT15_pct.2
# diffex_ad_svmadd <- merge(x = diffex_ad_svmadd, y = markers, by = "gene", all.x = TRUE, all.y = T)
# diffex_ad_svmadd <- diffex_ad_svmadd[ , -c(ncol(diffex_ad_svmadd)-1, ncol(diffex_ad_svmadd)-4)] #removes 2 columns
# 
# #replaces all NAs in pct.global
# i <- diffex_ad_svmadd[is.na(diffex_ad_svmadd$pct_global), "gene"]
# 
# for(gene_name in i){
#   pct_global <- markers[markers$gene == gene_name, 4] #find in markers
#   index <- which(diffex_ad_svmadd$gene == gene_name) #find row in diffex_hc
#   diffex_ad_svmadd$pct_global[index] <- pct_global #replace NA there
# }
# 
# # test <- test %>% rename(HC_basal_II_p_val_adj =  HC_basalII_p_val_adj)
# 
# setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
# saveRDS(diffex_ad_svmadd, file = "diffex_ad_svmadd.rds")
# write.xlsx(diffex_ad_svmadd, file = "diffex_ad_svmadd.xlsx")
```

```{r DEanalysis basalIII}
#separating all basal III subclusters based on disease condition
Idents(roj_rds) <- roj_rds$disease
roj_rds_hc <- subset(roj_rds, ident = "HC")
roj_rds_ad <- subset(roj_rds, ident = "AD")

diff_exp_hc <- FindMarkers(roj_rds_hc, group.by = "svm_added_anno", logfc.threshold = 0, ident.1 = "Basal III KRT15+", ident.2 = "Basal III ASS1+") #this is NOT DIFF expression!!!
diff_exp_ad <- FindMarkers(roj_rds_ad, group.by = "svm_added_anno", logfc.threshold = 0, ident.1 = "Basal III KRT15+", ident.2 = "Basal III ASS1+")

hc <- diff_exp_hc[diff_exp_hc$p_val_adj < 0.05, ] # this is DIFF expression!!!
hc["gene"] <- rownames(hc)
ad <- diff_exp_ad[diff_exp_ad$p_val_adj < 0.05, ]
ad["gene"] <- rownames(ad)

#global
diff_exp_hc_krt15 <- FindMarkers(roj_rds_hc, group.by = "svm_added_anno", ident.1 = "Basal III KRT15+")
diff_exp_hc_ass1 <- FindMarkers(roj_rds_hc, group.by = "svm_added_anno", ident.1 = "Basal III ASS1+")

table(diff_exp_hc_krt15$p_val_adj < 0.05)
table(diff_exp_hc_krt15$avg_log2FC > 1.5)

table(diff_exp_hc_ass1$p_val_adj < 0.05)
table(diff_exp_hc_ass1$avg_log2FC > 1.5)

diff_exp_hc_krt15 <- diff_exp_hc_krt15[diff_exp_hc_krt15$p_val_adj < 0.05, ] # this is DIFF expression!!!
diff_exp_hc_ass1 <- diff_exp_hc_ass1[diff_exp_hc_ass1$p_val_adj < 0.05, ] # this is DIFF expression!!!

#save DE list
df_list <- list("DE_healthy" = hc, "DE_AD" = ad)
# write.xlsx(df_list, file = "DE_basalIIIsubtypes.xlsx")
```

```{r DE overlap healthy + AD}
ggvenn(list("hc" = hc[hc$avg_log2FC > 0, ]$gene, "ad" = ad[ad$avg_log2FC > 0, ]$gene)) + ggtitle("upregulated in KRT15+ basal III")

ggvenn(list("hc" = hc[hc$avg_log2FC < 0, ]$gene, "ad" = ad[ad$avg_log2FC < 0, ]$gene)) + ggtitle("upregulated in ASS1+ basal III")
```

```{r ATF3, Pol II pause break release check }
Upstream pathway proteins: AFF1, HEXM1
Rapid-response genes: ATF3, DUSP1, RND3 
Induced diff TFs: GRHL3, OVOL1, PRDM1, ZNF750
Repressed proliferation genes: MKI67, FOXM1, AURKB

VlnPlot(roj_rds, group.by = "svm_annotation", features = c("ATF3", "DUSP1", "RND3", "GRHL3", "OVOL1", "PRDM1", "ZNF750", "MKI67"), split.by = "disease", stack = T, flip = T) 

```


# TP63 target gene enrichment

```{r in KRT15 / ASS1 cells}
#load targets p63 ChIP-seq
p63targetsall <- readxl::read_excel("embr0016-0863-sd13.xlsx", sheet = "ST12A") #open sup 13, table12 of Kouwenhoven EN, Oti M, Niehues H, van Heeringen SJ, Schalkwijk J, Stunnenberg HG, van Bokhoven H, Zhou H. Transcription factor p63 bookmarks and regulates dynamic enhancers during epidermal differentiation. EMBO Rep. 2015 Jul;16(7):863-78. doi: 10.15252/embr.201439941. Epub 2015 Jun 1. PMID: 26034101; PMCID: PMC4515125.
p63targetsall <- p63targetsall$...10 #select gene list
p63targetsall <- unique(p63targetsall[3:41598]) #Trim header to remove non gene values

#evaluate enrichment in DE genes (DE between KRT15/ASS1 in healthy or AD resp)
krt15_hc_DE <- rownames(diff_exp_hc[diff_exp_hc$avg_log2FC > 0 & diff_exp_hc$p_val_adj < 0.05, ])
ass1_hc_DE <- rownames(diff_exp_hc[diff_exp_hc$avg_log2FC < 0 & diff_exp_hc$p_val_adj < 0.05, ])
krt15_ad_DE <- rownames(diff_exp_ad[diff_exp_ad$avg_log2FC > 0 & diff_exp_ad$p_val_adj < 0.05, ])
ass1_ad_DE <- rownames(diff_exp_ad[diff_exp_ad$avg_log2FC < 0 & diff_exp_ad$p_val_adj < 0.05, ])

table(p63targetsall %in% krt15_hc_DE)
table(p63targetsall %in% ass1_hc_DE)
table(p63targetsall %in% krt15_ad_DE)
table(p63targetsall %in% ass1_ad_DE)
# p63targetsall[p63targetsall %in% krt15_hc_DE == T]

#evaluate enrichment in all KRT15/ASS1 marker genes (for HC or AD KRT15/ASS1 population resp)
krt15_hc <- rownames(diff_exp_hc[diff_exp_hc$avg_log2FC > 0, ])
ass1_hc <- rownames(diff_exp_hc[diff_exp_hc$avg_log2FC < 0, ])
krt15_ad <- rownames(diff_exp_ad[diff_exp_ad$avg_log2FC > 0, ])
ass1_ad <- rownames(diff_exp_ad[diff_exp_ad$avg_log2FC < 0, ])

table(p63targetsall %in% krt15_hc)
table(p63targetsall %in% ass1_hc)
table(p63targetsall %in% krt15_ad)
table(p63targetsall %in% ass1_ad)
as.data.frame(table(basal3_dake$b3_subanno_dis)) #cell numbers krt15/ass1 subpopulations

#background
table(p63targetsall %in% roj_rds@assays[["RNA"]]@counts@Dimnames[[1]])
length(roj_rds@assays[["RNA"]]@counts@Dimnames[[1]])


```

```{r in all HC + AD cells}
#load targets p63 ChIP-seq
p63targetsall <- readxl::read_excel("embr0016-0863-sd13.xlsx", sheet = "ST12A") #open sup 13, table12 of Kouwenhoven EN, Oti M, Niehues H, van Heeringen SJ, Schalkwijk J, Stunnenberg HG, van Bokhoven H, Zhou H. Transcription factor p63 bookmarks and regulates dynamic enhancers during epidermal differentiation. EMBO Rep. 2015 Jul;16(7):863-78. doi: 10.15252/embr.201439941. Epub 2015 Jun 1. PMID: 26034101; PMCID: PMC4515125.
p63targetsall <- p63targetsall$...10 #select gene list
p63targetsall <- unique(p63targetsall[3:41598]) #Trim header to remove non gene values

#define subpopulation marker genes 
HCkrt15_global <- diffex_hc[diffex_hc$HC_KRT15_p_val_adj < 0.05 & !is.na(diffex_hc$HC_KRT15_p_val_adj), ]
HCass1_global <- diffex_hc[diffex_hc$HC_ASS1_p_val_adj < 0.05 & !is.na(diffex_hc$HC_ASS1_p_val_adj), ]
HCbasalII_global <- diffex_hc[diffex_hc$HC_basal_II_p_val_adj < 0.05 & !is.na(diffex_hc$HC_basal_II_p_val_adj), ]
HCbasalI_global <- diffex_hc[diffex_hc$HC_basal_I_p_val_adj < 0.05 & !is.na(diffex_hc$HC_basal_I_p_val_adj), ]
HCbasalIV_global <- diffex_hc[diffex_hc$HC_basal_IV_p_val_adj < 0.05 & !is.na(diffex_hc$HC_basal_IV_p_val_adj), ]
HCspinosum_global <- diffex_hc[diffex_hc$HC_spinosum_p_val_adj < 0.05 & !is.na(diffex_hc$HC_spinosum_p_val_adj), ]
HCgranulosum_global <- diffex_hc[diffex_hc$HC_granulosum_p_val_adj < 0.05 & !is.na(diffex_hc$HC_granulosum_p_val_adj), ]

ADkrt15_global <- diffex_ad[diffex_ad$AD_KRT15_p_val_adj < 0.05 & !is.na(diffex_ad$AD_KRT15_p_val_adj), ]
ADass1_global <- diffex_ad[diffex_ad$AD_ASS1_p_val_adj < 0.05 & !is.na(diffex_ad$AD_ASS1_p_val_adj), ]
ADbasalII_global <- diffex_ad[diffex_ad$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad$AD_basal_II_p_val_adj), ]
ADbasalI_global <- diffex_ad[diffex_ad$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad$AD_basal_I_p_val_adj), ]
ADbasalIV_global <- diffex_ad[diffex_ad$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad$AD_basal_IV_p_val_adj), ]
ADspinosum_global <- diffex_ad[diffex_ad$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad$AD_spinosum_p_val_adj), ]
ADgranulosum_global <- diffex_ad[diffex_ad$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad$AD_granulosum_p_val_adj), ]

#check TP63 target gene enrichment
krt15_p63_hc <- HCkrt15_global[HCkrt15_global$gene %in% p63targetsall, ]
table(krt15_p63_hc$HC_KRT15_avg_log2FC > 0)

#does logFC has to be > 0? krt15 p63 target
krt15_p63_ad <- ADkrt15_global[p63targetsall %in% ADkrt15_global$gene, ]
krt15_p63_ad <- krt15_p63_ad[!is.na(krt15_p63_ad$gene), ]


test <- HCkrt15_global[p63targetsall %in% HCkrt15_global$gene, "gene"]

#evaluate enrichment in DE genes (DE between KRT15/ASS1 in healthy or AD resp)
krt15_hc_DE <- rownames(diff_exp_hc[diff_exp_hc$avg_log2FC > 0 & diff_exp_hc$p_val_adj < 0.05, ])
ass1_hc_DE <- rownames(diff_exp_hc[diff_exp_hc$avg_log2FC < 0 & diff_exp_hc$p_val_adj < 0.05, ])
krt15_ad_DE <- rownames(diff_exp_ad[diff_exp_ad$avg_log2FC > 0 & diff_exp_ad$p_val_adj < 0.05, ])
ass1_ad_DE <- rownames(diff_exp_ad[diff_exp_ad$avg_log2FC < 0 & diff_exp_ad$p_val_adj < 0.05, ])

table(p63targetsall %in% krt15_hc_DE)
table(p63targetsall %in% ass1_hc_DE)
table(p63targetsall %in% krt15_ad_DE)
table(p63targetsall %in% ass1_ad_DE)
# p63targetsall[p63targetsall %in% krt15_hc_DE == T]

#evaluate enrichment in all KRT15/ASS1 marker genes (for HC or AD KRT15/ASS1 population resp)
krt15_hc <- rownames(diff_exp_hc[diff_exp_hc$avg_log2FC > 0, ])
ass1_hc <- rownames(diff_exp_hc[diff_exp_hc$avg_log2FC < 0, ])
krt15_ad <- rownames(diff_exp_ad[diff_exp_ad$avg_log2FC > 0, ])
ass1_ad <- rownames(diff_exp_ad[diff_exp_ad$avg_log2FC < 0, ])

table(p63targetsall %in% krt15_hc)
table(p63targetsall %in% ass1_hc)
table(p63targetsall %in% krt15_ad)
table(p63targetsall %in% ass1_ad)
as.data.frame(table(basal3_dake$b3_subanno_dis)) #cell numbers krt15/ass1 subpopulations

#background
table(p63targetsall %in% roj_rds@assays[["RNA"]]@counts@Dimnames[[1]])
length(roj_rds@assays[["RNA"]]@counts@Dimnames[[1]])
```

# Literature checks, collaborations

```{r vis TRx2 and CARD14}
FeaturePlot(roj_rds, features = c('CARD14', 'TREX2')) # assay = 'RNA'
VlnPlot(roj_rds, group.by = "svm_added_anno" , features = c('CARD14', 'TREX2'), split.by = "disease", flip = T)
```

```{r vis ACOD1}
#Visualization 
VlnPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_added_anno" , features = c("ACOD1", "KRT15", "ASS1"), stack = T, split.by = "disease", flip = T)
VlnPlot(roj_rds[, roj_rds$disease == "AD"], group.by = "svm_added_anno" , features = c("ACOD1"), stack = T, split.by = "disease", flip = T)

ggcells(x = roj_rds,
        mapping = aes(x = descriptive_name, y = ACOD1),
        exprs_values = "logcounts") + 
  geom_violin(fill = 'coral2') + theme_bw() + ggtitle("log transformed ACOD1 expression") +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))
```

```{r vis SPHK2 and target genes}
genes <- c("AHR", "SPHK2", "S1P", "A4GALT", "ASAH2", "CERS1", "CERS6", "SGPL1", "SMPDL3B", "SGMS1", "SPHK2", "SPTLC1", "CYP1A1", "SPTLC2", "CYP1A1", "CYP1B1")
VlnPlot(roj_rds, group.by = "svm_added_anno" , features = genes, stack = T, split.by = "disease", split.plot = TRUE, flip = T)

#show overlap of both gene expression in UMAP, co-expression
FeaturePlot(roj_rds[, roj_rds$disease == "AD"], features = c("AHR", "SPTLC1", "SGMS1", "SPTLC2")) # assay = 'RNA'
FeaturePlot(roj_rds[, roj_rds$disease == "AD"], features = c("AHR", "SPTLC2"), cols = c("red", "blue"), blend = T)
FeaturePlot(roj_rds[, roj_rds$disease == "AD"], features = c("SPTLC1", "SGMS1"), cols = c("red", "blue"), blend = T)
FeaturePlot(roj_rds[, roj_rds$disease == "AD"], features = c("AHR", "CYP1B1"), cols = c("red", "green"), blend = T)
```

```{r literature check ATF3 in healthy}
#enrichment rapid-response genes
RRG <- readxl::read_excel("C:/Users/spare/OneDrive - Radboudumc/Literature, knowledge/Skin/KC differentiation/raid-response genes.xlsx")
colnames(RRG) <- RRG[3, ]
RRG <- RRG[-c(1:3), ]

genes <- c("ATF3", "DUSP1", "RND3", "GRHL3", "ZNF750", "OVOL1", "PRDM1", "AFF1", "HEXIM1", "MKI67", "FOXM1","AURKB") 
VlnPlot(roj_rds, group.by = "svm_added_anno" , features = c("ATF3", "DUSP1", "RND3", "GRHL3", "ZNF750", "OVOL1", "PRDM1", "AFF1", "HEXIM1", "CDK9", "MKI67", "FOXM1","AURKB"), stack = T,  flip = T, split.by = "disease", split.plot = TRUE,) #split.by = "disease", split.plot = TRUE,

FeaturePlot(roj_rds[, roj_rds$disease == "HC"], features = c("ATF3", "DUSP1", "RND3")) # assay = 'RNA'
VlnPlot(roj_rds[, roj_rds$disease == "HC"], group.by = "svm_added_anno" , features = c("ATF3", "DUSP1", "RND3"), stack = T,  flip = T) 

#gene marker lists
Idents(roj_rds) <- roj_rds$disease
roj_rds_hc <- subset(roj_rds, ident = "HC")

diff_exp_hc_krt15 <- FindMarkers(roj_rds_hc, group.by = "svm_added_anno", ident.1 = "Basal III KRT15+")
diff_exp_hc_ass1 <- FindMarkers(roj_rds_hc, group.by = "svm_added_anno", ident.1 = "Basal III ASS1+")
diff_exp_hc_basalI <- FindMarkers(roj_rds_hc, group.by = "svm_added_anno", ident.1 = "BASAL I")
diff_exp_hc_basalII <- FindMarkers(roj_rds_hc, group.by = "svm_added_anno", ident.1 = "BASAL II")
diff_exp_hc_basalIV <- FindMarkers(roj_rds_hc, group.by = "svm_added_anno", ident.1 = "BASAL IV")
diff_exp_hc_gran <- FindMarkers(roj_rds_hc, group.by = "svm_added_anno", ident.1 = "Stratum Granulosum")

diff_exp_hc_krt15 <- diff_exp_hc_krt15[diff_exp_hc_krt15$p_val_adj < 0.05, ] # this is DIFF expression!!!
# diff_exp_hc_krt15["gene"] <- rownames(diff_exp_hc_krt15)
diff_exp_hc_ass1 <- diff_exp_hc_ass1[diff_exp_hc_ass1$p_val_adj < 0.05, ] # this is DIFF expression!!!
diff_exp_hc_basalI <- diff_exp_hc_basalI[diff_exp_hc_basalI$p_val_adj < 0.05, ]
diff_exp_hc_basalII <- diff_exp_hc_basalII[diff_exp_hc_basalII$p_val_adj < 0.05, ]
diff_exp_hc_basalIV <- diff_exp_hc_basalIV[diff_exp_hc_basalIV$p_val_adj < 0.05, ]
diff_exp_hc_gran <- diff_exp_hc_gran[diff_exp_hc_gran$p_val_adj < 0.05, ]

ggvenn::ggvenn(list('Rapid-response' = RRG$Gene, 'HC_KRT15' = rownames(diff_exp_hc_krt15)))
ggvenn::ggvenn(list('Rapid-response' = RRG$Gene, 'HC_ASS1' = rownames(diff_exp_hc_ass1)))
ggvenn::ggvenn(list('Rapid-response' = RRG$Gene, 'HC_Basal_I' = rownames(diff_exp_hc_basalI)))
ggvenn::ggvenn(list('Rapid-response' = RRG$Gene, 'HC_Basal_II' = rownames(diff_exp_hc_basalII)))
ggvenn::ggvenn(list('Rapid-response' = RRG$Gene, 'HC_Basal_IV' = rownames(diff_exp_hc_basalIV)))
ggvenn::ggvenn(list('Rapid-response' = RRG$Gene, 'HC_StrGran' = rownames(diff_exp_hc_gran)))


RRG$Gene[RRG$Gene %in% rownames(diff_exp_hc_krt15)]
c("ATF3") %in% RRG$Gene[RRG$Gene %in% rownames(diff_exp_hc_krt15)]
c("ATF3") %in% RRG$Gene[RRG$Gene %in% rownames(diff_exp_hc_ass1)]
c("ATF3") %in% RRG$Gene[RRG$Gene %in% rownames(diff_exp_hc_basalI)]
c("ATF3") %in% RRG$Gene[RRG$Gene %in% rownames(diff_exp_hc_basalII)]
RRG$Gene[RRG$Gene %in% rownames(diff_exp_hc_basalIV)]
c("ATF3") %in% RRG$Gene[RRG$Gene %in% rownames(diff_exp_hc_basalIV)]
c("ATF3") %in% RRG$Gene[RRG$Gene %in% rownames(diff_exp_hc_gran)]

Idents(roj_rds) <- roj_rds$svm_annotation
VlnPlot(roj_rds[ , roj_rds$disease == "HC"], features = c("ATF3", "DUSP1", "RND3"), split.by = "disease", stack = T, flip = T)
Idents(roj_rds) <- roj_rds$svm_added_anno
VlnPlot(roj_rds[ , roj_rds$disease == "HC"], features = c("ATF3", "DUSP1", "RND3"), split.by = "disease", stack = T, flip = T)
```

```{r literature check ATF3 in AD}
#enrichment rapid-response genes
RRG <- readxl::read_excel("C:/Users/spare/OneDrive - Radboudumc/Literature, knowledge/Skin/KC differentiation/raid-response genes.xlsx")
colnames(RRG) <- RRG[3, ]
RRG <- RRG[-c(1:3), ]

Idents(roj_rds) <- roj_rds$svm_annotation
VlnPlot(roj_rds, features = c("ATF3", "DUSP1", "RND3"), split.by = "disease", stack = T, flip = T)
VlnPlot(roj_rds, features = c("OVOL1", "GRHL3", "ZNF750", "PRDM1"), split.by = "disease", stack = T, flip = T)

Idents(roj_rds) <- roj_rds$svm_annotation
VlnPlot(roj_rds, features = c("ATF3", "DUSP1", "RND3"), split.by = "disease", stack = T, flip = T)
VlnPlot(roj_rds, features = c("OVOL1", "GRHL3", "ZNF750", "PRDM1"), split.by = "disease", stack = T, flip = T)
VlnPlot(roj_rds, features = c("MKI67", "FOXM1", "AURKB"), split.by = "disease", stack = T, flip = T)
VlnPlot(roj_rds, features = c("AFF1", "HEXIM1"), split.by = "disease", stack = T, flip = T)

VlnPlot(roj_rds, features = c("PRKCA", "PRKCB", "PRKCG", "PRKCD", "PRKCE","PRKCH","PRKCQ"), split.by = "disease", assay = "RNA", stack = T, flip = T)

FeaturePlot(roj_rds, features = c("ATF3"), split.by = "disease") # assay = 'RNA'

#background
counts_sum <- rowSums(roj_rds@assays[["RNA"]])
background <- rownames(roj_rds@assays[["RNA"]][counts_sum > 1])

RRG[!(RRG$Gene %in% background), "Gene"]
RRGexpr <- RRG[!RRG$Gene %in% c("LINC00673", "LOC284454"), ]
gene <- RRGexpr$Gene
RRGexpr$Gene <- NULL
RRGexpr$`KL1 Log2(FoldChange)` <- as.numeric(RRGexpr$`KL1 Log2(FoldChange)`)
RRGexpr$`KL2 Log2(FoldChange)` <- as.numeric(RRGexpr$`KL2 Log2(FoldChange)`)
RRGexpr$mean <- rowMeans(RRGexpr)
RRGexpr$Gene <- gene

#enrichment of RRGs in DE genes
diffexp_ad_basIII <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_III_p_val_adj), "gene"]
diffexp_ad_basII <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_II_p_val_adj), "gene"]
diffexp_ad_basI <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_I_p_val_adj), "gene"]
diffexp_ad_basIV <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_IV_p_val_adj), "gene"]
diffexp_ad_spin <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_spinosum_p_val_adj), "gene"]
diffexp_ad_gran <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_granulosum_p_val_adj), "gene"]

ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Basal III: DE in AD' = diffexp_ad_basIII), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Basal II: DE in AD' = diffexp_ad_basII), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Basal I: DE in AD' = diffexp_ad_basI), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Basal IV: DE in AD' = diffexp_ad_basIV), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Stratum spinosum: DE in AD' = diffexp_ad_spin), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Stratum granulosum: DE in AD' = diffexp_ad_gran), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'all expressed genes' = background), show_percentage =  F)

#enrichment of RRGs in DE upregulated genes
diffexp_ad_basIIIup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_III_avg_log2FC > 0 & diffex_ad_hc_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_III_p_val_adj),  "gene"]
diffexp_ad_basIIup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_II_avg_log2FC > 0 & diffex_ad_hc_svm$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_II_p_val_adj), "gene"]
diffexp_ad_basIup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_I_avg_log2FC > 0 & diffex_ad_hc_svm$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_I_p_val_adj), "gene"]
diffexp_ad_basIVup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_IV_avg_log2FC > 0 & diffex_ad_hc_svm$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_IV_p_val_adj), "gene"]
diffexp_ad_spinup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_spinosum_avg_log2FC > 0 & diffex_ad_hc_svm$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_spinosum_p_val_adj), "gene"]
diffexp_ad_granup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_granulosum_avg_log2FC > 0 & diffex_ad_hc_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_granulosum_p_val_adj), "gene"]
diffexp_ad_grandown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_granulosum_avg_log2FC < 0 & diffex_ad_hc_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_granulosum_p_val_adj), "gene"]

ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Basal III: DE in AD' = diffexp_ad_basIII), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expr' = RRGexpr$Gene, 'Basal III: diff upregulated in AD' = diffexp_ad_basIIIup), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Basal II: DE in AD' = diffexp_ad_basII), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expr' = RRGexpr$Gene, 'Basal II: diff upregulated in AD' = diffexp_ad_basIIup), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Basal I: DE in AD' = diffexp_ad_basI), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expr' = RRGexpr$Gene, 'Basal I: diff upregulated in AD' = diffexp_ad_basIup), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Basal IV: DE in AD' = diffexp_ad_basIV), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expr' = RRGexpr$Gene, 'Basal IV: diff upregulated in AD' = diffexp_ad_basIVup), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Stratum spinosum: DE in AD' = diffexp_ad_spin), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expr' = RRGexpr$Gene, 'Spinosum: diff upregulated in AD' = diffexp_ad_spinup), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'Stratum granulosum: DE in AD' = diffexp_ad_gran), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expr' = RRGexpr$Gene, 'Granulosum: diff upregulated in AD' = diffexp_ad_granup), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expr' = RRGexpr$Gene, 'Granulosum: diff downregulated in AD' = diffexp_ad_grandown), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'all expressed genes' = background), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'all DE genes AD' = rownames(markers)), show_percentage =  F)


ggvenn::ggvenn(list('TP63 targets' = p63targetsall, 'all expressed genes' = background), show_percentage =  F)

#TP63 targets?
ggvenn::ggvenn(list('TP63 targets' = p63targetsall, 'RRGs in diff upregulated basal III' = RRGexpr$Gene[RRGexpr$Gene %in% diffexp_ad_basIIIup]), show_percentage =  F)
ggvenn::ggvenn(list('TP63 targets' = p63targetsall, 'RRGs in diff downregulated granulosum' = RRGexpr$Gene[RRGexpr$Gene %in% diffexp_ad_grandown]), show_percentage =  F)
ggvenn::ggvenn(list('TP63 targets' = p63targetsall, 'rapid response genes' = RRGexpr$Gene), show_percentage =  F)


table(RRGexpr$Gene %in% p63targetsall)

ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'all DE genes AD' = rownames(markers)), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'all DE up AD' = rownames(markers)[markers$avg_log2FC > 0]), show_percentage =  F)
ggvenn::ggvenn(list('Rapid-response expressed' = RRGexpr$Gene, 'all DE down AD' = rownames(markers)[markers$avg_log2FC < 0]), show_percentage =  F)


View(markers[markers$avg_log2FC > 0, ])
```

```{r ATF3 - check early differentiation signature}
KL24h <- readxl::read_excel("C:/Users/spare/OneDrive - Radboudumc/Literature, knowledge/Skin/KC differentiation/RNA-seq KL treatment 24h.xlsx", sheet = "24-hour KL all") #opens Suppl data 3 of Lloyd et al 2022
colnames(KL24h) <- KL24h[3, ]
KL24h <- KL24h[-c(1:3), ]

Cadiff <- readxl::read_excel("C:/Users/spare/OneDrive - Radboudumc/Literature, knowledge/Skin/KC differentiation/RNA-seq Cadiff KCs 4 days.xlsx") #opens Suppl data 1 of Lloyd et al 2022
colnames(Cadiff) <- Cadiff[3, ]
Cadiff <- Cadiff[-c(1:3), ]
Cadiff2 <- c(Cadiff$`Cluster I Genes`, Cadiff$`Cluster II Genes`, Cadiff$`Cluster III Genes`)

EarlyDiff <- KL24h[KL24h$Gene %in% Cadiff2, ]

#enrichment of RRGs in DE genes
diffexp_ad_basIII <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_III_p_val_adj), "gene"]
diffexp_ad_basII <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_II_p_val_adj), "gene"]
diffexp_ad_basI <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_I_p_val_adj), "gene"]
diffexp_ad_basIV <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_IV_p_val_adj), "gene"]
diffexp_ad_spin <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_spinosum_p_val_adj), "gene"]
diffexp_ad_gran <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_granulosum_p_val_adj), "gene"]

ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal III: DE in AD' = diffexp_ad_basIII), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal II: DE in AD' = diffexp_ad_basII), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal I: DE in AD' = diffexp_ad_basI), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal IV: DE in AD' = diffexp_ad_basIV), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Stratum spinosum: DE in AD' = diffexp_ad_spin), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Stratum granulosum: DE in AD' = diffexp_ad_gran), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'all expressed genes' = background), show_percentage =  F)

#enrichment of RRGs in DE upregulated genes
diffexp_ad_basIIIup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_III_avg_log2FC > 0 & diffex_ad_hc_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_III_p_val_adj),  "gene"]
diffexp_ad_basIIIdown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_III_avg_log2FC < 0 & diffex_ad_hc_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_III_p_val_adj),  "gene"]
diffexp_ad_basIIup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_II_avg_log2FC > 0 & diffex_ad_hc_svm$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_II_p_val_adj), "gene"]
diffexp_ad_basIIdown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_II_avg_log2FC < 0 & diffex_ad_hc_svm$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_II_p_val_adj), "gene"]
diffexp_ad_basIup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_I_avg_log2FC > 0 & diffex_ad_hc_svm$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_I_p_val_adj), "gene"]
diffexp_ad_basIdown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_I_avg_log2FC < 0 & diffex_ad_hc_svm$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_I_p_val_adj), "gene"]
diffexp_ad_basIVup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_IV_avg_log2FC > 0 & diffex_ad_hc_svm$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_IV_p_val_adj), "gene"]
diffexp_ad_basIVdown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_IV_avg_log2FC < 0 & diffex_ad_hc_svm$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_IV_p_val_adj), "gene"]
diffexp_ad_spinup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_spinosum_avg_log2FC > 0 & diffex_ad_hc_svm$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_spinosum_p_val_adj), "gene"]
diffexp_ad_spindown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_spinosum_avg_log2FC < 0 & diffex_ad_hc_svm$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_spinosum_p_val_adj), "gene"]
diffexp_ad_granup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_granulosum_avg_log2FC > 0 & diffex_ad_hc_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_granulosum_p_val_adj), "gene"]
diffexp_ad_grandown <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_granulosum_avg_log2FC < 0 & diffex_ad_hc_svm$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_granulosum_p_val_adj), "gene"]

ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal III: DE in AD' = diffexp_ad_basIII), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal III: diff upregulated in AD' = diffexp_ad_basIIIup), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal III: diff downregulated in AD' = diffexp_ad_basIIIdown), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal II: DE in AD' = diffexp_ad_basII), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal II: diff upregulated in AD' = diffexp_ad_basIIup), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal II: diff downregulated in AD' = diffexp_ad_basIIdown), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal I: DE in AD' = diffexp_ad_basI), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal I: diff upregulated in AD' = diffexp_ad_basIup), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal I: diff downregulated in AD' = diffexp_ad_basIdown), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal IV: DE in AD' = diffexp_ad_basIV), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal IV: diff upregulated in AD' = diffexp_ad_basIVup), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Basal IV: diff downregulated in AD' = diffexp_ad_basIVdown), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Stratum spinosum: DE in AD' = diffexp_ad_spin), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Spinosum: diff upregulated in AD' = diffexp_ad_spinup), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Spinosum: diff downregulated in AD' = diffexp_ad_spindown), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Stratum granulosum: DE in AD' = diffexp_ad_gran), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Granulosum: diff upregulated in AD' = diffexp_ad_granup), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'Granulosum: diff downregulated in AD' = diffexp_ad_grandown), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'all expressed genes' = background), show_percentage =  F)
ggvenn::ggvenn(list('CDK9 early diff signature' = EarlyDiff$Gene, 'all DE genes AD' = rownames(markers)), show_percentage =  F)


EarlyDiff$Gene[!(EarlyDiff$Gene %in% background)]
EarlyDiff$Gene[EarlyDiff$Gene %in% diffexp_ad_basIIIup]

GO <- enrichGO(
  gene = EarlyDiff$Gene[EarlyDiff$Gene %in% diffexp_ad_basIIIup],
  OrgDb = org.Hs.eg.db,
  universe = background,
  keyType = 'SYMBOL',
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff  = 0.05
)

View(as.data.frame(GO))
selected <- GO[GO$Description == "epithelial cell proliferation", asis = T]
cnetplot(selected)

GO <- enrichGO(
  gene = EarlyDiff$Gene[EarlyDiff$Gene %in% diffexp_ad_grandown],
  OrgDb = org.Hs.eg.db,
  universe = background,
  keyType = 'SYMBOL',
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff  = 0.05
)

View(as.data.frame(GO))
selected <- GO[GO$Description == "epithelial cell proliferation", asis = T]
cnetplot(selected)

p63targetsall[p63targetsall %in% RRG$Gene[RRG$Gene %in% diffexp_ad_basIIIup]]

GO <- enrichGO(
  gene = p63targetsall[p63targetsall %in% RRG$Gene[RRG$Gene %in% diffexp_ad_basIIIup]],
  OrgDb = org.Hs.eg.db,
  universe = background,
  keyType = 'SYMBOL',
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff  = 0.05
)

View(as.data.frame(GO))
```

```{r TP63 KO genes enrichment in RRGs}
p63LOF <- readxl::read_excel("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/Shen - TP63 KO patients can be rescued by PRIMA (2013).xls") #a list of genes with an expression change >1.5 in a microarray assay of a monolayer cell culture of 3 TP63 LOF mutated patient lines. Flaws: 2D vs 3D vs single-cell!, 1.5 FC very high, microarray also captures a limited number of genes
p63LOFexpr <- p63LOF[p63LOF$gene %in% background, ]

#let's check the overlap of those microarray EEC genes with our (potentially TP63 controlled) RRGs
ggvenn::ggvenn(list('Microarray EEC patients' = p63LOF$gene, 'Rapid response genes' = RRG$Gene))
table(p63LOF$gene %in% RRG$Gene)
table(p63LOF$gene %in% background)
TP63RRG <- p63LOF$gene[p63LOF$gene %in% RRG$Gene]
# p63LOF[p63LOF$gene %in% RRG$Gene, ]
# RRG[RRG$Gene %in% TP63RRG, ]

#let's check if the genes that do overlap are upregulated in basal III (or in AD)
# diffexp_ad_basIIIup <- diffex_ad_hc_svm[diffex_ad_hc_svm$AD_basal_III_avg_log2FC > 0 & diffex_ad_hc_svm$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_ad_hc_svm$AD_basal_III_p_val_adj),  "gene"]
BasalIIIupRRG <- RRG$Gene[RRG$Gene %in% diffexp_ad_basIIIup]
ggvenn::ggvenn(list('Microarray EEC patients' = p63LOFexpr$gene, 'RRGs upregulated in AD basal III' = BasalIIIupRRG))
p63LOFexpr$gene[p63LOFexpr$gene %in% BasalIIIupRRG]

diffex_ad_hc_svm[diffex_ad_hc_svm$gene %in% TP63RRG, ]
table(TP63RRG %in% background)

#let's check what TP63LOF genes do in AD (regardless of if they're part of the RRGs)
ggvenn::ggvenn(list('Microarray EEC patients expressed' = p63LOFexpr$gene, 'AD DEGs' = diffex_ad_hc_svm$gene))
ggvenn::ggvenn(list('Microarray EEC patients expressed' = p63LOFexpr$gene, 'AD diff upregulated genes' = diffex_ad_hc_svm$gene[diffex_ad_hc_svm$AD_avg_log2FC > 0]))
ggvenn::ggvenn(list('Microarray EEC patients expressed' = p63LOFexpr$gene, 'AD diff downregulated genes' = diffex_ad_hc_svm$gene[diffex_ad_hc_svm$AD_avg_log2FC < 0]))


#let's check which cell type they are associated with 
diffex_p63LOF_DEAD <- diffex_ad_hc_svm[diffex_ad_hc_svm$gene %in% p63LOFexpr$gene, ]
p63LOF_DEADbasalIII <- diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_III_p_val_adj), "gene"]
p63LOF_DEADbasalII <- diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_II_p_val_adj), "gene"]
p63LOF_DEADbasalI <- diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_I_p_val_adj), "gene"]
p63LOF_DEADbasalIV <- diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_IV_p_val_adj), "gene"]
p63LOF_DEADspin <- diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_spinosum_p_val_adj), "gene"]
p63LOF_DEADgran <- diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_granulosum_p_val_adj), "gene"]

#and are these genes up- or dowregulated by TP63?
table(p63LOF$`Avg fold change`[p63LOF$gene %in% p63LOF_DEADbasalIII] < 1) #if avg FC < 1 then its upregulated in TP63 LOF mutation -> TP63 downregulates gene
table(p63LOF$`Avg fold change`[p63LOF$gene %in% p63LOF_DEADbasalII] < 1)
table(p63LOF$`Avg fold change`[p63LOF$gene %in% p63LOF_DEADbasalI] < 1)
table(p63LOF$`Avg fold change`[p63LOF$gene %in% p63LOF_DEADbasalIV] < 1)
table(p63LOF$`Avg fold change`[p63LOF$gene %in% p63LOF_DEADspin] < 1)
table(p63LOF$`Avg fold change`[p63LOF$gene %in% p63LOF_DEADgran] < 1)

#and let's check what they do there
table(diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_III_p_val_adj), "AD_basal_III_avg_log2FC"] > 0)
table(diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_II_p_val_adj), "AD_basal_II_avg_log2FC"] > 0)
table(diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_I_p_val_adj), "AD_basal_I_avg_log2FC"] > 0)
table(diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_IV_p_val_adj), "AD_basal_IV_avg_log2FC"] > 0)
table(diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_spinosum_p_val_adj), "AD_spinosum_avg_log2FC"] > 0)
table(diffex_p63LOF_DEAD[diffex_p63LOF_DEAD$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_granulosum_p_val_adj), "AD_granulosum_avg_log2FC"] > 0)

#let's check if the genes upregulated by TP63 (EEC microaaray) are also upregulated in AD 
a <- p63LOF$gene[p63LOF$gene %in% p63LOF_DEADbasalIII & p63LOF$`Avg fold change` > 1]
b <- diffex_p63LOF_DEAD$gene[diffex_p63LOF_DEAD$AD_basal_III_avg_log2FC > 0 & diffex_p63LOF_DEAD$AD_basal_III_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_III_p_val_adj)] 
table (a %in% b)
a[a%in%b] 

GO <- enrichGO(
  gene = a[a%in%b],
  OrgDb = org.Hs.eg.db,
  universe = background,
  keyType = 'SYMBOL',
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff  = 0.05
)

a <- p63LOF$gene[p63LOF$gene %in% p63LOF_DEADbasalII & p63LOF$`Avg fold change` > 1]
b <- diffex_p63LOF_DEAD$gene[diffex_p63LOF_DEAD$AD_basal_II_avg_log2FC > 0 & diffex_p63LOF_DEAD$AD_basal_II_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_II_p_val_adj)] 
table (a %in% b)

a <- p63LOF$gene[p63LOF$gene %in% p63LOF_DEADbasalI & p63LOF$`Avg fold change` > 1]
b <- diffex_p63LOF_DEAD$gene[diffex_p63LOF_DEAD$AD_basal_I_avg_log2FC > 0 & diffex_p63LOF_DEAD$AD_basal_I_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_I_p_val_adj)] 
table (a %in% b)

a <- p63LOF$gene[p63LOF$gene %in% p63LOF_DEADbasalIV & p63LOF$`Avg fold change` > 1]
b <- diffex_p63LOF_DEAD$gene[diffex_p63LOF_DEAD$AD_basal_IV_avg_log2FC > 0 & diffex_p63LOF_DEAD$AD_basal_IV_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_basal_IV_p_val_adj)] 
table (a %in% b)

a <- p63LOF$gene[p63LOF$gene %in% p63LOF_DEADspin & p63LOF$`Avg fold change` > 1]
b <- diffex_p63LOF_DEAD$gene[diffex_p63LOF_DEAD$AD_spinosum_avg_log2FC > 0 & diffex_p63LOF_DEAD$AD_spinosum_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_spinosum_p_val_adj)] 
table (a %in% b)

#checking downregulated genes!
a <- p63LOF$gene[p63LOF$gene %in% p63LOF_DEADgran & p63LOF$`Avg fold change` < 1]
b <- diffex_p63LOF_DEAD$gene[diffex_p63LOF_DEAD$AD_granulosum_avg_log2FC < 0 & diffex_p63LOF_DEAD$AD_granulosum_p_val_adj < 0.05 & !is.na(diffex_p63LOF_DEAD$AD_granulosum_p_val_adj)] 
table (a %in% b)
```


```{r TFAP2A expression}
Idents(roj_rds) <- "svm_annotation"
FeaturePlot(roj_rds[, roj_rds$disease == "HC"], features = c('TFAP2A'), ncol = 1)
VlnPlot(roj_rds[, roj_rds$disease == "HC"], cols = colors_subtypes, features = "TFAP2A")

FeaturePlot(roj_rds[, roj_rds$disease == "AD"], features = c('TFAP2A'), ncol = 1)
VlnPlot(roj_rds[, roj_rds$disease == "AD"], cols = colors_subtypes, features = "TFAP2A")

VlnPlot(roj_rds, features = "TFAP2A", split.by = "disease", pt.size = 0)

Idents(roj_rds) <- "disease"
boxplot(roj_rds, features = "TFAP2A", split.by = "disease")
markers <- FindMarkers(roj_rds, group.by = "disease", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0, min.pct = 0.05) #comparison of changes in AD compared to HC, log2FC up means more expressed in first group (AD)
markers$gene <- rownames(markers)

#boxplot TFAP2A expression HC vs AD
Idents(roj_rds) <- "disease"

roj_rds_AP2a <- subset(roj_rds, features = c("TFAP2A")) #feature of interest

#reshape the table to get it in the correct format to create a boxplot
matrix_AP2a <-as.matrix(roj_rds_AP2a@assays[["RNA"]]@data)

library(reshape2)
library(ggplot2)
library(ggpubr)                               

dataframe_with_columns_AP2a <-setNames(melt(matrix_AP2a), c('rows', 'col' ,'expression')) #use matrix as input
dataframe_with_columns_AP2a$disease <- roj_rds_AP2a@meta.data$disease
# dataframe_with_columns_AP2a$test <- colnames(roj_rds_AP2a)
# all(dataframe_with_columns_AP2a$test == dataframe_with_columns_AP2a$col) #true

#create the boxplot
#On the x-axis are the features
#when checking with data pints : geom_jitter(color="black") +
ggplot(dataframe_with_columns_AP2a, aes(rows, expression, fill=disease)) +
  geom_boxplot()+
  stat_compare_means(label = "p.format")  #wilcox.test
  #theme(legend.position = "none")
```

```{r TSLP check}
Idents(roj_rds) <- "svm_annotation"
FeaturePlot(roj_rds, features = c('TSLP'), split.by = "disease", ncol = 1)
VlnPlot(roj_rds, features = c('TSLP'), split.by = "disease", cols = colors_subtypes) + theme(axis.title.x=element_blank())

Idents(roj_rds) <- "disease"
boxplot(roj_rds, features = "TSLP", split.by = "disease")
markers <- FindMarkers(roj_rds, group.by = "disease", ident.1 = "AD", ident.2 = "HC", logfc.threshold = 0, min.pct = 0.05) #comparison of changes in AD compared to HC, log2FC up means more expressed in first group (AD)
markers$gene <- rownames(markers)

#boxplot TFAP2A expression HC vs AD
Idents(roj_rds) <- "disease"

roj_rds_AP2a <- subset(roj_rds, features = c("TSLP")) #feature of interest

#reshape the table to get it in the correct format to create a boxplot
matrix_AP2a <-as.matrix(roj_rds_AP2a@assays[["RNA"]]@data)

library(reshape2)
library(ggplot2)
library(ggpubr)                               

dataframe_with_columns_AP2a <-setNames(melt(matrix_AP2a), c('rows', 'col' ,'expression')) #use matrix as input
dataframe_with_columns_AP2a$disease <- roj_rds_AP2a@meta.data$disease
# dataframe_with_columns_AP2a$test <- colnames(roj_rds_AP2a)
# all(dataframe_with_columns_AP2a$test == dataframe_with_columns_AP2a$col) #true

#create the boxplot
#On the x-axis are the features
#when checking with data pints : geom_jitter(color="black") +
ggplot(dataframe_with_columns_AP2a, aes(rows, expression, fill=disease)) +
  geom_boxplot()+
  stat_compare_means(label = "p.format")  #wilcox.test
  #theme(legend.position = "none")
```

# Evaluating expression of AHR signature genes in healthy / change in AD?

```{r AHRsig}
AHRsig <- readxl::read_excel("C:/Users/spare/OneDrive - Radboudumc/Results/Pan-tissue AHR signature_Opitz(2020).xlsx") #open table S2 of Sadik et. al 2020
AHRsig <- AHRsig[-c(1:2), ]
colnames(AHRsig) <- c("gene", "entrezID")
AHRsig <- AHRsig$gene

ggvenn::ggvenn(list('AHRsig' = AHRsig, 'HC_KRT15' = names(HC_KRT15)))
ggvenn::ggvenn(list('AHRsig' = AHRsig, 'HC_ASS1' = names(HC_ASS1)))
ggvenn::ggvenn(list('AHRsig' = AHRsig, 'AD_KRT15' = names(ad_KRT15)))
ggvenn::ggvenn(list('AHRsig' = AHRsig, 'AD_ASS1' = names(ad_ASS1)))

ggvenn::ggvenn(list('AHRsig' = AHRsig, 'KC_KRT15' = c(k15up, k15down)))
AHRsig[AHRsig %in% c(k15up, k15down)]

AHRsig[AHRsig %in% names(HC_KRT15)]
AHRsig[AHRsig %in% names(HC_ASS1)]
AHRsig[AHRsig %in% names(ad_KRT15)]
AHRsig[AHRsig %in% names(ad_ASS1)]

GSE_targets <- data.frame("term" = c("AHRsignature"),
                          "gene" = AHRsig)

#make ranked gene lists
HC_KRT15 <- hc %>% filter(avg_log2FC > 0) %>% dplyr::pull(avg_log2FC)
names(HC_KRT15) <- hc %>% filter(avg_log2FC > 0) %>% dplyr::pull(gene)
HC_KRT15 <- sort(HC_KRT15, decreasing = TRUE)

HC_ASS1 <- hc %>% filter(avg_log2FC < 0) %>% dplyr::pull(avg_log2FC)
names(HC_ASS1) <- hc %>% filter(avg_log2FC < 0) %>% dplyr::pull(gene)
HC_ASS1 <- sort(HC_ASS1, decreasing = TRUE)

ad_KRT15 <- ad %>% filter(avg_log2FC > 0) %>% dplyr::pull(avg_log2FC)
names(ad_KRT15) <- ad %>% filter(avg_log2FC > 0) %>% dplyr::pull(gene)
ad_KRT15 <- sort(ad_KRT15, decreasing = TRUE)

ad_ASS1 <- ad %>% filter(avg_log2FC < 0) %>% dplyr::pull(avg_log2FC)
names(ad_ASS1) <- ad %>% filter(avg_log2FC < 0) %>% dplyr::pull(gene)
ad_ASS1 <- sort(ad_ASS1, decreasing = TRUE)

#####Run GSEA on the comparison####

GSEA_hcKRT15 <- GSEA(
  HC_KRT15,
  TERM2GENE = GSE_targets,
  pvalueCutoff = 1,
  eps = 0,
  maxGSSize = 10000,
  minGSSize = 1
  )

GSEA_hcASS1 <- GSEA(
  HC_ASS1,
  TERM2GENE = GSE_targets,
  pvalueCutoff = 1,
  eps = 0,
  maxGSSize = 10000,
  minGSSize = 1
  )

GSEA_adKRT15 <- GSEA(
  ad_KRT15,
  TERM2GENE = GSE_targets,
  pvalueCutoff = 1,
  eps = 0,
  maxGSSize = 10000,
  minGSSize = 1
  )

GSEA_adASS1 <- GSEA(
  ad_ASS1,
  TERM2GENE = GSE_targets,
  pvalueCutoff = 1,
  eps = 0,
  maxGSSize = 10000,
  minGSSize = 1
  )

gseaplot(GSEA_hcKRT15, by = "all", title = GSEA_hcKRT15$Description[1], geneSetID = 1)
gseaplot(GSEA_hcASS1, by = "all", title = GSEA_hcASS1$Description[1], geneSetID = 1)
gseaplot(GSEA_adKRT15, by = "all", title = GSEA_adKRT15$Description[1], geneSetID = 1)
gseaplot(GSEA_adASS1, by = "all", title = GSEA_adASS1$Description[1], geneSetID = 1)

GSEA_hcKRT15_df <- as.data.frame(GSEA_hcKRT15)
GSEA_hcASS1_df <- as.data.frame(GSEA_hcASS1)
GSEA_adKRT15_df <- as.data.frame(GSEA_adKRT15)
GSEA_adASS1_df <- as.data.frame(GSEA_adASS1)

```

# Save

```{r save}
setwd("C:/Users/spare/OneDrive - Radboudumc/Results/KCsinAD/FPcomp/KCinAD")
saveRDS(roj_rds, "roj_rds.rds")
saveRDS(basal3_dake, "basal3_dake.rds")
```
